{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TRABALHO BD"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A partir de webscrapping, ir buscar o número de ids escolhido (máx 20) de termos à escolha do utilizador e importar as informações para a base de dados"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "WEBSCRAPPING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 730,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['https://www.ncbi.nlm.nih.gov/gene/?term=a ']\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "\n",
    "def validateInsertedText(item):\n",
    "    return re.match(r'[\\d|a-z|A-Z]+(\\_?[\\d|a-z|A-Z]+)*', item)\n",
    "\n",
    "def url_get(item):\n",
    "    url_list = []\n",
    "    url = \"https://www.ncbi.nlm.nih.gov/gene/?term={}\".format(item)\n",
    "    url_list.append(url)\n",
    "    return url_list\n",
    "\n",
    "query = input(\"Defina o termo que pretende pesquisar [bacteria/yeast/human...] \\n Se o termo tiver mais do que uma palavra, separe por _: \")\n",
    "try:\n",
    "    \n",
    "    if not validateInsertedText(query):\n",
    "        raise Exception\n",
    "    else:   \n",
    "        print(url_get(query))\n",
    "\n",
    "except Exception:\n",
    "    print(f'Failed to obtain erro .l. next time be a goob boy :). This is what you inserted:\"{query}\"')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 713,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests, sys, json\n",
    "content = []\n",
    "for url in url_get(query):\n",
    "    r = requests.get(url)\n",
    "    content.append(r.content)\n",
    "# print(content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 714,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# Parsing the HTML\n",
    "for c in content:\n",
    "    soup = BeautifulSoup(r.content, 'html.parser')\n",
    "    res = soup.get_text()\n",
    "# print(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 715,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['9615',\n",
       " '249359',\n",
       " '403879',\n",
       " '442858',\n",
       " '403811',\n",
       " '403869',\n",
       " '403802',\n",
       " '475738',\n",
       " '442942',\n",
       " '403885',\n",
       " '403424',\n",
       " '474860',\n",
       " '489652',\n",
       " '403418',\n",
       " '403980',\n",
       " '804479',\n",
       " '610255',\n",
       " '403803',\n",
       " '477032',\n",
       " '403955',\n",
       " '404306']"
      ]
     },
     "execution_count": 715,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "existe = re.findall(r'\\s*ID:\\s*\\d*(?=\\D)', res, re.DOTALL)\n",
    "id_genes = ','.join(existe)\n",
    "id_g = re.sub(r'\\s*ID:\\n*\\s+', '', id_genes)\n",
    "id_gene = id_g.split(',')\n",
    "id_gene\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 719,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INVALID\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    number_of_genes = int(input(\"Número de genes a obter (máx: 20): \"))\n",
    "    print(number_of_genes)\n",
    "    genes = id_gene[0:number_of_genes]\n",
    "    print(genes)\n",
    "except Exception:\n",
    "    print(\"INVALID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#BUSCAR LINKS DOS GENES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 661,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['https://www.ncbi.nlm.nih.gov/nuccore/56784',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/66380993',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/66380996',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/8912104',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/8912103',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/69752209',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/66380995',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/60851505',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/64220641']"
      ]
     },
     "execution_count": 661,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import time\n",
    "def url_get_id(item):\n",
    "    url_id = []\n",
    "    for id in genes:\n",
    "        url = f\"https://www.ncbi.nlm.nih.gov/nuccore/{id}\"\n",
    "        url_id.append(url)\n",
    "    return url_id\n",
    "url_get_id(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 662,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "id_content = []\n",
    "for url in url_get_id(genes):\n",
    "    r = requests.get(url)\n",
    "    id_content.append(r.content)\n",
    "# id_content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 663,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       X67302                   369 bp    mRNA    linear   ROD 26-JUL-2016\n",
      "DEFINITION  R.norvegicus mRNA for Oct 2 Pou-homeodomain protein.\n",
      "ACCESSION   X67302\n",
      "VERSION     X67302.1\n",
      "KEYWORDS    homeodomain protein; Oct-2 gene; POU domain protein.\n",
      "SOURCE      Rattus norvegicus (Norway rat)\n",
      "  ORGANISM  Rattus norvegicus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Rattus.\n",
      "REFERENCE   1  (bases 1 to 369)\n",
      "  AUTHORS   Briton-Jones,C.M. and Rodgers,R.J.\n",
      "  TITLE     Cloning and sequencing of POU-Homeo domain gene expressed in rat\n",
      "            ovaries\n",
      "  JOURNAL   Unpublished\n",
      "REFERENCE   2  (bases 1 to 369)\n",
      "  AUTHORS   Rodgers,R.J.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (07-JUL-1992) R.J. Rodgers, Flinders Univ. of South\n",
      "            Australia, Dept. of Medicine, Bedford Park, SA 5042, AUSTRALIA\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..369\n",
      "                     /organism=\"Rattus norvegicus\"\n",
      "                     /mol_type=\"mRNA\"\n",
      "                     /strain=\"Porton-Wistar\"\n",
      "                     /db_xref=\"taxon:10116\"\n",
      "                     /haplotype=\"diploid\"\n",
      "                     /cell_line=\"ovary\"\n",
      "                     /tissue_type=\"ovary\"\n",
      "                     /dev_stage=\"adult cyclic & pregnant\"\n",
      "     gene            <1..>369\n",
      "                     /gene=\"Oct 2\"\n",
      "     CDS             <1..>369\n",
      "                     /gene=\"Oct 2\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"Oct 2 Pou-Homeo domain protein\"\n",
      "                     /protein_id=\"CAA47715.1\"\n",
      "                     /db_xref=\"GOA:Q63388\"\n",
      "                     /db_xref=\"HSSP:1OCT\"\n",
      "                     /db_xref=\"InterPro:IPR000327\"\n",
      "                     /db_xref=\"InterPro:IPR001356\"\n",
      "                     /db_xref=\"InterPro:IPR009057\"\n",
      "                     /db_xref=\"InterPro:IPR010982\"\n",
      "                     /db_xref=\"InterPro:IPR012287\"\n",
      "                     /db_xref=\"InterPro:IPR013847\"\n",
      "                     /db_xref=\"UniProtKB/TrEMBL:Q63388\"\n",
      "                     /translation=\"FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLE\n",
      "                     KWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQ\n",
      "                     KPTSEEILLIAEQLHMEKEVI\"\n",
      "     misc_feature    1..153\n",
      "                     /gene=\"Oct 2\"\n",
      "                     /note=\"Pou-specific domain\"\n",
      "     misc_feature    154..234\n",
      "                     /gene=\"Oct 2\"\n",
      "                     /note=\"linker\"\n",
      "     misc_feature    218..369\n",
      "                     /gene=\"Oct 2\"\n",
      "                     /note=\"Pou-homeo domain\"\n",
      "ORIGIN      \n",
      "        1 ttcacacagg gtgatgtggg cctggccatg ggcaagctct atggcaacga cttcagccaa\n",
      "       61 acgaccattt cccgcttcga ggcccttaac ctgagcttca agaacatgtg taaactcaag\n",
      "      121 cccctcctgg agaagtggct caacgacgca gagactatgt ctgtggattc aagcctaccc\n",
      "      181 agcccaaacc agctgagcag ccccagcctg ggtttcgacg ggctgccggg gcggagacgc\n",
      "      241 aagaagagga ccagcatcga gacgaatgtc cgcttcgcct tagagaagag tttcctagcg\n",
      "      301 aaccagaagc ctacctcaga ggagatcctg ctgatcgcag agcagctgca catggaaaag\n",
      "      361 gaagtgatc\n",
      "//\n",
      "\n",
      "LOCUS       BW778499                 339 bp    mRNA    linear   EST 04-JAN-2011\n",
      "DEFINITION  BW778499 Amphioxus Branchiostoma floridae unpublished cDNA library,\n",
      "            gastrula whole animal Branchiostoma floridae cDNA clone bfga051e13\n",
      "            5', mRNA sequence.\n",
      "ACCESSION   BW778499\n",
      "VERSION     BW778499.1\n",
      "DBLINK      BioSample: SAMN00176279\n",
      "KEYWORDS    EST.\n",
      "SOURCE      Branchiostoma floridae (Florida lancelet)\n",
      "  ORGANISM  Branchiostoma floridae\n",
      "            Eukaryota; Metazoa; Chordata; Cephalochordata; Leptocardii;\n",
      "            Amphioxiformes; Branchiostomatidae; Branchiostoma.\n",
      "REFERENCE   1  (bases 1 to 339)\n",
      "  AUTHORS   Yu,J.K., Satou,Y., Holland,N.D., Shin-i,T., Kohara,Y., Satoh,N.,\n",
      "            Bronner-Fraser,M. and Holland,L.Z.\n",
      "  TITLE     Axial patterning in cephalochordates and the evolution of the\n",
      "            organizer\n",
      "  JOURNAL   Nature 445 (7128), 613-617 (2007)\n",
      "   PUBMED   17237766\n",
      "COMMENT     Contact: Tadasu Shin-i\n",
      "            Genome Biology Lab., Center For Genetic Resource Information\n",
      "            National Institute of Genetics\n",
      "            1111 Yata, Mishima, Shizuoka 411-8540, Japan\n",
      "            Tel: 81-55-981-6856\n",
      "            Fax: 81-55-981-6855\n",
      "            Email: tshini@genes.nig.ac.jp, URL: http://dolphin.lab.nig.ac.jp\n",
      "            If you want to have a cDNA clone for this EST or if you have any\n",
      "            questions, please send an e-mail to Nori Satoh (norisky@oist.jp).\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..339\n",
      "                     /organism=\"Branchiostoma floridae\"\n",
      "                     /mol_type=\"mRNA\"\n",
      "                     /db_xref=\"taxon:7739\"\n",
      "                     /clone=\"bfga051e13\"\n",
      "                     /tissue_type=\"whole animal\"\n",
      "                     /clone_lib=\"SAMN00176279 Amphioxus Branchiostoma floridae\n",
      "                     unpublished cDNA library, gastrula whole animal\"\n",
      "                     /dev_stage=\"gastrula\"\n",
      "ORIGIN      \n",
      "        1 atctataaca cctaccaatg gcggacgcac ctccagccgg taggggaggt tttcgcggcg\n",
      "       61 ggttcgggga tcgcggcagg ggccgtggac ggggccgagg tcgcggcagg ggccgtggac\n",
      "      121 ggggccgcgg aggcaaggag gaaaaggagt ggatgccaat gaccaagctt ggtcgtcttg\n",
      "      181 tgaaagacct gaagatcaag tccctggagg atatctacct gttctcactt cccatcaagg\n",
      "      241 agtttgagat cattgacttt ttcctgggtt cggcactgaa ggatgaggtg ctgaagatca\n",
      "      301 tgcctgttca gaagcagacc cgtgccggtc agcgcacca\n",
      "//\n",
      "\n",
      "LOCUS       BW778502                 451 bp    mRNA    linear   EST 04-JAN-2011\n",
      "DEFINITION  BW778502 Amphioxus Branchiostoma floridae unpublished cDNA library,\n",
      "            gastrula whole animal Branchiostoma floridae cDNA clone bfga051e16\n",
      "            5', mRNA sequence.\n",
      "ACCESSION   BW778502\n",
      "VERSION     BW778502.1\n",
      "DBLINK      BioSample: SAMN00176279\n",
      "KEYWORDS    EST.\n",
      "SOURCE      Branchiostoma floridae (Florida lancelet)\n",
      "  ORGANISM  Branchiostoma floridae\n",
      "            Eukaryota; Metazoa; Chordata; Cephalochordata; Leptocardii;\n",
      "            Amphioxiformes; Branchiostomatidae; Branchiostoma.\n",
      "REFERENCE   1  (bases 1 to 451)\n",
      "  AUTHORS   Yu,J.K., Satou,Y., Holland,N.D., Shin-i,T., Kohara,Y., Satoh,N.,\n",
      "            Bronner-Fraser,M. and Holland,L.Z.\n",
      "  TITLE     Axial patterning in cephalochordates and the evolution of the\n",
      "            organizer\n",
      "  JOURNAL   Nature 445 (7128), 613-617 (2007)\n",
      "   PUBMED   17237766\n",
      "COMMENT     Contact: Tadasu Shin-i\n",
      "            Genome Biology Lab., Center For Genetic Resource Information\n",
      "            National Institute of Genetics\n",
      "            1111 Yata, Mishima, Shizuoka 411-8540, Japan\n",
      "            Tel: 81-55-981-6856\n",
      "            Fax: 81-55-981-6855\n",
      "            Email: tshini@genes.nig.ac.jp, URL: http://dolphin.lab.nig.ac.jp\n",
      "            If you want to have a cDNA clone for this EST or if you have any\n",
      "            questions, please send an e-mail to Nori Satoh (norisky@oist.jp).\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..451\n",
      "                     /organism=\"Branchiostoma floridae\"\n",
      "                     /mol_type=\"mRNA\"\n",
      "                     /db_xref=\"taxon:7739\"\n",
      "                     /clone=\"bfga051e16\"\n",
      "                     /tissue_type=\"whole animal\"\n",
      "                     /clone_lib=\"SAMN00176279 Amphioxus Branchiostoma floridae\n",
      "                     unpublished cDNA library, gastrula whole animal\"\n",
      "                     /dev_stage=\"gastrula\"\n",
      "ORIGIN      \n",
      "        1 tctaagaaca ttgtaaggaa tcacagcttg tatctccgaa tctaagacgt cttcgttata\n",
      "       61 cagagtctcg ggaccacaac tggtaaacaa agatgtcatc agaagaagaa gactacctgt\n",
      "      121 cccccaggtc cacaagtatc agtaacgggg agagcagttt ctacaccagc gctcaaacgc\n",
      "      181 acctgtctga ctcacaggac gacgcagacc tgaagttctc aggccggtcc aagctgagtg\n",
      "      241 aagacagtat tgatgagttt atgagcagct gtggaaccta cgtctcctat aatgcacggt\n",
      "      301 attacaatgg acaagtcatc accaccttgt cgtcaaagaa cacgctgttc cgggcagagt\n",
      "      361 ttgacctgca acactgtata caggagctgc atgatgcaag aatacagatt cccagtgacg\n",
      "      421 acgtctctgc ggccatcgcc agcatcctgg g\n",
      "//\n",
      "\n",
      "LOCUS       BB232249                 291 bp    mRNA    linear   EST 04-JUL-2000\n",
      "DEFINITION  BB232249 RIKEN full-length enriched, 3 days neonate thymus Mus\n",
      "            musculus cDNA clone A630036M17 3' similar to AF236887 Mus musculus\n",
      "            protein kinase ATR (Atr) mRNA, mRNA sequence.\n",
      "ACCESSION   BB232249\n",
      "VERSION     BB232249.1\n",
      "DBLINK      BioSample: SAMN00159912\n",
      "KEYWORDS    EST.\n",
      "SOURCE      Mus musculus (house mouse)\n",
      "  ORGANISM  Mus musculus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Mus; Mus.\n",
      "REFERENCE   1  (bases 1 to 291)\n",
      "  AUTHORS   Konno,H., Aizawa,K., Akahira,S., Akiyama,J., Arakawa,T.,\n",
      "            Carninci,P., Endo,T., Fukuda,S., Fukunishi,Y., Hara,A., Hayatsu,N.,\n",
      "            Hirozane,T., Hori,F., Ishii,Y., Ishikawa,J., Ishikawa,T., Itoh,M.,\n",
      "            Izawa,M., Kadota,K., Kagawa,I., Kai,C., Kawai,J., Kikuchi,N.,\n",
      "            Kiyosawa,H., Kojima,Y., Kondo,S., Koya,S., Kurihara,C.,\n",
      "            Kusakabe,M., Matsuyama,T., Miki,R., Mizuno,Y., Nakamura,M., Oda,H.,\n",
      "            Okazaki,Y., Ono,T., Owa,C., Saito,H., Sakai,C., Sato,K.,\n",
      "            Shibata,K., Shibata,Y., Shigemoto,Y., Shinagawa,A., Shiraki,T.,\n",
      "            Sogabe,Y., Sugahara,Y., Suzuki,H., Suzuki,H., Tagawa,A.,\n",
      "            Takahashi,F., Tominaga,N., Toya,T., Tsunoda,Y., Watahiki,A.,\n",
      "            Watanabe,S., Yamamura,T., Yamanaka,I., Yano,R., Yasunishi,A.,\n",
      "            Yokota,T., Yoshida,K., Yoshiki,A., Yoshino,M., Muramatsu,M. and\n",
      "            Hayashizaki,Y.\n",
      "  TITLE     RIKEN Mouse ESTs (Konno,H., et al.)\n",
      "  JOURNAL   Unpublished (2000)\n",
      "COMMENT     Contact: Yoshihide Hayashizaki\n",
      "            Omics Science Center, RIKEN Yokohama Institute\n",
      "            The Institute of Physical and Chemical Research (RIKEN)\n",
      "            1-7-22 Suehiro-cho, Tsurumi-ku, Yokohama, Kanagawa 230-0045, Japan\n",
      "            Email: URL:http://www.osc.riken.jp/\n",
      "            Carninci,P., Nishiyama,Y., Westover,A., Itoh,M., Nagaoka,S.,\n",
      "            Sasaki,N., Okazaki,Y., Muramatsu,M. and Hayashizaki,Y.\n",
      "            Thermostabilization and thermoactivation of thermolabile enzymes by\n",
      "            trehalose and its application for the synthesis of full length\n",
      "            cDNA. Proc. Natl. Acad. Sci. U.S.A. 95 (2), 520-524 (1998)\n",
      "             Itoh,M., Kitsunai,T., Akiyama,J., Shibata,K., Izawa,M., Kawai,J.,\n",
      "            Tomaru,Y., Carninci,P., Shibata,Y., Ozawa,Y., Muramatsu,M.,\n",
      "            Okazaki,Y. and Hayashizaki,Y.\n",
      "             Automated filtration-based high-throughput plasmid preparation\n",
      "            system. Genome Res. 9 (5), 463-470 (1999)\n",
      "             Carninci,P. and Hayashizaki,Y.\n",
      "             High-efficiency full-length cDNA cloning. Methods Enzymol. 303,\n",
      "            19-44 (1999)\n",
      "             Please visit our web site (http://genome.rtc.riken.go.jp) for\n",
      "            further details.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..291\n",
      "                     /organism=\"Mus musculus\"\n",
      "                     /mol_type=\"mRNA\"\n",
      "                     /db_xref=\"taxon:10090\"\n",
      "                     /clone=\"A630036M17\"\n",
      "                     /tissue_type=\"thymus\"\n",
      "                     /clone_lib=\"SAMN00159912 RIKEN full-length enriched, 3\n",
      "                     days neonate thymus\"\n",
      "                     /dev_stage=\"3 days neonate\"\n",
      "                     /lab_host=\"DH10B\"\n",
      "                     /note=\"Site_1: SalI; Site_2: BamHI; cDNA library was\n",
      "                     prepared and sequenced in Mouse Genome Encyclopedia\n",
      "                     Project of Genome Exploration Research Group in Riken\n",
      "                     Genomic Sciences Center and Genome Science Laboratory in\n",
      "                     RIKEN. Division of Experimental Animal Research in Riken\n",
      "                     contributed to prepare mouse tissues. 1st strand cDNA was\n",
      "                     primed with a primer [5'\n",
      "                     GAGAGAGAGAAGGATCCAAGAGCTCTTTTTTTTTTTTTTTTVN 3'], cDNA was\n",
      "                     prepared by using trehalose thermo-activated reverse\n",
      "                     transcriptase and subsequently enriched for full-length by\n",
      "                     cap-trapper. cDNA went through one round of normalization\n",
      "                     to Rot = 20.0 and subtraction to Rot = 459.0. Second\n",
      "                     strand cDNA was prepared with the primer adapter of\n",
      "                     sequence [5' GAGAGAGAGATTCTCGAGTTAATTAAATTAATCCCCCCCCCCCCC\n",
      "                     3']. cDNA was cleaved with XhoI and BamHI. Vector: a\n",
      "                     modified pBluescript KS(+) after bulk excision from Lambda\n",
      "                     FLC I.\"\n",
      "ORIGIN      \n",
      "        1 acttctcctg ctaccatact ctcttctctt cccctgatct aggatcagcg catgttctca\n",
      "       61 ggccccactc atgaaagaca ggaggatgcc tcggcacgag taattctcgg aatactttga\n",
      "      121 gtttttcaga ttaagctgct gactttggta gcatacactg gcgaagctcc tttcctttca\n",
      "      181 tatacactcc tacaaagcag aatgatttta attgtttgta cagccaaatg agaaaagaac\n",
      "      241 caagaataat tttgttaaaa tattaaagtt tcaacttttg ctacaactga a\n",
      "//\n",
      "\n",
      "LOCUS       BB232248                 297 bp    mRNA    linear   EST 04-JUL-2000\n",
      "DEFINITION  BB232248 RIKEN full-length enriched, 3 days neonate thymus Mus\n",
      "            musculus cDNA clone A630036M16 3', mRNA sequence.\n",
      "ACCESSION   BB232248\n",
      "VERSION     BB232248.1\n",
      "DBLINK      BioSample: SAMN00159912\n",
      "KEYWORDS    EST.\n",
      "SOURCE      Mus musculus (house mouse)\n",
      "  ORGANISM  Mus musculus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Mus; Mus.\n",
      "REFERENCE   1  (bases 1 to 297)\n",
      "  AUTHORS   Konno,H., Aizawa,K., Akahira,S., Akiyama,J., Arakawa,T.,\n",
      "            Carninci,P., Endo,T., Fukuda,S., Fukunishi,Y., Hara,A., Hayatsu,N.,\n",
      "            Hirozane,T., Hori,F., Ishii,Y., Ishikawa,J., Ishikawa,T., Itoh,M.,\n",
      "            Izawa,M., Kadota,K., Kagawa,I., Kai,C., Kawai,J., Kikuchi,N.,\n",
      "            Kiyosawa,H., Kojima,Y., Kondo,S., Koya,S., Kurihara,C.,\n",
      "            Kusakabe,M., Matsuyama,T., Miki,R., Mizuno,Y., Nakamura,M., Oda,H.,\n",
      "            Okazaki,Y., Ono,T., Owa,C., Saito,H., Sakai,C., Sato,K.,\n",
      "            Shibata,K., Shibata,Y., Shigemoto,Y., Shinagawa,A., Shiraki,T.,\n",
      "            Sogabe,Y., Sugahara,Y., Suzuki,H., Suzuki,H., Tagawa,A.,\n",
      "            Takahashi,F., Tominaga,N., Toya,T., Tsunoda,Y., Watahiki,A.,\n",
      "            Watanabe,S., Yamamura,T., Yamanaka,I., Yano,R., Yasunishi,A.,\n",
      "            Yokota,T., Yoshida,K., Yoshiki,A., Yoshino,M., Muramatsu,M. and\n",
      "            Hayashizaki,Y.\n",
      "  TITLE     RIKEN Mouse ESTs (Konno,H., et al.)\n",
      "  JOURNAL   Unpublished (2000)\n",
      "COMMENT     Contact: Yoshihide Hayashizaki\n",
      "            Omics Science Center, RIKEN Yokohama Institute\n",
      "            The Institute of Physical and Chemical Research (RIKEN)\n",
      "            1-7-22 Suehiro-cho, Tsurumi-ku, Yokohama, Kanagawa 230-0045, Japan\n",
      "            Email: URL:http://www.osc.riken.jp/\n",
      "            Carninci,P., Nishiyama,Y., Westover,A., Itoh,M., Nagaoka,S.,\n",
      "            Sasaki,N., Okazaki,Y., Muramatsu,M. and Hayashizaki,Y.\n",
      "            Thermostabilization and thermoactivation of thermolabile enzymes by\n",
      "            trehalose and its application for the synthesis of full length\n",
      "            cDNA. Proc. Natl. Acad. Sci. U.S.A. 95 (2), 520-524 (1998)\n",
      "             Itoh,M., Kitsunai,T., Akiyama,J., Shibata,K., Izawa,M., Kawai,J.,\n",
      "            Tomaru,Y., Carninci,P., Shibata,Y., Ozawa,Y., Muramatsu,M.,\n",
      "            Okazaki,Y. and Hayashizaki,Y.\n",
      "             Automated filtration-based high-throughput plasmid preparation\n",
      "            system. Genome Res. 9 (5), 463-470 (1999)\n",
      "             Carninci,P. and Hayashizaki,Y.\n",
      "             High-efficiency full-length cDNA cloning. Methods Enzymol. 303,\n",
      "            19-44 (1999)\n",
      "             Please visit our web site (http://genome.rtc.riken.go.jp) for\n",
      "            further details.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..297\n",
      "                     /organism=\"Mus musculus\"\n",
      "                     /mol_type=\"mRNA\"\n",
      "                     /db_xref=\"taxon:10090\"\n",
      "                     /clone=\"A630036M16\"\n",
      "                     /tissue_type=\"thymus\"\n",
      "                     /clone_lib=\"SAMN00159912 RIKEN full-length enriched, 3\n",
      "                     days neonate thymus\"\n",
      "                     /dev_stage=\"3 days neonate\"\n",
      "                     /lab_host=\"DH10B\"\n",
      "                     /note=\"Site_1: SalI; Site_2: BamHI; cDNA library was\n",
      "                     prepared and sequenced in Mouse Genome Encyclopedia\n",
      "                     Project of Genome Exploration Research Group in Riken\n",
      "                     Genomic Sciences Center and Genome Science Laboratory in\n",
      "                     RIKEN. Division of Experimental Animal Research in Riken\n",
      "                     contributed to prepare mouse tissues. 1st strand cDNA was\n",
      "                     primed with a primer [5'\n",
      "                     GAGAGAGAGAAGGATCCAAGAGCTCTTTTTTTTTTTTTTTTVN 3'], cDNA was\n",
      "                     prepared by using trehalose thermo-activated reverse\n",
      "                     transcriptase and subsequently enriched for full-length by\n",
      "                     cap-trapper. cDNA went through one round of normalization\n",
      "                     to Rot = 20.0 and subtraction to Rot = 459.0. Second\n",
      "                     strand cDNA was prepared with the primer adapter of\n",
      "                     sequence [5' GAGAGAGAGATTCTCGAGTTAATTAAATTAATCCCCCCCCCCCCC\n",
      "                     3']. cDNA was cleaved with XhoI and BamHI. Vector: a\n",
      "                     modified pBluescript KS(+) after bulk excision from Lambda\n",
      "                     FLC I.\"\n",
      "ORIGIN      \n",
      "        1 atcctgggca acattaaaaa tagtctacat taggtttttt tccagaaacc ttttgtaaca\n",
      "       61 aacagtatct cgttatgtaa ctccagcacc aagctttgct tgtatttagt caactcagct\n",
      "      121 tggaaatccg cccttgggcc ccactccatt ctcagtccag cacatctagt agcagtgtgg\n",
      "      181 cggaaacact tttatggcaa catggagcag tattagattt tatgtagctg acaatgcagt\n",
      "      241 attaaagctc agtattttgc atatttttag catttacaaa tatttttgct ttagtgc\n",
      "//\n",
      "\n",
      "LOCUS       AAHY01223610             742 bp    DNA     linear   ROD 07-JUL-2005\n",
      "DEFINITION  Mus musculus strain 129X1/SvJ chromosome Un CRA_211000020833787,\n",
      "            whole genome shotgun sequence.\n",
      "ACCESSION   AAHY01223610 AAHY01000000\n",
      "VERSION     AAHY01223610.1\n",
      "DBLINK      BioProject: PRJNA11785\n",
      "            BioSample: SAMN03004379\n",
      "KEYWORDS    WGS.\n",
      "SOURCE      Mus musculus (house mouse)\n",
      "  ORGANISM  Mus musculus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Mus; Mus.\n",
      "REFERENCE   1  (bases 1 to 742)\n",
      "  AUTHORS   Mural,R.J., Adams,M.D., Myers,E.W., Smith,H.O., Miklos,G.L.,\n",
      "            Wides,R., Halpern,A., Li,P.W., Sutton,G.G., Nadeau,J.,\n",
      "            Salzberg,S.L., Holt,R.A., Kodira,C.D., Lu,F., Chen,L., Deng,Z.,\n",
      "            Evangelista,C.C., Gan,W., Heiman,T.J., Li,J., Li,Z., Merkulov,G.V.,\n",
      "            Milshina,N.V., Naik,A.K., Qi,R., Shue,B.C., Wang,A., Wang,J.,\n",
      "            Wang,X., Yan,X., Ye,J., Yooseph,S., Zhao,Q., Zheng,L., Zhu,S.C.,\n",
      "            Biddick,K., Bolanos,R., Delcher,A.L., Dew,I.M., Fasulo,D.,\n",
      "            Flanigan,M.J., Huson,D.H., Kravitz,S.A., Miller,J.R., Mobarry,C.M.,\n",
      "            Reinert,K., Remington,K.A., Zhang,Q., Zheng,X.H., Nusskern,D.R.,\n",
      "            Lai,Z., Lei,Y., Zhong,W., Yao,A., Guan,P., Ji,R.R., Gu,Z.,\n",
      "            Wang,Z.Y., Zhong,F., Xiao,C., Chiang,C.C., Yandell,M.,\n",
      "            Wortman,J.R., Amanatides,P.G., Hladun,S.L., Pratts,E.C.,\n",
      "            Johnson,J.E., Dodson,K.L., Woodford,K.J., Evans,C.A., Gropman,B.,\n",
      "            Rusch,D.B., Venter,E., Wang,M., Smith,T.J., Houck,J.T.,\n",
      "            Tompkins,D.E., Haynes,C., Jacob,D., Chin,S.H., Allen,D.R.,\n",
      "            Dahlke,C.E., Sanders,R., Li,K., Liu,X., Levitsky,A.A.,\n",
      "            Majoros,W.H., Chen,Q., Xia,A.C., Lopez,J.R., Donnelly,M.T.,\n",
      "            Newman,M.H., Glodek,A., Kraft,C.L., Nodell,M., Ali,F., An,H.J.,\n",
      "            Baldwin-Pitts,D., Beeson,K.Y., Cai,S., Carnes,M., Carver,A.,\n",
      "            Caulk,P.M., Center,A., Chen,Y.H., Cheng,M.L., Coyne,M.D.,\n",
      "            Crowder,M., Danaher,S., Davenport,L.B., Desilets,R., Dietz,S.M.,\n",
      "            Doup,L., Dullaghan,P., Ferriera,S., Fosler,C.R., Gire,H.C.,\n",
      "            Gluecksmann,A., Gocayne,J.D., Gray,J., Hart,B., Haynes,J.,\n",
      "            Hoover,J., Howland,T., Ibegwam,C., Jalali,M., Johns,D., Kline,L.,\n",
      "            Ma,D.S., MacCawley,S., Magoon,A., Mann,F., May,D., McIntosh,T.C.,\n",
      "            Mehta,S., Moy,L., Moy,M.C., Murphy,B.J., Murphy,S.D., Nelson,K.A.,\n",
      "            Nuri,Z., Parker,K.A., Prudhomme,A.C., Puri,V.N., Qureshi,H.,\n",
      "            Raley,J.C., Reardon,M.S., Regier,M.A., Rogers,Y.H., Romblad,D.L.,\n",
      "            Schutz,J., Scott,J.L., Scott,R., Sitter,C.D., Smallwood,M.,\n",
      "            Sprague,A.C., Stewart,E., Strong,R.V., Suh,E., Sylvester,K.,\n",
      "            Thomas,R., Tint,N.N., Tsonis,C., Wang,G., Wang,G., Williams,M.S.,\n",
      "            Williams,S.M., Windsor,S.M., Wolfe,K., Wu,M.M., Zaveri,J.,\n",
      "            Chaturvedi,K., Gabrielian,A.E., Ke,Z., Sun,J., Subramanian,G.,\n",
      "            Venter,J.C., Pfannkoch,C.M., Barnstead,M. and Stephenson,L.D.\n",
      "  TITLE     A comparison of whole-genome shotgun-derived mouse chromosome 16\n",
      "            and the human genome\n",
      "  JOURNAL   Science 296 (5573), 1661-1671 (2002)\n",
      "   PUBMED   12040188\n",
      "  REMARK    Erratum:[Science 2002 Aug 23;297(5585):1278]\n",
      "REFERENCE   2  (bases 1 to 742)\n",
      "  AUTHORS   Mural,R.J., Adams,M.D., Myers,E.W., Smith,H.O. and Venter,C.J.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (02-JUN-2005) Celera Genomics, 45 W. Gude Dr., Rockville,\n",
      "            MD 20850, USA\n",
      "COMMENT     This is the July 2001 combined whole genome shotgun assembly of Mus\n",
      "            musculus. It contains 27 million Celera reads on four Mus musculus\n",
      "            strains (129X1/SvJ, 129S1/SvImJ, DBA/2J and A/J), 13 million reads\n",
      "            on C57BL/6J from the NCBI Trace Archive, 0.4 million BAC end\n",
      "            sequences from TIGR, and unfinished and finished BACs pulled from\n",
      "            NCBI (Nature 2002. 420:520-562). The assembly process relied on\n",
      "            Celera's paired reads and BAC end reads for long range order and\n",
      "            orientation. Its scaffolds were mapped to chromosomes using STS\n",
      "            maps. For more detailed information about whole genome sequencing\n",
      "            and Celera's assembly process, please refer to Venter, J.C. et al.\n",
      "            Science 2001. 291:1304-1351.\n",
      "            Annotation was added to the scaffolds in June 2007.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..742\n",
      "                     /organism=\"Mus musculus\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /strain=\"129X1/SvJ; 129S1/SvImJ; DBA/2J; A/J\"\n",
      "                     /db_xref=\"taxon:10090\"\n",
      "                     /chromosome=\"Un\"\n",
      "                     /note=\"on a degenerate scaffold\"\n",
      "ORIGIN      \n",
      "        1 tggaaaataa gaaaaatcca cttgacgact tgaaaaatga cgaaatcact aatagacgtg\n",
      "       61 aaaaatgtga aatgcacact gaaggacctg gaatatggcg agaaaactga aaatcacgga\n",
      "      121 aaatgagaaa tacacacttt aggacgtgaa atatggcgag gaaaactgaa aaaggtggaa\n",
      "      181 aatttagaaa tgtcctctgt aggacgtgaa atatggcaag aaaactgaaa atcacggata\n",
      "      241 atgagaaaca tacacttgac gacttgaaaa atggcgaaat cactaaaaaa cgtgaaaaat\n",
      "      301 gagaaatgca cactgaagga cctggaatat ggcgagaaaa ctgaaaatca aggaaaatga\n",
      "      361 gaaatacaca ctttaggacg tgaaatatgg cgaggaaaac tgaaaaaggt ggaaaattta\n",
      "      421 gaaatgtcca ctgtaggacg tggaatatgg caagaaaact gaaaatcata aaaaatgaga\n",
      "      481 aacatccact tgacgacttg aaaaatgacg aaatcactaa aaaacgtgaa aaatgtgaaa\n",
      "      541 tgcacactga aggacctgga atatggcgag aaaactgaaa atcacggaaa atgagtaata\n",
      "      601 cacactttag gacgtgaaat atggcgagga aaactgaaaa aggtggaaaa tttagaaatg\n",
      "      661 tccactgtac gacgtggaat atggcaagaa cactgaaaat catgggaaaa tgagaaacat\n",
      "      721 ccacttgacg acttgaaaaa tg\n",
      "//\n",
      "\n",
      "LOCUS       BW778501                 336 bp    mRNA    linear   EST 04-JAN-2011\n",
      "DEFINITION  BW778501 Amphioxus Branchiostoma floridae unpublished cDNA library,\n",
      "            gastrula whole animal Branchiostoma floridae cDNA clone bfga051e15\n",
      "            5', mRNA sequence.\n",
      "ACCESSION   BW778501\n",
      "VERSION     BW778501.1\n",
      "DBLINK      BioSample: SAMN00176279\n",
      "KEYWORDS    EST.\n",
      "SOURCE      Branchiostoma floridae (Florida lancelet)\n",
      "  ORGANISM  Branchiostoma floridae\n",
      "            Eukaryota; Metazoa; Chordata; Cephalochordata; Leptocardii;\n",
      "            Amphioxiformes; Branchiostomatidae; Branchiostoma.\n",
      "REFERENCE   1  (bases 1 to 336)\n",
      "  AUTHORS   Yu,J.K., Satou,Y., Holland,N.D., Shin-i,T., Kohara,Y., Satoh,N.,\n",
      "            Bronner-Fraser,M. and Holland,L.Z.\n",
      "  TITLE     Axial patterning in cephalochordates and the evolution of the\n",
      "            organizer\n",
      "  JOURNAL   Nature 445 (7128), 613-617 (2007)\n",
      "   PUBMED   17237766\n",
      "COMMENT     Contact: Tadasu Shin-i\n",
      "            Genome Biology Lab., Center For Genetic Resource Information\n",
      "            National Institute of Genetics\n",
      "            1111 Yata, Mishima, Shizuoka 411-8540, Japan\n",
      "            Tel: 81-55-981-6856\n",
      "            Fax: 81-55-981-6855\n",
      "            Email: tshini@genes.nig.ac.jp, URL: http://dolphin.lab.nig.ac.jp\n",
      "            If you want to have a cDNA clone for this EST or if you have any\n",
      "            questions, please send an e-mail to Nori Satoh (norisky@oist.jp).\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..336\n",
      "                     /organism=\"Branchiostoma floridae\"\n",
      "                     /mol_type=\"mRNA\"\n",
      "                     /db_xref=\"taxon:7739\"\n",
      "                     /clone=\"bfga051e15\"\n",
      "                     /tissue_type=\"whole animal\"\n",
      "                     /clone_lib=\"SAMN00176279 Amphioxus Branchiostoma floridae\n",
      "                     unpublished cDNA library, gastrula whole animal\"\n",
      "                     /dev_stage=\"gastrula\"\n",
      "ORIGIN      \n",
      "        1 aagatcgtgt gttctcgaaa ctgaaggtta taaaaaccat tttcagcgta accaaagcca\n",
      "       61 gaaggacgtc agaaccggac cagattcttc acatagggac cagtgcatga tttaggagca\n",
      "      121 aaattctcaa atacaagaga aagctcgcac tctcaccatg gtaaaggaaa tgaagtacta\n",
      "      181 cgacatcctt ggtgtcaaac caagtgccac gccagcggaa ctgaagaagg cgtacaggaa\n",
      "      241 acttgcaatg aagtatcatc ctgacaagaa cccagatgca ggggacaagt tcaaagaaat\n",
      "      301 ttccttggca tatgaagtcc tgtcagacga gaagaa\n",
      "//\n",
      "\n",
      "LOCUS       AAGJ01210774             910 bp    DNA     linear   INV 11-MAR-2005\n",
      "DEFINITION  Strongylocentrotus purpuratus isolate A3 Spur_1.1_Reptig6721, whole\n",
      "            genome shotgun sequence.\n",
      "ACCESSION   AAGJ01210774 AAGJ01000000\n",
      "VERSION     AAGJ01210774.1\n",
      "KEYWORDS    WGS.\n",
      "SOURCE      Strongylocentrotus purpuratus\n",
      "  ORGANISM  Strongylocentrotus purpuratus\n",
      "            Eukaryota; Metazoa; Echinodermata; Eleutherozoa; Echinozoa;\n",
      "            Echinoidea; Euechinoidea; Echinacea; Echinoida;\n",
      "            Strongylocentrotidae; Strongylocentrotus.\n",
      "REFERENCE   1  (bases 1 to 910)\n",
      "  AUTHORS   Song,X.-Z., Shen,Y., Liu,Y., Adams,C., Agbai,O. II, Allen,C.,\n",
      "            Alsbrooks,S., Archer,P., Arredondo,H., Bandaranaike,D., Bangura,L.,\n",
      "            Beltran,B., Beraducci,A., Biswalo,K., Blyth,P., Buhay,C., Burch,P.,\n",
      "            Cadoree,I., Canada,A., Cardenas,V., Carter,K., Cavazos,I.,\n",
      "            Chacko,J., Chahrour,M., Chavez,D., Chen,A., Chen,G., Chen,R.,\n",
      "            Cheng,M.-T., Chu,J., Clerc,K., Cockrell,R., Coyle,M., Cree,A.,\n",
      "            Curry,S., Dai,W., Davila,M.L., Davis,C., Davy-Carroll,L., De\n",
      "            Anda,C., Delgado,O., Denson,S., Deramo,C., Ding,Y., Dinh,H.,\n",
      "            Donlin,J., McCauley,S., Dugan-Rocha,S., Dunn,A., Durbin,K.,\n",
      "            Dziuda,D., Egan,A., Escotto,M., Espinosa,V., Fa,M., Fernandez,S.,\n",
      "            Fernando,P., Flagg,N., Forbes,L., Fowler,G.J., Fu,Q., Fuh,E.,\n",
      "            Garcia,A., Garcia,R., Garner,T., Gench,S., Ghose,S., Gill,R.,\n",
      "            Gonzalez,D., Guevara,W., Holder,M., Haaland,W., Haeberlen,K.,\n",
      "            Hall,B., Hamid,H., Hamilton,K., Harbes,B., Havlak,P., Hawes,A.,\n",
      "            Hawkins,E., Hayes,S., Hernandez,J., Hines,S., Hitchens,M.,\n",
      "            Hogues,M., Hollins,B., Howell,L.T., Hume,J., Imo,K., Jackson,L.,\n",
      "            Jacob,L., Jiang,H., Johnson,B., Johnson,R., Kelly,S., Keys,T.,\n",
      "            Khan,Z., King,L., Kovar,C., Kowis,A., Lara,F., Lee,K., Lee,S.,\n",
      "            LeGall,F.I., Lemon,S., Lewis,L., Li,B., Li,Y., Li,Z., Linnell,M.,\n",
      "            Liu,W., Liu,Y.-S., Liyanage,D., London,P., Lopez,J.,\n",
      "            Lorensuhewa,L., Lozado,R., Madu,R., Maheshwari,M., Mahoney,C.,\n",
      "            Malloy,K., Mansouri,D., Martinez,E., Mercadao,C., Metzker,M.,\n",
      "            Minja,E., Morgan,M., Morris,S., Munidasa,M., Nazareth,L., Ngo,D.,\n",
      "            Nguyen,N., Nott,A., Nwaokelemeh,O., Obregon,M., Ochi-Okorie,C.,\n",
      "            Odeh,E., Okwuonu,G., Okwuonu,K., Parker,D., Pasternak,S., Patel,B.,\n",
      "            Patel,V., Paul,H., Perez,A., Perez,L., Pham,T., Primus,E.,\n",
      "            Pu,L.-L., Puazo,M., Qin,X., Quinn,A., Quiroz,J., Rabata,D.,\n",
      "            Rachlin,E., Ren,Y., Rodriguez,F., Rojas,A., Ruiz,S.J., Sana,M.,\n",
      "            Sanders,W., Santibanez,J., Santos,R., Savery,G., Scherer,S.,\n",
      "            Shen,H., Sisson,I., Sorelle,R., Svatek,A., Taylor,E., Taylor,T.,\n",
      "            Thomas,N., Thorn,R., Thornton,R., Trejos,Z., Usmani,K., Vargo,C.,\n",
      "            Verduzco,D., Villasana,D., Waldron,L., Wang,Q., Wang,S., Warren,J.,\n",
      "            Wei,X., Wheeler,D., Williams,G., Williams,R., Wright,R., Wu,J.,\n",
      "            Yakub,S., Yan,K., Yuan,Y., Yu,F., Zhang,J., Zhang,L., Zhou,J.,\n",
      "            Davidson,E., Cameron,R.A., Sodergren,E., Worley,K.C., Muzny,D.,\n",
      "            Weinstock,G. and Gibbs,R.A.\n",
      "  CONSRTM   Human Genome Sequencing Center\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (14-FEB-2005) Human Genome Sequencing Center, Baylor\n",
      "            College of Medicine, One Baylor Plaza, Houston, TX 77030, USA\n",
      "COMMENT     A number of contigs have been removed because they may contain\n",
      "            contamination.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..910\n",
      "                     /organism=\"Strongylocentrotus purpuratus\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /isolate=\"A3\"\n",
      "                     /db_xref=\"taxon:7668\"\n",
      "ORIGIN      \n",
      "        1 tggggttagg ggggttctta gccccaccta tagtacgggg gttaagccta ttccactttc\n",
      "       61 tgttcagact ggtttttacc aaagtgggct aaagttggct aaccccacct atagtacggg\n",
      "      121 gttaactaac cctgcaaaat agcagggcta actccgcaat tgtggtgcta actcctgcta\n",
      "      181 accccgcaat tgcggtgcta agcaaatgaa tgagaacaga aacgaacaat agtccggtac\n",
      "      241 tatgcaatgg agcgtatcgc agctggcaag atgcgaagta tgtccaatca taacattcac\n",
      "      301 gatatccggg gttaacgaaa tgcagtgaga tggaaattgt tctagtataa tgcacctatt\n",
      "      361 gtaatattgt tgtgcattat ccatatattt tgatcatgat ggaaaaataa ataacctaac\n",
      "      421 taaaggccct tctactcagc ccaaacacca ttttgagata acagactgtc gtgcgaacct\n",
      "      481 aactaaaggc ccttctactc agcccaaaca ccattttgag ataacagact gtcgtgcgaa\n",
      "      541 cctaactaaa ggcccttcta ctcagcccaa acaccatttt gagttaacag actgtcgtgc\n",
      "      601 gaacctaact aaaggccctt ctactcagcc caaacaccat tttgagttaa cagactgtcg\n",
      "      661 tgctaaccta actaaaggcc cttctactca gcccaaacac cattttgagt taacagactg\n",
      "      721 tcgtgcgaac ctaactaaag gcccttctac tcagcccaaa caccattttg agttcacaga\n",
      "      781 ctgtcgtgcg aacctaacta aaggcccttc tactcagccc aaacaccatt ttgagttaac\n",
      "      841 agactgtcgt gcgaacctaa ctaaaagccc ttctactcag cccaaacacc attttgagtt\n",
      "      901 cacagactgt\n",
      "//\n",
      "\n",
      "LOCUS       AAGV01843290            1143 bp    DNA     linear   MAM 13-MAY-2005\n",
      "DEFINITION  Dasypus novemcinctus isolate 3-136 cont1.843289, whole genome\n",
      "            shotgun sequence.\n",
      "ACCESSION   AAGV01843290 AAGV01000000\n",
      "VERSION     AAGV01843290.1\n",
      "DBLINK      Project: 12594\n",
      "KEYWORDS    WGS.\n",
      "SOURCE      Dasypus novemcinctus (nine-banded armadillo)\n",
      "  ORGANISM  Dasypus novemcinctus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Xenarthra; Dasypodidae; Dasypus.\n",
      "REFERENCE   1  (bases 1 to 1143)\n",
      "  AUTHORS   Lindblad-Toh,K., Chang,J.L., Gnerre,S., Clamp,M. and Lander,E.S.\n",
      "  CONSRTM   The Genome Sequencing Platform, The Genome Assembly Team\n",
      "  TITLE     Annotating the human genome using low coverage mammalian genomes\n",
      "  JOURNAL   Unpublished\n",
      "REFERENCE   2  (bases 1 to 1143)\n",
      "  AUTHORS   Lindblad-Toh,K., Chang,J.L., Gnerre,S., Clamp,M. and Lander,E.S.\n",
      "  CONSRTM   The Genome Sequencing Platform, The Genome Assembly Team\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (05-MAY-2005) Broad Institute of MIT and Harvard, 320\n",
      "            Charles Street, Cambridge, MA 02141, USA\n",
      "COMMENT     This version of the project represents approximately 2x coverage of\n",
      "            the genome.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..1143\n",
      "                     /organism=\"Dasypus novemcinctus\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /isolate=\"3-136\"\n",
      "                     /db_xref=\"taxon:9361\"\n",
      "                     /country=\"USA:National Hansen's Disease Programs at\n",
      "                     Louisiana State University, School of Veterinary Medicine\"\n",
      "ORIGIN      \n",
      "        1 ggggtaagtc caataaaccc agagtgtagg agttgcaagt ctgttgaggc tcagggcctg\n",
      "       61 gctatcacat ggacaatcca gagattcagg tcccctgagt atacactaaa ccccactgcc\n",
      "      121 aaccataggt ccggtaaaag tgacagtaga ggcttgtgaa caaagatcac acctgagtcc\n",
      "      181 aactccatca cactcaggaa cacaaactcc aaaatagggc caactgacat ggcactgaac\n",
      "      241 tccatctgct atgaccgtag aacctgggag tctctgtagc cctcagaaga accaatacct\n",
      "      301 ggggatgtat ctactttatc tgtctctggg actctgctga ggtatgcata agggcgaccc\n",
      "      361 ctctgataat ctcctggctc ttttttggag actcatagcc atataaactc atttgtcatt\n",
      "      421 tccatttccc ccttttattc aaggtcaaaa agcattttaa ctcctgatat tacatgtagg\n",
      "      481 ctgagatatt ctgttggtct agttgaccct tttattcaag gtcattttct agctacatca\n",
      "      541 tcaactggta cttggtcgta atccctcaac gccagggagg ctcagcccta ggagtcatgt\n",
      "      601 cccttgctgg ggggaaggca gtgcatttac atgctgagtt cggcttcgag actggccaca\n",
      "      661 tttgagcaac atggaggatc tcaggaggta actcttaggc accctgcagc tctaggccta\n",
      "      721 gttcttattt caggcgcaca ggctcacagg catagtcatt agtatcaagg gctcattgtt\n",
      "      781 ggaccatcct tttttggtct tagccattgc acttgggaga ttgttgctgt tccattaggg\n",
      "      841 aatgtgatag acctcccctg gctaggaact cagcactccc tcagttgtcg tttttaactg\n",
      "      901 taaccactat gaaaatatcc aaacattttt gtataccctg gagaactccc tgccaaccat\n",
      "      961 gtgtctaaca tcccacacca gtattcctcg cctgccattg ctgaacctct ctgtaatcca\n",
      "     1021 aaacttcttc gaaaatgaag cctaatatat tgttaggttc cattaatagt aaaatggaat\n",
      "     1081 atagcgatga gtttaaaggt tagatataga atatatacta ctttagaaaa attaaggtaa\n",
      "     1141 aaa\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# Making a GET request\n",
    "#r = requests.get('https://www.ncbi.nlm.nih.gov/nuccore/{}'.format(genes))\n",
    "# Parsing the HTML\n",
    "locus = \"\"\n",
    "locus_list = []\n",
    "for c in id_content:\n",
    "\tsoup = BeautifulSoup(c, 'html.parser')\n",
    "\n",
    "# Procurar um tag meta com um determinado atributo\n",
    "\n",
    "\tlines = soup.find_all('meta', {'name':\"ncbi_uidlist\"} )\n",
    "\n",
    "\tid = \"\"\n",
    "\turl = \"\"\n",
    "\tfor line in lines:\n",
    "\t\t# print(line)\n",
    "\t\t# if 'name' in line.attrs:\n",
    "\t\t# \tprint(line.attrs['name'])\n",
    "\t\tif 'content' in line.attrs:\n",
    "\t\t\t# print(line.attrs['content'])\t\t\n",
    "\t\t\tid = line.attrs['content']\n",
    "\n",
    "\tif id:\n",
    "\t\turl =\"https://www.ncbi.nlm.nih.gov/sviewer/viewer.fcgi?id={}&db=nuccore&report=genbank&conwithfeat=on&hide-cdd=on&retmode=text&ncbi_phid=CE88F25338A215A1000000000483042A&withmarkup=on&tool=portal&log$=seqview&maxdownloadsize=1000000\".format(id)\n",
    "\n",
    "\tr2 = requests.get(url)\n",
    "\tlocus += r2.content.decode('utf-8')\n",
    "\n",
    "\tif \"Error\" not in r2.content.decode('utf-8'):\n",
    "\t\tlocus_list.append(r2.content.decode('utf-8'))\n",
    "print(locus)\n",
    "\t#utf-8 para converter de bytes para string\n",
    "\t#https://stackoverflow.com/questions/31019854/typeerror-cant-use-a-string-pattern-on-a-bytes-like-object-in-re-findall"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "LINK NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 664,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['https://www.ncbi.nlm.nih.gov/nuccore/X67302',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/BW778499',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/BW778502',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/BB232249',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/BB232248',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/AAHY01223610',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/BW778501',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/AAGJ01210774',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/AAGV01843290']"
      ]
     },
     "execution_count": 664,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "id_lista = []\n",
    "# print(genes)\n",
    "existe = re.findall(r'ACCESSION\\s+[^\\s]+', locus)\n",
    "#\\s+ um ou mais espaços\n",
    "#[^\\s]+ indica um conjunto de caracteres que se podem repetir uma ou mais vezes\n",
    "\n",
    "if existe:\n",
    "    for c in existe:\n",
    "        m = re.match(r'ACCESSION\\s+([^\\s]+)', c )\n",
    "        if m:\n",
    "            id = m.group(1)\n",
    "            id_lista.append(\"https://www.ncbi.nlm.nih.gov/nuccore/{}\".format( id ) )\n",
    "else:\n",
    "    print( \"Padrão não encontrado\" )\n",
    "id_lista\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ACCESSION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 665,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X67302 ',\n",
       " 'BW778499 ',\n",
       " 'BW778502 ',\n",
       " 'BB232249 ',\n",
       " 'BB232248 ',\n",
       " 'AAHY01223610 AAHY01000000 ',\n",
       " 'BW778501 ',\n",
       " 'AAGJ01210774 AAGJ01000000 ',\n",
       " 'AAGV01843290 AAGV01000000 ']"
      ]
     },
     "execution_count": 665,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_list = []\n",
    "\n",
    "existe = re.findall(r'ACCESSION\\s+.*?(?=VERSION)', locus, re.DOTALL)\n",
    "#\\s+ um ou mais espaços\n",
    "#\\.* encontra o caractere ponto final zero ou mais vezes\n",
    "#?= something que faz match com VERSION mas não consome estes carateres\n",
    "if existe:\n",
    "    for accession in existe:\n",
    "        m = re.match( r'ACCESSION\\s+(.+)', accession, re.DOTALL )\n",
    "        acc_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "#(.+) agrupar um ou mais caracteres dentro de um grupo\n",
    "#re.sub(r'\\s+', ' ' ...) substitui um ou mais espaços por apenas um espaço\n",
    "acc_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "VERSION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 666,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X67302.1 ',\n",
       " 'BW778499.1 ',\n",
       " 'BW778502.1 ',\n",
       " 'BB232249.1 ',\n",
       " 'BB232248.1 ',\n",
       " 'AAHY01223610.1 ',\n",
       " 'BW778501.1 ',\n",
       " 'AAGJ01210774.1 ',\n",
       " 'AAGV01843290.1 ']"
      ]
     },
     "execution_count": 666,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "version_list = []\n",
    "\n",
    "existe = re.findall(r'VERSION.*?(?=KEYWORDS|DBLINK)', locus, re.DOTALL)\n",
    "#\\s+ um ou mais espaços\n",
    "#\\.* encontra o caractere ponto final zero ou mais vezes\n",
    "#?= something que faz match com VERSION mas não consome estes carateres\n",
    "if existe:\n",
    "    for vers in existe:\n",
    "        m = re.match( r'VERSION\\s+(.+)', vers, re.DOTALL )\n",
    "        version_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "#(.+) agrupar um ou mais caracteres dentro de um grupo\n",
    "#re.sub(r'\\s+', ' ' ...) substitui um ou mais espaços por apenas um espaço\n",
    "version_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "DEFINITION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 667,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['R.norvegicus mRNA for Oct 2 Pou-homeodomain protein. ',\n",
       " \"BW778499 Amphioxus Branchiostoma floridae unpublished cDNA library, gastrula whole animal Branchiostoma floridae cDNA clone bfga051e13 5', mRNA sequence. \",\n",
       " \"BW778502 Amphioxus Branchiostoma floridae unpublished cDNA library, gastrula whole animal Branchiostoma floridae cDNA clone bfga051e16 5', mRNA sequence. \",\n",
       " \"BB232249 RIKEN full-length enriched, 3 days neonate thymus Mus musculus cDNA clone A630036M17 3' similar to AF236887 Mus musculus protein kinase ATR (Atr) mRNA, mRNA sequence. \",\n",
       " \"BB232248 RIKEN full-length enriched, 3 days neonate thymus Mus musculus cDNA clone A630036M16 3', mRNA sequence. \",\n",
       " 'Mus musculus strain 129X1/SvJ chromosome Un CRA_211000020833787, whole genome shotgun sequence. ',\n",
       " \"BW778501 Amphioxus Branchiostoma floridae unpublished cDNA library, gastrula whole animal Branchiostoma floridae cDNA clone bfga051e15 5', mRNA sequence. \",\n",
       " 'Strongylocentrotus purpuratus isolate A3 Spur_1.1_Reptig6721, whole genome shotgun sequence. ',\n",
       " 'Dasypus novemcinctus isolate 3-136 cont1.843289, whole genome shotgun sequence. ']"
      ]
     },
     "execution_count": 667,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "defi_list = []\n",
    "existe = re.findall(r'DEFINITION\\s+.*?(?=ACCESSION)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for definition in existe:\n",
    "        m = re.match( r'DEFINITION\\s+(.+)', definition, re.DOTALL )\n",
    "        defi_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "defi_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "LOCUS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 668,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X67302 369 bp mRNA linear ROD 26-JUL-2016 ',\n",
       " 'BW778499 339 bp mRNA linear EST 04-JAN-2011 ',\n",
       " 'BW778502 451 bp mRNA linear EST 04-JAN-2011 ',\n",
       " 'BB232249 291 bp mRNA linear EST 04-JUL-2000 ',\n",
       " 'BB232248 297 bp mRNA linear EST 04-JUL-2000 ',\n",
       " 'AAHY01223610 742 bp DNA linear ROD 07-JUL-2005 ',\n",
       " 'BW778501 336 bp mRNA linear EST 04-JAN-2011 ',\n",
       " 'AAGJ01210774 910 bp DNA linear INV 11-MAR-2005 ',\n",
       " 'AAGV01843290 1143 bp DNA linear MAM 13-MAY-2005 ']"
      ]
     },
     "execution_count": 668,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loc_list = []\n",
    "existe = re.findall(r'LOCUS\\s+.*?(?=DEFINITION)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for loc in existe:\n",
    "        m = re.match( r'LOCUS\\s+(.+)', loc, re.DOTALL )\n",
    "        loc_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "loc_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "LOCUS NAME"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 669,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X67302',\n",
       " 'BW778499',\n",
       " 'BW778502',\n",
       " 'BB232249',\n",
       " 'BB232248',\n",
       " 'AAHY01223610',\n",
       " 'BW778501',\n",
       " 'AAGJ01210774',\n",
       " 'AAGV01843290']"
      ]
     },
     "execution_count": 669,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loc_name_list = []\n",
    "existe = re.findall(r'LOCUS\\s + [^\\s]+', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for id in existe:\n",
    "        m = re.match( r'LOCUS\\s+(.+)', id, re.DOTALL )\n",
    "        loc_name_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "loc_name_list\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "NUMBER OF BP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 670,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['369 bp',\n",
       " '339 bp',\n",
       " '451 bp',\n",
       " '291 bp',\n",
       " '297 bp',\n",
       " '742 bp',\n",
       " '336 bp',\n",
       " '910 bp',\n",
       " '1143 bp']"
      ]
     },
     "execution_count": 670,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loc_bp_list = []\n",
    "loc = []\n",
    "\n",
    "existe = re.findall(r'LOCUS\\s+.*?(?=DEFINITION)', locus, re.DOTALL)\n",
    "# print(existe)\n",
    "if existe:\n",
    "    for c in existe:\n",
    "        loc.append(list(filter(None, c.split(' '))))\n",
    "for x in range(len(loc)):\n",
    "    loc_bp_list.append(' '.join(loc[x][2:4])) \n",
    "\n",
    "#cria lista com os valores do locus, e escolhe os 3º e 4º caracteres, que correspondem ao nr de bp\n",
    "loc_bp_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MOLECULE TYPE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 671,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['mRNA linear',\n",
       " 'mRNA linear',\n",
       " 'mRNA linear',\n",
       " 'mRNA linear',\n",
       " 'mRNA linear',\n",
       " 'DNA linear',\n",
       " 'mRNA linear',\n",
       " 'DNA linear',\n",
       " 'DNA linear']"
      ]
     },
     "execution_count": 671,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mol_type = []\n",
    "for m in range(len(loc)):\n",
    "    if \"linear\" in loc[m][5]:\n",
    "        mol_type.append(' '.join(loc[m][4:6]))\n",
    "    else:\n",
    "        mol_type.append(' '.join(loc[m][4:5]))\n",
    "#temos que corrigir esta porque as mol types podem ser de uma, duas ou trÊs palavras..\n",
    "mol_type"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GENBANK DIVISION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 672,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ROD', 'EST', 'EST', 'EST', 'EST', 'ROD', 'EST', 'INV', 'MAM']"
      ]
     },
     "execution_count": 672,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gb_div = []\n",
    "for gb in range(len(loc)):\n",
    "    gb_div.append(' '.join(loc[gb][6:7]))\n",
    "gb_div"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MODIFICATION DATE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 673,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['26-JUL-2016',\n",
       " '04-JAN-2011',\n",
       " '04-JAN-2011',\n",
       " '04-JUL-2000',\n",
       " '04-JUL-2000',\n",
       " '07-JUL-2005',\n",
       " '04-JAN-2011',\n",
       " '11-MAR-2005',\n",
       " '13-MAY-2005']"
      ]
     },
     "execution_count": 673,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mod_date = []\n",
    "for md in range(len(loc)):\n",
    "    mod_date.append(''.join(loc[md][7]).replace(\"\\n\",\"\"))\n",
    "mod_date"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SOURCE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 674,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Rattus norvegicus (Norway rat) ',\n",
       " 'Branchiostoma floridae (Florida lancelet) ',\n",
       " 'Branchiostoma floridae (Florida lancelet) ',\n",
       " 'Mus musculus (house mouse) ',\n",
       " 'Mus musculus (house mouse) ',\n",
       " 'Mus musculus (house mouse) ',\n",
       " 'Branchiostoma floridae (Florida lancelet) ',\n",
       " 'Strongylocentrotus purpuratus ',\n",
       " 'Dasypus novemcinctus (nine-banded armadillo) ']"
      ]
     },
     "execution_count": 674,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "source_list = []\n",
    "existe = re.findall(r'SOURCE\\s+.*?(?=ORGANISM)', locus, re.DOTALL)\n",
    "# print(existe)\n",
    "if existe:\n",
    "    for source in existe:\n",
    "        m = re.match( r'SOURCE\\s+(.+)', source, re.DOTALL )\n",
    "        source_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "source_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ORGANISM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 675,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Rattus norvegicus',\n",
       " 'Branchiostoma floridae',\n",
       " 'Branchiostoma floridae',\n",
       " 'Mus musculus',\n",
       " 'Mus musculus',\n",
       " 'Mus musculus',\n",
       " 'Branchiostoma floridae',\n",
       " 'Strongylocentrotus purpuratus',\n",
       " 'Dasypus novemcinctus']"
      ]
     },
     "execution_count": 675,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_list = []\n",
    "existe = re.findall(r'ORGANISM\\s+.*?(?=\\n)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for org in existe:\n",
    "        m = re.match( r'ORGANISM\\s+(.+)', org, re.DOTALL )\n",
    "        org_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "org_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "FEATURES - CDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 676,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 677,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['source', 'gene', 'CDS', 'misc_feature', 'misc_feature', 'misc_feature', 'source', 'source', 'source', 'source', 'source', 'source', 'source', 'source']\n",
      "* [['[0:369](+)'], ['[0:339](+)'], ['[0:451](+)'], ['[0:291](+)'], ['[0:297](+)'], ['[0:742](+)'], ['[0:336](+)'], ['[0:910](+)'], ['[0:1143](+)']]\n",
      "-> [[['Oct 2']], [], [], [], [], [], [], [], []]\n",
      "9\n",
      "p [[[], [], ['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']], [], [], []], [[]], [[]], [[]], [[]], [[]], [[]], [[]], [[]]]\n",
      "-> [[[], [], ['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']], [], [], []], [[]], [[]], [[]], [[]], [[]], [[]], [[]], [[]]]\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "from Bio import SeqFeature\n",
    "from Bio.SeqFeature import SeqFeature, FeatureLocation\n",
    "\n",
    "Entrez.email = \"pg49849@alunos.uminho.pt\"\n",
    "\n",
    "features=[]\n",
    "CDS=[] \n",
    "source = []\n",
    "gene = []\n",
    "\n",
    "\n",
    "for acc in acc_list:\n",
    "    handle = Entrez.efetch(db=\"nucleotide\", rettype = \"gb\", retmode= \"text\", id=acc)\n",
    "    \n",
    "    for record in SeqIO.parse(handle,\"gb\"):\n",
    "        \n",
    "        for feature in record.features:\n",
    "            features.append(feature.type)\n",
    "\n",
    "        aux_source = []\n",
    "        aux_gene = []\n",
    "\n",
    "\n",
    "        for feature in record.features:\n",
    "            if feature.type == \"source\":\n",
    "                aux_source.append(str(feature.location))\n",
    "\n",
    "            if feature.type == 'gene':\n",
    "                aux_gene.append(feature.qualifiers[\"gene\"])\n",
    " \n",
    "        gene.append(aux_gene)\n",
    "        source.append(aux_source)\n",
    "\n",
    "        record_CDS = []\n",
    "        for feature in record.features:\n",
    "            CDS_aux=[]\n",
    "            if feature.type == \"CDS\":\n",
    "                CDS_aux.append(str(feature.location))\n",
    "                CDS_aux.append(feature.qualifiers[\"product\"])\n",
    "                CDS_aux.append(feature.qualifiers[\"protein_id\"])\n",
    "                CDS_aux.append(feature.qualifiers[\"translation\"])\n",
    "                # print('->' , CDS_aux)\n",
    "            record_CDS.append(CDS_aux)\n",
    "            \n",
    "        CDS.append(record_CDS)\n",
    "\n",
    "# print(source)\n",
    "# print(gene)\n",
    "# print(exon)\n",
    "# print(intron)\n",
    "print(features)\n",
    "print('*',source)\n",
    "print('->',gene)\n",
    "print(len(CDS))\n",
    "print('p', CDS)  \n",
    "print('->', CDS) "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SEQUENCE/ORIGIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 678,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ttcacacagggtgatgtgggcctggccatgggcaagctctatggcaacgacttcagccaaacgaccatttcccgcttcgaggcccttaacctgagcttcaagaacatgtgtaaactcaagcccctcctggagaagtggctcaacgacgcagagactatgtctgtggattcaagcctacccagcccaaaccagctgagcagccccagcctgggtttcgacgggctgccggggcggagacgcaagaagaggaccagcatcgagacgaatgtccgcttcgccttagagaagagtttcctagcgaaccagaagcctacctcagaggagatcctgctgatcgcagagcagctgcacatggaaaaggaagtgatc',\n",
       " 'atctataacacctaccaatggcggacgcacctccagccggtaggggaggttttcgcggcgggttcggggatcgcggcaggggccgtggacggggccgaggtcgcggcaggggccgtggacggggccgcggaggcaaggaggaaaaggagtggatgccaatgaccaagcttggtcgtcttgtgaaagacctgaagatcaagtccctggaggatatctacctgttctcacttcccatcaaggagtttgagatcattgactttttcctgggttcggcactgaaggatgaggtgctgaagatcatgcctgttcagaagcagacccgtgccggtcagcgcacca',\n",
       " 'tctaagaacattgtaaggaatcacagcttgtatctccgaatctaagacgtcttcgttatacagagtctcgggaccacaactggtaaacaaagatgtcatcagaagaagaagactacctgtcccccaggtccacaagtatcagtaacggggagagcagtttctacaccagcgctcaaacgcacctgtctgactcacaggacgacgcagacctgaagttctcaggccggtccaagctgagtgaagacagtattgatgagtttatgagcagctgtggaacctacgtctcctataatgcacggtattacaatggacaagtcatcaccaccttgtcgtcaaagaacacgctgttccgggcagagtttgacctgcaacactgtatacaggagctgcatgatgcaagaatacagattcccagtgacgacgtctctgcggccatcgccagcatcctggg',\n",
       " 'acttctcctgctaccatactctcttctcttcccctgatctaggatcagcgcatgttctcaggccccactcatgaaagacaggaggatgcctcggcacgagtaattctcggaatactttgagtttttcagattaagctgctgactttggtagcatacactggcgaagctcctttcctttcatatacactcctacaaagcagaatgattttaattgtttgtacagccaaatgagaaaagaaccaagaataattttgttaaaatattaaagtttcaacttttgctacaactgaa',\n",
       " 'atcctgggcaacattaaaaatagtctacattaggtttttttccagaaaccttttgtaacaaacagtatctcgttatgtaactccagcaccaagctttgcttgtatttagtcaactcagcttggaaatccgcccttgggccccactccattctcagtccagcacatctagtagcagtgtggcggaaacacttttatggcaacatggagcagtattagattttatgtagctgacaatgcagtattaaagctcagtattttgcatatttttagcatttacaaatatttttgctttagtgc',\n",
       " 'tggaaaataagaaaaatccacttgacgacttgaaaaatgacgaaatcactaatagacgtgaaaaatgtgaaatgcacactgaaggacctggaatatggcgagaaaactgaaaatcacggaaaatgagaaatacacactttaggacgtgaaatatggcgaggaaaactgaaaaaggtggaaaatttagaaatgtcctctgtaggacgtgaaatatggcaagaaaactgaaaatcacggataatgagaaacatacacttgacgacttgaaaaatggcgaaatcactaaaaaacgtgaaaaatgagaaatgcacactgaaggacctggaatatggcgagaaaactgaaaatcaaggaaaatgagaaatacacactttaggacgtgaaatatggcgaggaaaactgaaaaaggtggaaaatttagaaatgtccactgtaggacgtggaatatggcaagaaaactgaaaatcataaaaaatgagaaacatccacttgacgacttgaaaaatgacgaaatcactaaaaaacgtgaaaaatgtgaaatgcacactgaaggacctggaatatggcgagaaaactgaaaatcacggaaaatgagtaatacacactttaggacgtgaaatatggcgaggaaaactgaaaaaggtggaaaatttagaaatgtccactgtacgacgtggaatatggcaagaacactgaaaatcatgggaaaatgagaaacatccacttgacgacttgaaaaatg',\n",
       " 'aagatcgtgtgttctcgaaactgaaggttataaaaaccattttcagcgtaaccaaagccagaaggacgtcagaaccggaccagattcttcacatagggaccagtgcatgatttaggagcaaaattctcaaatacaagagaaagctcgcactctcaccatggtaaaggaaatgaagtactacgacatccttggtgtcaaaccaagtgccacgccagcggaactgaagaaggcgtacaggaaacttgcaatgaagtatcatcctgacaagaacccagatgcaggggacaagttcaaagaaatttccttggcatatgaagtcctgtcagacgagaagaa',\n",
       " 'tggggttaggggggttcttagccccacctatagtacgggggttaagcctattccactttctgttcagactggtttttaccaaagtgggctaaagttggctaaccccacctatagtacggggttaactaaccctgcaaaatagcagggctaactccgcaattgtggtgctaactcctgctaaccccgcaattgcggtgctaagcaaatgaatgagaacagaaacgaacaatagtccggtactatgcaatggagcgtatcgcagctggcaagatgcgaagtatgtccaatcataacattcacgatatccggggttaacgaaatgcagtgagatggaaattgttctagtataatgcacctattgtaatattgttgtgcattatccatatattttgatcatgatggaaaaataaataacctaactaaaggcccttctactcagcccaaacaccattttgagataacagactgtcgtgcgaacctaactaaaggcccttctactcagcccaaacaccattttgagataacagactgtcgtgcgaacctaactaaaggcccttctactcagcccaaacaccattttgagttaacagactgtcgtgcgaacctaactaaaggcccttctactcagcccaaacaccattttgagttaacagactgtcgtgctaacctaactaaaggcccttctactcagcccaaacaccattttgagttaacagactgtcgtgcgaacctaactaaaggcccttctactcagcccaaacaccattttgagttcacagactgtcgtgcgaacctaactaaaggcccttctactcagcccaaacaccattttgagttaacagactgtcgtgcgaacctaactaaaagcccttctactcagcccaaacaccattttgagttcacagactgt',\n",
       " 'ggggtaagtccaataaacccagagtgtaggagttgcaagtctgttgaggctcagggcctggctatcacatggacaatccagagattcaggtcccctgagtatacactaaaccccactgccaaccataggtccggtaaaagtgacagtagaggcttgtgaacaaagatcacacctgagtccaactccatcacactcaggaacacaaactccaaaatagggccaactgacatggcactgaactccatctgctatgaccgtagaacctgggagtctctgtagccctcagaagaaccaatacctggggatgtatctactttatctgtctctgggactctgctgaggtatgcataagggcgacccctctgataatctcctggctcttttttggagactcatagccatataaactcatttgtcatttccatttcccccttttattcaaggtcaaaaagcattttaactcctgatattacatgtaggctgagatattctgttggtctagttgacccttttattcaaggtcattttctagctacatcatcaactggtacttggtcgtaatccctcaacgccagggaggctcagccctaggagtcatgtcccttgctggggggaaggcagtgcatttacatgctgagttcggcttcgagactggccacatttgagcaacatggaggatctcaggaggtaactcttaggcaccctgcagctctaggcctagttcttatttcaggcgcacaggctcacaggcatagtcattagtatcaagggctcattgttggaccatccttttttggtcttagccattgcacttgggagattgttgctgttccattagggaatgtgatagacctcccctggctaggaactcagcactccctcagttgtcgtttttaactgtaaccactatgaaaatatccaaacatttttgtataccctggagaactccctgccaaccatgtgtctaacatcccacaccagtattcctcgcctgccattgctgaacctctctgtaatccaaaacttcttcgaaaatgaagcctaatatattgttaggttccattaatagtaaaatggaatatagcgatgagtttaaaggttagatatagaatatatactactttagaaaaattaaggtaaaaa']"
      ]
     },
     "execution_count": 678,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_origin = []\n",
    "existe = re.findall(r'ORIGIN\\s+.*?(?=//)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for nuc in existe:\n",
    "        m = re.match( r'ORIGIN\\s+(.+)', nuc, re.DOTALL)\n",
    "        a = re.sub(r'\\s+', '', m.group(1) ) \n",
    "        num = re.sub(r'\\d+','', a)\n",
    "        seq_origin.append(num)\n",
    "seq_origin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 679,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[97, 105, 65, 102],\n",
       " [73, 83, 66, 117],\n",
       " [131, 116, 98, 106],\n",
       " [85, 67, 90, 49],\n",
       " [83, 62, 100, 52],\n",
       " [336, 102, 136, 168],\n",
       " [121, 72, 66, 77],\n",
       " [275, 234, 230, 171],\n",
       " [312, 272, 321, 238]]"
      ]
     },
     "execution_count": 679,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_count = []\n",
    "for seq in seq_origin:\n",
    "    #contagem de nucleotidos\n",
    "    A= []\n",
    "    C= []\n",
    "    T= []\n",
    "    G= []\n",
    "    for n in seq:\n",
    "        if n == 'a':\n",
    "            A.append('a')\n",
    "        if n == 'c':\n",
    "            C.append('c')\n",
    "        if n == 't':\n",
    "            T.append('t')\n",
    "        if n == 'g':\n",
    "            G.append('g')\n",
    "        \n",
    "    a = len(A)\n",
    "    c = len(C)\n",
    "    t = len(T)\n",
    "    g = len(G)\n",
    "    seq_count.append([a,c,t,g])\n",
    "    #fazer append de um array\n",
    "seq_count\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 680,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[369, 339, 451, 291, 297, 742, 336, 910, 1143]"
      ]
     },
     "execution_count": 680,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "length = []\n",
    "for seq in seq_origin:\n",
    "    length.append(len(seq))\n",
    "length"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "JOURNAL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 681,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Unpublished ',\n",
       " 'Submitted (07-JUL-1992) R.J. Rodgers, Flinders Univ. of South Australia, Dept. of Medicine, Bedford Park, SA 5042, AUSTRALIA ',\n",
       " 'Nature 445 (7128), 613-617 (2007) ',\n",
       " 'Nature 445 (7128), 613-617 (2007) ',\n",
       " 'Unpublished (2000) ',\n",
       " 'Unpublished (2000) ',\n",
       " 'Science 296 (5573), 1661-1671 (2002) ',\n",
       " 'Submitted (02-JUN-2005) Celera Genomics, 45 W. Gude Dr., Rockville, MD 20850, USA ',\n",
       " 'Nature 445 (7128), 613-617 (2007) ',\n",
       " 'Submitted (14-FEB-2005) Human Genome Sequencing Center, Baylor College of Medicine, One Baylor Plaza, Houston, TX 77030, USA ',\n",
       " 'Unpublished ',\n",
       " 'Submitted (05-MAY-2005) Broad Institute of MIT and Harvard, 320 Charles Street, Cambridge, MA 02141, USA ']"
      ]
     },
     "execution_count": 681,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "journ_list = []\n",
    "existe = re.findall(r'JOURNAL\\s+.*?(?=PUBMED|REFERENCE|FEATURES|COMMENT)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for journal in existe:\n",
    "        m = re.match( r'JOURNAL\\s+(.+)', journal, re.DOTALL )\n",
    "        journ_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "journ_list\n",
    "#| dá-nos operador ou\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "AUTHORS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 475,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['McFadden,G.I., Gilson,P.R., Hofmann,C.J., Adcock,G.J. and Maier,U.G. ',\n",
       " 'McFadden,G.I. ',\n",
       " 'Lang,G., Wotton,D., Owen,M.J., Sewell,W.A., Brown,M.H., Mason,D.Y., Crumpton,M.J. and Kioussis,D. ',\n",
       " 'Benito,J.M., Lovrich,G.A., Abate,C.M. and Sineriz,F. ',\n",
       " 'Benito,J.M., Lovrich,G.A., Abate,C.M. and Sineriz,F. ',\n",
       " 'Satou,Y., Shin-i,T., Kohara,Y. and Satoh,N. ',\n",
       " 'Buell,C.R., Hart,A., Zismann,V., Karamycheva,S.A., Day,B., Staskawicz,B., Jin,H. and Baker,B. ',\n",
       " 'Zhao,S., Adams,M.D., Nierman,W., Malek,J., de Jong,P. and Venter,J.C. ',\n",
       " 'Venter,J.C., Adams,M.D., Myers,E.W., Li,P.W., Mural,R.J., Sutton,G.G., Smith,H.O., Yandell,M., Evans,C.A., Holt,R.A., Gocayne,J.D., Amanatides,P., Ballew,R.M., Huson,D.H., Wortman,J.R., Zhang,Q., Kodira,C.D., Zheng,X.H., Chen,L., Skupski,M., Subramanian,G., Thomas,P.D., Zhang,J., Gabor Miklos,G.L., Nelson,C., Broder,S., Clark,A.G., Nadeau,J., McKusick,V.A., Zinder,N., Levine,A.J., Roberts,R.J., Simon,M., Slayman,C., Hunkapiller,M., Bolanos,R., Delcher,A., Dew,I., Fasulo,D., Flanigan,M., Florea,L., Halpern,A., Hannenhalli,S., Kravitz,S., Levy,S., Mobarry,C., Reinert,K., Remington,K., Abu-Threideh,J., Beasley,E., Biddick,K., Bonazzi,V., Brandon,R., Cargill,M., Chandramouliswaran,I., Charlab,R., Chaturvedi,K., Deng,Z., Di Francesco,V., Dunn,P., Eilbeck,K., Evangelista,C., Gabrielian,A.E., Gan,W., Ge,W., Gong,F., Gu,Z., Guan,P., Heiman,T.J., Higgins,M.E., Ji,R.R., Ke,Z., Ketchum,K.A., Lai,Z., Lei,Y., Li,Z., Li,J., Liang,Y., Lin,X., Lu,F., Merkulov,G.V., Milshina,N., Moore,H.M., Naik,A.K., Narayan,V.A., Neelam,B., Nusskern,D., Rusch,D.B., Salzberg,S., Shao,W., Shue,B., Sun,J., Wang,Z., Wang,A., Wang,X., Wang,J., Wei,M., Wides,R., Xiao,C., Yan,C., Yao,A., Ye,J., Zhan,M., Zhang,W., Zhang,H., Zhao,Q., Zheng,L., Zhong,F., Zhong,W., Zhu,S., Zhao,S., Gilbert,D., Baumhueter,S., Spier,G., Carter,C., Cravchik,A., Woodage,T., Ali,F., An,H., Awe,A., Baldwin,D., Baden,H., Barnstead,M., Barrow,I., Beeson,K., Busam,D., Carver,A., Center,A., Cheng,M.L., Curry,L., Danaher,S., Davenport,L., Desilets,R., Dietz,S., Dodson,K., Doup,L., Ferriera,S., Garg,N., Gluecksmann,A., Hart,B., Haynes,J., Haynes,C., Heiner,C., Hladun,S., Hostin,D., Houck,J., Howland,T., Ibegwam,C., Johnson,J., Kalush,F., Kline,L., Koduru,S., Love,A., Mann,F., May,D., McCawley,S., McIntosh,T., McMullen,I., Moy,M., Moy,L., Murphy,B., Nelson,K., Pfannkoch,C., Pratts,E., Puri,V., Qureshi,H., Reardon,M., Rodriguez,R., Rogers,Y.H., Romblad,D., Ruhfel,B., Scott,R., Sitter,C., Smallwood,M., Stewart,E., Strong,R., Suh,E., Thomas,R., Tint,N.N., Tse,S., Vech,C., Wang,G., Wetter,J., Williams,S., Williams,M., Windsor,S., Winn-Deen,E., Wolfe,K., Zaveri,J., Zaveri,K., Abril,J.F., Guigo,R., Campbell,M.J., Sjolander,K.V., Karlak,B., Kejariwal,A., Mi,H., Lazareva,B., Hatton,T., Narechania,A., Diemer,K., Muruganujan,A., Guo,N., Sato,S., Bafna,V., Istrail,S., Lippert,R., Schwartz,R., Walenz,B., Yooseph,S., Allen,D., Basu,A., Baxendale,J., Blick,L., Caminha,M., Carnes-Stine,J., Caulk,P., Chiang,Y.H., Coyne,M., Dahlke,C., Mays,A., Dombroski,M., Donnelly,M., Ely,D., Esparham,S., Fosler,C., Gire,H., Glanowski,S., Glasser,K., Glodek,A., Gorokhov,M., Graham,K., Gropman,B., Harris,M., Heil,J., Henderson,S., Hoover,J., Jennings,D., Jordan,C., Jordan,J., Kasha,J., Kagan,L., Kraft,C., Levitsky,A., Lewis,M., Liu,X., Lopez,J., Ma,D., Majoros,W., McDaniel,J., Murphy,S., Newman,M., Nguyen,T., Nguyen,N., Nodell,M., Pan,S., Peck,J., Peterson,M., Rowe,W., Sanders,R., Scott,J., Simpson,M., Smith,T., Sprague,A., Stockwell,T., Turner,R., Venter,E., Wang,M., Wen,M., Wu,D., Wu,M., Xia,A., Zandieh,A. and Zhu,X. ',\n",
       " 'Istrail,S., Sutton,G., Florea,L., Halpern,A.L., Mobarry,C.M., Lippert,R., Walenz,B., Shatkay,H., Dew,I., Miller,J.R., Flanigan,M.J., Edwards,N.J., Bolanos,R., Fasulo,D., Halldorsson,B.V., Hannenhalli,S., Turner,R., Yooseph,S., Lu,F., Nusskern,D.R., Shue,B.C., Zheng,X.H., Zhong,F., Delcher,A.L., Huson,D.H., Kravitz,S.A., Mouchard,L., Reinert,K., Remington,K.A., Clark,A.G., Waterman,M.S., Eichler,E.E., Adams,M.D., Hunkapiller,M.W., Myers,E.W. and Venter,J.C. ',\n",
       " 'Venter,J.C., Adams,M.D., Myers,E.W., Li,P.W., Mural,R.J., Sutton,G.G., Smith,H.O., Yandell,M., Evans,C.A., Holt,R.A., Gocayne,J.D., Amanatides,P., Ballew,R.M., Huson,D.H., Wortman,J.R., Zhang,Q., Kodira,C.D., Zheng,X.H., Chen,L., Skupski,M., Subramanian,G., Thomas,P.D., Zhang,J., Gabor Miklos,G.L., Nelson,C., Broder,S., Clark,A.G., Nadeau,J., McKusick,V.A., Zinder,N., Levine,A.J., Roberts,R.J., Simon,M., Slayman,C., Hunkapiller,M., Bolanos,R., Delcher,A., Dew,I., Fasulo,D., Flanigan,M., Florea,L., Halpern,A., Hannenhalli,S., Kravitz,S., Levy,S., Mobarry,C., Reinert,K., Remington,K., Abu-Threideh,J., Beasley,E., Biddick,K., Bonazzi,V., Brandon,R., Cargill,M., Chandramouliswaran,I., Charlab,R., Chaturvedi,K., Deng,Z., Di Francesco,V., Dunn,P., Eilbeck,K., Evangelista,C., Gabrielian,A.E., Gan,W., Ge,W., Gong,F., Gu,Z., Guan,P., Heiman,T.J., Higgins,M.E., Ji,R.R., Ke,Z., Ketchum,K.A., Lai,Z., Lei,Y., Li,Z., Li,J., Liang,Y., Lin,X., Lu,F., Merkulov,G.V., Milshina,N., Moore,H.M., Naik,A.K., Narayan,V.A., Neelam,B., Nusskern,D., Rusch,D.B., Salzberg,S., Shao,W., Shue,B., Sun,J., Wang,Z., Wang,A., Wang,X., Wang,J., Wei,M., Wides,R., Xiao,C., Yan,C., Yao,A., Ye,J., Zhan,M., Zhang,W., Zhang,H., Zhao,Q., Zheng,L., Zhong,F., Zhong,W., Zhu,S., Zhao,S., Gilbert,D., Baumhueter,S., Spier,G., Carter,C., Cravchik,A., Woodage,T., Ali,F., An,H., Awe,A., Baldwin,D., Baden,H., Barnstead,M., Barrow,I., Beeson,K., Busam,D., Carver,A., Center,A., Cheng,M.L., Curry,L., Danaher,S., Davenport,L., Desilets,R., Dietz,S., Dodson,K., Doup,L., Ferriera,S., Garg,N., Gluecksmann,A., Hart,B., Haynes,J., Haynes,C., Heiner,C., Hladun,S., Hostin,D., Houck,J., Howland,T., Ibegwam,C., Johnson,J., Kalush,F., Kline,L., Koduru,S., Love,A., Mann,F., May,D., McCawley,S., McIntosh,T., McMullen,I., Moy,M., Moy,L., Murphy,B., Nelson,K., Pfannkoch,C., Pratts,E., Puri,V., Qureshi,H., Reardon,M., Rodriguez,R., Rogers,Y.H., Romblad,D., Ruhfel,B., Scott,R., Sitter,C., Smallwood,M., Stewart,E., Strong,R., Suh,E., Thomas,R., Tint,N.N., Tse,S., Vech,C., Wang,G., Wetter,J., Williams,S., Williams,M., Windsor,S., Winn-Deen,E., Wolfe,K., Zaveri,J., Zaveri,K., Abril,J.F., Guigo,R., Campbell,M.J., Sjolander,K.V., Karlak,B., Kejariwal,A., Mi,H., Lazareva,B., Hatton,T., Narechania,A., Diemer,K., Muruganujan,A., Guo,N., Sato,S., Bafna,V., Istrail,S., Lippert,R., Schwartz,R., Walenz,B., Yooseph,S., Allen,D., Basu,A., Baxendale,J., Blick,L., Caminha,M., Carnes-Stine,J., Caulk,P., Chiang,Y.H., Coyne,M., Dahlke,C., Mays,A., Dombroski,M., Donnelly,M., Ely,D., Esparham,S., Fosler,C., Gire,H., Glanowski,S., Glasser,K., Glodek,A., Gorokhov,M., Graham,K., Gropman,B., Harris,M., Heil,J., Henderson,S., Hoover,J., Jennings,D., Jordan,C., Jordan,J., Kasha,J., Kagan,L., Kraft,C., Levitsky,A., Lewis,M., Liu,X., Lopez,J., Ma,D., Majoros,W., McDaniel,J., Murphy,S., Newman,M., Nguyen,T., Nguyen,N., Nodell,M., Pan,S., Peck,J., Peterson,M., Rowe,W., Sanders,R., Scott,J., Simpson,M., Smith,T., Sprague,A., Stockwell,T., Turner,R., Venter,E., Wang,M., Wen,M., Wu,D., Wu,M., Xia,A., Zandieh,A. and Zhu,X. ',\n",
       " 'Istrail,S., Sutton,G., Florea,L., Halpern,A.L., Mobarry,C.M., Lippert,R., Walenz,B., Shatkay,H., Dew,I., Miller,J.R., Flanigan,M.J., Edwards,N.J., Bolanos,R., Fasulo,D., Halldorsson,B.V., Hannenhalli,S., Turner,R., Yooseph,S., Lu,F., Nusskern,D.R., Shue,B.C., Zheng,X.H., Zhong,F., Delcher,A.L., Huson,D.H., Kravitz,S.A., Mouchard,L., Reinert,K., Remington,K.A., Clark,A.G., Waterman,M.S., Eichler,E.E., Adams,M.D., Hunkapiller,M.W., Myers,E.W. and Venter,J.C. ',\n",
       " 'Kirkness,E.F., Bafna,V., Halpern,A.L., Levy,S., Remington,K., Rusch,D.B., Delcher,A.L., Pop,M., Wang,W., Fraser,C.M. and Venter,J.C. ',\n",
       " 'Kirkness,E., Bafna,V., Halpern,A.L., Levy,S., Remington,K., Rusch,D.B., Delcher,A., Pop,M., Wang,W., Venter,C. and Fraser,C.M. ']"
      ]
     },
     "execution_count": 475,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "authors_list = []\n",
    "existe = re.findall(r'AUTHORS\\s+.*?(?=TITLE)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for auth in existe:\n",
    "        m = re.match( r'AUTHORS\\s+(.+)', auth, re.DOTALL )\n",
    "        authors_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "authors_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TITLE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 476,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Evidence that an amoeba acquired a chloroplast by retaining part of an engulfed eukaryotic alga ',\n",
       " 'Direct Submission ',\n",
       " 'The structure of the human CD2 gene and its expression in transgenic mice ',\n",
       " 'Microbial populations from the subantarctic marine ecosystem ',\n",
       " 'Direct Submission ',\n",
       " 'Expressed genes in Ciona intestinalis (2002c) ',\n",
       " 'Generation of EST sequences from Nicotiana benthamiana ',\n",
       " 'Use of BAC End Sequences from Library RPCI-11 for Sequence-Ready Map Building ',\n",
       " 'The sequence of the human genome ',\n",
       " 'Direct Submission ',\n",
       " 'The sequence of the human genome ',\n",
       " 'Direct Submission ',\n",
       " 'The dog genome: survey sequencing and comparative analysis ',\n",
       " 'Direct Submission ']"
      ]
     },
     "execution_count": 476,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "titles = []\n",
    "existe = re.findall(r'TITLE\\s+.*?(?=JOURNAL)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for title in existe:\n",
    "        m = re.match( r'TITLE\\s+(.+)', title, re.DOTALL )\n",
    "        titles.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "titles"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "PUBMED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 682,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1723776', '1723776', '12040188 ', '1723776']\n",
      "[[''], ['1723776'], ['1723776'], [''], [''], ['12040188 '], ['1723776'], [''], ['']]\n"
     ]
    }
   ],
   "source": [
    "pub_list = []\n",
    "existe = re.findall(r'PUBMED\\s+.*?(?=REFERENCE|COMMENT)', locus, re.DOTALL)\n",
    "# print(existe)\n",
    "if existe:\n",
    "    for pubm in existe:\n",
    "        m = re.match( r'PUBMED\\s+(.+)?(?=\\w)', pubm, re.DOTALL )\n",
    "        pub_list.append(re.sub(r'\\s+\\w+.*', ' ', m.group(1) ) )\n",
    "print(pub_list)\n",
    "\n",
    "#procurar os ficheiros que tem o campo PUBMED, se não tiver vai valor vazio na lista (para fazer a relação genbank-PM)\n",
    "file_locus_pub_list = []\n",
    "for x in locus_list:\n",
    "    aux_pub_list = []\n",
    "    existe = re.findall(r'PUBMED\\s+.*?(?=REFERENCE|COMMENT)', x, re.DOTALL)\n",
    "    # print(existe)\n",
    "    if existe:\n",
    "        for pubm in existe:\n",
    "            m = re.match( r'PUBMED\\s+(.+)?(?=\\w)', pubm, re.DOTALL )\n",
    "\n",
    "            aux_pub_list.append(re.sub(r'\\s+\\w+.*', ' ', m.group(1) ) )\n",
    "    else:\n",
    "        aux_pub_list.append('')\n",
    "    file_locus_pub_list.append(aux_pub_list)\n",
    "\n",
    "print(file_locus_pub_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 683,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['https://pubmed.ncbi.nlm.nih.gov/1723776/', 'https://pubmed.ncbi.nlm.nih.gov/1723776/', 'https://pubmed.ncbi.nlm.nih.gov/12040188 /', 'https://pubmed.ncbi.nlm.nih.gov/1723776/']\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "def url_get_id_pm(item):\n",
    "    url_id_pm = []\n",
    "    for id_pm in item:\n",
    "        url = f\"https://pubmed.ncbi.nlm.nih.gov/{id_pm}/\"\n",
    "        url_id_pm.append(url)\n",
    "    return url_id_pm\n",
    "\n",
    "pm = \"\"\n",
    "soup = []\n",
    "content_pm = []\n",
    "urls = url_get_id_pm(pub_list)\n",
    "print(urls)\n",
    "\n",
    "for url in urls:\n",
    "\tr2 = requests.get(url)\n",
    "\tpm += r2.content.decode('utf-8')\n",
    "\tsoup.append(BeautifulSoup(r2.content, 'html.parser'))\n",
    "\tcontent_pm.append(r2.content.decode('utf-8'))\n",
    "\n",
    "#print(soup)\n",
    "#print(content_pm)\n",
    "\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "AUTHOR, JOURNAL, TITLE, AFFILIATION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 684,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "from Bio import SeqIO\n",
    "from Bio import Entrez\n",
    "from Bio import Medline\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 685,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "['A comparison of whole-genome shotgun-derived mouse chromosome 16 and the human genome.', 'Comparison of three intracellular markers for combined electrophysiological, morphological and immunohistochemical analyses.']\n",
      "['The high degree of similarity between the mouse and human genomes is demonstrated through analysis of the sequence of mouse chromosome 16 (Mmu 16), which was obtained as part of a whole-genome shotgun assembly of the mouse genome. The mouse genome is about 10% smaller than the human genome, owing to a lower repetitive DNA content. Comparison of the structure and protein-coding potential of Mmu 16 with that of the homologous segments of the human genome identifies regions of conserved synteny with human chromosomes (Hsa) 3, 8, 12, 16, 21, and 22. Gene content and order are highly conserved between Mmu 16 and the syntenic blocks of the human genome. Of the 731 predicted genes on Mmu 16, 509 align with orthologs on the corresponding portions of the human genome, 44 are likely paralogous to these genes, and 164 genes have homologs elsewhere in the human genome; there are 14 genes for which we could find no human counterpart.', 'Hypothalamic paraventricular and supraoptic neurons were recorded intracellularly in coronal slices and injected with Lucifer yellow, ethidium bromide or biocytin. Electrical properties, morphological staining and neurophysin immunohistochemistry were compared among the 3 markers. Lucifer yellow electrodes had a high resistance and frequently blocked during experiments. Neurons recorded with Lucifer yellow electrodes had low input resistances and low-amplitude, broad spikes. Lucifer yellow labeling in whole mount was highly fluorescent, revealing distal dendrites and axons. Of cells injected with Lucifer yellow, 64% were recovered but were faint after immunohistochemical processing. Recordings with ethidium bromide electrodes were similar to controls, although electrode blockage sometimes occurred. Only somata and proximal dendrites of ethidium bromide-filled neurons were visible in whole-mount. Forty percent of cells injected with ethidium bromide were recovered after immunohistochemical processing; these were invariably faint. Recordings with biocytin-filled electrodes were similar to control recordings. Biocytin-filled, HRP-labeled cells showed distal dendrites and often dendritic spines and axons in 50-75-microns sections. Seventy percent of biocytin-injected cells labeled with fluorescent markers were recovered and remained strongly labeled after immunohistochemical processing. Biocytin had the best electrical and staining properties for combined electrophysiological and anatomical studies.']\n",
      "[['Mural RJ', 'Adams MD', 'Myers EW', 'Smith HO', 'Miklos GL', 'Wides R', 'Halpern A', 'Li PW', 'Sutton GG', 'Nadeau J', 'Salzberg SL', 'Holt RA', 'Kodira CD', 'Lu F', 'Chen L', 'Deng Z', 'Evangelista CC', 'Gan W', 'Heiman TJ', 'Li J', 'Li Z', 'Merkulov GV', 'Milshina NV', 'Naik AK', 'Qi R', 'Shue BC', 'Wang A', 'Wang J', 'Wang X', 'Yan X', 'Ye J', 'Yooseph S', 'Zhao Q', 'Zheng L', 'Zhu SC', 'Biddick K', 'Bolanos R', 'Delcher AL', 'Dew IM', 'Fasulo D', 'Flanigan MJ', 'Huson DH', 'Kravitz SA', 'Miller JR', 'Mobarry CM', 'Reinert K', 'Remington KA', 'Zhang Q', 'Zheng XH', 'Nusskern DR', 'Lai Z', 'Lei Y', 'Zhong W', 'Yao A', 'Guan P', 'Ji RR', 'Gu Z', 'Wang ZY', 'Zhong F', 'Xiao C', 'Chiang CC', 'Yandell M', 'Wortman JR', 'Amanatides PG', 'Hladun SL', 'Pratts EC', 'Johnson JE', 'Dodson KL', 'Woodford KJ', 'Evans CA', 'Gropman B', 'Rusch DB', 'Venter E', 'Wang M', 'Smith TJ', 'Houck JT', 'Tompkins DE', 'Haynes C', 'Jacob D', 'Chin SH', 'Allen DR', 'Dahlke CE', 'Sanders R', 'Li K', 'Liu X', 'Levitsky AA', 'Majoros WH', 'Chen Q', 'Xia AC', 'Lopez JR', 'Donnelly MT', 'Newman MH', 'Glodek A', 'Kraft CL', 'Nodell M', 'Ali F', 'An HJ', 'Baldwin-Pitts D', 'Beeson KY', 'Cai S', 'Carnes M', 'Carver A', 'Caulk PM', 'Center A', 'Chen YH', 'Cheng ML', 'Coyne MD', 'Crowder M', 'Danaher S', 'Davenport LB', 'Desilets R', 'Dietz SM', 'Doup L', 'Dullaghan P', 'Ferriera S', 'Fosler CR', 'Gire HC', 'Gluecksmann A', 'Gocayne JD', 'Gray J', 'Hart B', 'Haynes J', 'Hoover J', 'Howland T', 'Ibegwam C', 'Jalali M', 'Johns D', 'Kline L', 'Ma DS', 'MacCawley S', 'Magoon A', 'Mann F', 'May D', 'McIntosh TC', 'Mehta S', 'Moy L', 'Moy MC', 'Murphy BJ', 'Murphy SD', 'Nelson KA', 'Nuri Z', 'Parker KA', 'Prudhomme AC', 'Puri VN', 'Qureshi H', 'Raley JC', 'Reardon MS', 'Regier MA', 'Rogers YH', 'Romblad DL', 'Schutz J', 'Scott JL', 'Scott R', 'Sitter CD', 'Smallwood M', 'Sprague AC', 'Stewart E', 'Strong RV', 'Suh E', 'Sylvester K', 'Thomas R', 'Tint NN', 'Tsonis C', 'Wang G', 'Wang G', 'Williams MS', 'Williams SM', 'Windsor SM', 'Wolfe K', 'Wu MM', 'Zaveri J', 'Chaturvedi K', 'Gabrielian AE', 'Ke Z', 'Sun J', 'Subramanian G', 'Venter JC', 'Pfannkoch CM', 'Barnstead M', 'Stephenson LD'], ['Tasker JG', 'Hoffman NW', 'Dudek FE']]\n",
      "[['Celera Genomics, 45 West Gude Drive, Rockville, MD 20850, USA. richard.mural@celera.com'], ['Mental Retardation Research Center, UCLA School of Medicine.']]\n",
      "['Science. 2002 May 31;296(5573):1661-71. doi: 10.1126/science.1069193.', 'J Neurosci Methods. 1991 Jul;38(2-3):129-43. doi: 10.1016/0165-0270(91)90163-t.']\n"
     ]
    }
   ],
   "source": [
    "Ids=[]\n",
    "title_list = []\n",
    "abstract = []\n",
    "author = []\n",
    "affiliation = []\n",
    "journal = []\n",
    "database = 'PubMed'\n",
    "Entrez.email= \"pg49836@alunos.uminho.pt\"\n",
    "# idlist= pub_list\n",
    "idlist = []\n",
    "handle = Entrez.efetch(db=database, id=pub_list, rettype=\"medline\", retmode=\"text\") \n",
    "records = Medline.parse(handle)\n",
    "description=[]\n",
    "for info in records:\n",
    "    title_list.append(info.get(\"TI\", \"-\"))\n",
    "    author.append(info.get(\"AU\", \"-\"))\n",
    "    journal.append(info.get(\"SO\", \"-\"))\n",
    "    affiliation.append(info.get(\"AD\", \"-\") )\n",
    "    abstract.append(info.get(\"AB\", \"-\"))\n",
    "    print()\n",
    "print(title_list)\n",
    "print(abstract)\n",
    "print(author)\n",
    "print(affiliation)\n",
    "print(journal)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "CONNECTION TO DATABASE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 686,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INSERT INTO LOCUS (Locus_name, Sequence_length, Molecule_type, GB_division, Source, Organism, Accession) VALUES (\"X67302\", \"369\", \"mRNA linear\", \"ROD\", \"Rattus norvegicus (Norway rat) \", \"Rattus norvegicus\", \"X67302 \"), (\"BW778499\", \"339\", \"mRNA linear\", \"EST\", \"Branchiostoma floridae (Florida lancelet) \", \"Branchiostoma floridae\", \"BW778499 \"), (\"BW778502\", \"451\", \"mRNA linear\", \"EST\", \"Branchiostoma floridae (Florida lancelet) \", \"Branchiostoma floridae\", \"BW778502 \"), (\"BB232249\", \"291\", \"mRNA linear\", \"EST\", \"Mus musculus (house mouse) \", \"Mus musculus\", \"BB232249 \"), (\"BB232248\", \"297\", \"mRNA linear\", \"EST\", \"Mus musculus (house mouse) \", \"Mus musculus\", \"BB232248 \"), (\"AAHY01223610\", \"742\", \"DNA linear\", \"ROD\", \"Mus musculus (house mouse) \", \"Mus musculus\", \"AAHY01223610 AAHY01000000 \"), (\"BW778501\", \"336\", \"mRNA linear\", \"EST\", \"Branchiostoma floridae (Florida lancelet) \", \"Branchiostoma floridae\", \"BW778501 \"), (\"AAGJ01210774\", \"910\", \"DNA linear\", \"INV\", \"Strongylocentrotus purpuratus \", \"Strongylocentrotus purpuratus\", \"AAGJ01210774 AAGJ01000000 \"), (\"AAGV01843290\", \"1143\", \"DNA linear\", \"MAM\", \"Dasypus novemcinctus (nine-banded armadillo) \", \"Dasypus novemcinctus\", \"AAGV01843290 AAGV01000000 \")\n",
      "9 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"127.0.0.1\",\n",
    "    user =\"user\",\n",
    "    password =\"password\",\n",
    "        database =\"teste\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `LOCUS` (\n",
    "        `Id_locus` INT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n",
    "        `Accession` VARCHAR(100) NOT NULL,\n",
    "        `Sequence_length` INT NULL,\n",
    "        `Molecule_type` VARCHAR(20) NULL,\n",
    "        `GB_division` VARCHAR(10) NULL,\n",
    "        `Source` VARCHAR(200) NULL,\n",
    "        `Organism` VARCHAR(200) NULL,\n",
    "        `Locus_name` VARCHAR(45) NULL\n",
    "        );\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    sql_loc = \"INSERT INTO LOCUS (Locus_name, Sequence_length, Molecule_type, GB_division, Source, Organism, Accession) VALUES\"\n",
    "    loc_name_list = list(dict.fromkeys(loc_name_list))  #retirar duplicados \n",
    "    for index in range(len(loc_name_list)):\n",
    "        #fazer a verificação se existe locus name repetido (evitar duplicados)\n",
    "        select_ids_loc = f\"SELECT Id_Locus FROM LOCUS WHERE Locus_name = \\\"{loc_name_list[index]}\\\"\"\n",
    "        Cursor.execute(select_ids_loc)\n",
    "        myresult_locus = Cursor.fetchall()\n",
    "        id_loc = None\n",
    "        for res_loc, in myresult_locus:\n",
    "            id_loc = res_loc\n",
    "        if id_loc is None: \n",
    "            if index == len(loc_name_list)-1:\n",
    "                sql_loc += f\" (\\\"{loc_name_list[index]}\\\", \\\"{length[index]}\\\", \\\"{mol_type[index]}\\\", \\\"{gb_div[index]}\\\", \\\"{source_list[index]}\\\", \\\"{org_list[index]}\\\", \\\"{acc_list[index]}\\\")\"\n",
    "            else:\n",
    "                sql_loc += f\" (\\\"{loc_name_list[index]}\\\", \\\"{length[index]}\\\", \\\"{mol_type[index]}\\\", \\\"{gb_div[index]}\\\", \\\"{source_list[index]}\\\", \\\"{org_list[index]}\\\", \\\"{acc_list[index]}\\\"),\"\n",
    "    print(sql_loc)\n",
    "    if sql_loc != \"INSERT INTO LOCUS (Locus_name, Sequence_length, Molecule_type, GB_division, Source, Organism, Accession) VALUES\":  #não executa nada se não forem dados valores diferentes dos já inseridos\n",
    "        if sql_loc.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "            sql_loc = sql_loc[:-1] #remove utimo caracter da string\n",
    "        Cursor.execute(sql_loc)\n",
    "\n",
    "    DataBase.commit()\n",
    "\n",
    "    print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 687,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"127.0.0.1\",\n",
    "    user =\"user\",\n",
    "    password =\"password\",\n",
    "        database =\"teste\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `GENBANK` (\n",
    "        `Id_GB` INT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n",
    "        `GB_id` VARCHAR(45) NOT NULL,\n",
    "        `Definition` VARCHAR(400) NULL,\n",
    "        `Modification_date` VARCHAR(11) NULL,\n",
    "        `Version` VARCHAR(20) NOT NULL UNIQUE,\n",
    "        `LOCUS_Id_locus` INT NOT NULL,\n",
    "        `ORIGIN_Id_origin` INT NULL\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    sql_gb = \"INSERT INTO GENBANK (GB_Id, Modification_date, Definition, Version, LOCUS_Id_locus, ORIGIN_Id_origin) VALUES\"\n",
    "    \n",
    "\n",
    "    for index in range(len(version_list)):\n",
    "        sql_insert_origin = \"INSERT INTO ORIGIN (Sequence, Length, Adenine, Citosine, Timine , Guanine) VALUES\"\n",
    "        select_ids_origin = f\"SELECT Id_Origin FROM ORIGIN WHERE Sequence = \\\"{seq_origin[index]}\\\"\"\n",
    "        Cursor.execute(select_ids_origin)\n",
    "        myresult_origin = Cursor.fetchall()\n",
    "        id_ori = None\n",
    "        for res_ori, in myresult_origin:\n",
    "            id_ori = res_ori\n",
    "\n",
    "        if id_ori is None: \n",
    "            if index == len(loc_name_list)-1:\n",
    "                sql_insert_origin += f\" (\\\"{seq_origin[index]}\\\", \\\"{length[index]}\\\", \\\"{seq_count[index][0]}\\\", \\\"{seq_count[index][1]}\\\", \\\"{seq_count[index][2]}\\\",  \\\"{seq_count[index][3]}\\\")\"\n",
    "            else:\n",
    "                sql_insert_origin += f\" (\\\"{seq_origin[index]}\\\", \\\"{length[index]}\\\", \\\"{seq_count[index][0]}\\\", \\\"{seq_count[index][1]}\\\", \\\"{seq_count[index][2]}\\\",  \\\"{seq_count[index][3]}\\\"),\"\n",
    "\n",
    "        if sql_insert_origin != \"INSERT INTO ORIGIN (Sequence, Length, Adenine, Citosine, Timine , Guanine) VALUES\":\n",
    "            # select_ids_origin += f\" (\\\"{seq_origin[index]}\\\", \\\"{length[index]}\\\", \\\"{seq_count[index][0]}\\\", \\\"{seq_count[index][1]}\\\", \\\"{seq_count[index][2]}\\\",  \\\"{seq_count[index][3]}\\\")\"\n",
    "            if sql_insert_origin.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                sql_insert_origin = sql_insert_origin[:-1] #remove utimo caracter da string\n",
    "            Cursor.execute(sql_insert_origin)\n",
    "\n",
    "            # print(Cursor.lastrowid)  #buscar o id do origin inserido\n",
    "            id_inserted_origin = Cursor.lastrowid\n",
    "        else:\n",
    "            id_inserted_origin = id_ori\n",
    "\n",
    "\n",
    "        #fazer a verificação se existe locus repetido (evitar duplicados)\n",
    "        select_locus = f\"SELECT Id_locus from LOCUS where Locus_name = '{loc_name_list[index]}'\"\n",
    "        Cursor.execute(select_locus)\n",
    "        myresult = Cursor.fetchall()\n",
    "        id_locus = None\n",
    "        for res, in myresult:\n",
    "            id_locus = res\n",
    "        \n",
    "        if id_locus is not None:\n",
    "            select_gb = f\"SELECT Id_GB from GENBANK where GB_Id = '{loc_name_list[index]}'\"\n",
    "            Cursor.execute(select_gb)\n",
    "            myresult_gb = Cursor.fetchall()\n",
    "            id_genbank = None\n",
    "            for res_gb, in myresult_gb:\n",
    "                id_genbank = res_gb\n",
    "            \n",
    "            if id_genbank is None:\n",
    "                if index == len(version_list)-1:\n",
    "                    sql_gb += f\" (\\\"{loc_name_list[index]}\\\", \\\"{mod_date[index]}\\\", \\\"{defi_list[index]}\\\", \\\"{version_list[index]}\\\", \\\"{id_locus}\\\", \\\"{id_inserted_origin}\\\")\"\n",
    "                else:\n",
    "                    sql_gb += f\" (\\\"{loc_name_list[index]}\\\", \\\"{mod_date[index]}\\\", \\\"{defi_list[index]}\\\", \\\"{version_list[index]}\\\", \\\"{id_locus}\\\", \\\"{id_inserted_origin}\\\"),\"\n",
    "                \n",
    "    if sql_gb != \"INSERT INTO GENBANK (GB_Id, Modification_date, Definition, Version, LOCUS_Id_locus, ORIGIN_Id_origin) VALUES\":\n",
    "\n",
    "        if sql_gb.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "            sql_gb = sql_gb[:-1] #remove utimo caracter da string\n",
    "            \n",
    "        Cursor.execute(sql_gb)\n",
    "\n",
    "        DataBase.commit()\n",
    "\n",
    "        print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 688,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INSERT INTO PUBMED (PM_ID, Journal, Title, Abstract) VALUES (\"1723776\", \"Science. 2002 May 31;296(5573):1661-71. doi: 10.1126/science.1069193.\", \"A comparison of whole-genome shotgun-derived mouse chromosome 16 and the human genome.\", \"The high degree of similarity between the mouse and human genomes is demonstrated through analysis of the sequence of mouse chromosome 16 (Mmu 16), which was obtained as part of a whole-genome shotgun assembly of the mouse genome. The mouse genome is about 10% smaller than the human genome, owing to a lower repetitive DNA content. Comparison of the structure and protein-coding potential of Mmu 16 with that of the homologous segments of the human genome identifies regions of conserved synteny with human chromosomes (Hsa) 3, 8, 12, 16, 21, and 22. Gene content and order are highly conserved between Mmu 16 and the syntenic blocks of the human genome. Of the 731 predicted genes on Mmu 16, 509 align with orthologs on the corresponding portions of the human genome, 44 are likely paralogous to these genes, and 164 genes have homologs elsewhere in the human genome; there are 14 genes for which we could find no human counterpart.\"), (\"12040188 \", \"J Neurosci Methods. 1991 Jul;38(2-3):129-43. doi: 10.1016/0165-0270(91)90163-t.\", \"Comparison of three intracellular markers for combined electrophysiological, morphological and immunohistochemical analyses.\", \"Hypothalamic paraventricular and supraoptic neurons were recorded intracellularly in coronal slices and injected with Lucifer yellow, ethidium bromide or biocytin. Electrical properties, morphological staining and neurophysin immunohistochemistry were compared among the 3 markers. Lucifer yellow electrodes had a high resistance and frequently blocked during experiments. Neurons recorded with Lucifer yellow electrodes had low input resistances and low-amplitude, broad spikes. Lucifer yellow labeling in whole mount was highly fluorescent, revealing distal dendrites and axons. Of cells injected with Lucifer yellow, 64% were recovered but were faint after immunohistochemical processing. Recordings with ethidium bromide electrodes were similar to controls, although electrode blockage sometimes occurred. Only somata and proximal dendrites of ethidium bromide-filled neurons were visible in whole-mount. Forty percent of cells injected with ethidium bromide were recovered after immunohistochemical processing; these were invariably faint. Recordings with biocytin-filled electrodes were similar to control recordings. Biocytin-filled, HRP-labeled cells showed distal dendrites and often dendritic spines and axons in 50-75-microns sections. Seventy percent of biocytin-injected cells labeled with fluorescent markers were recovered and remained strongly labeled after immunohistochemical processing. Biocytin had the best electrical and staining properties for combined electrophysiological and anatomical studies.\")\n",
      "2 record inserted.\n",
      "['1723776', '12040188 ']\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"127.0.0.1\",\n",
    "    user =\"user\",\n",
    "    password =\"password\",\n",
    "        database =\"teste\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `PUBMED` (\n",
    "        `Id_PM` INT NOT NULL AUTO_INCREMENT,\n",
    "        `PM_ID` INT NOT NULL,\n",
    "        `Journal` VARCHAR(100) NULL,\n",
    "        `Title` VARCHAR(500) NULL,\n",
    "        `Abstract` LONGTEXT NULL,\n",
    "        PRIMARY KEY (`Id_PM`)\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    \n",
    "    #print(journal)\n",
    "    #print(len(title_list))\n",
    "    #print(len(abstract))\n",
    "    #print(len(pub_list))\n",
    "    sql_pm = \"INSERT INTO PUBMED (PM_ID, Journal, Title, Abstract) VALUES\"\n",
    "    pub_list = list(dict.fromkeys(pub_list))  #tirar os duplicados (também n apanha o ultimo)\n",
    "    #print(pub_list)\n",
    "    for index in range(len(pub_list)):\n",
    "        select_pm = f\"SELECT Id_PM from PUBMED where PM_Id = '{pub_list[index]}'\"\n",
    "        Cursor.execute(select_pm)\n",
    "        myresult_pubmed = Cursor.fetchall()\n",
    "        id_pubmed = None\n",
    "        for res_pm, in myresult_pubmed:\n",
    "            id_pubmed = res_pm\n",
    "        \n",
    "        if id_pubmed is None:\n",
    "            if index == len(pub_list)-1:\n",
    "                sql_pm += f\" (\\\"{pub_list[index]}\\\", \\\"{journal[index]}\\\", \\\"{title_list[index]}\\\", \\\"{abstract[index]}\\\")\"\n",
    "            else:\n",
    "                sql_pm += f\" (\\\"{pub_list[index]}\\\", \\\"{journal[index]}\\\", \\\"{title_list[index]}\\\", \\\"{abstract[index]}\\\"),\"\n",
    "        \n",
    "    if sql_pm != \"INSERT INTO PUBMED (PM_ID, Journal, Title, Abstract) VALUES\":\n",
    "        # select_ids_origin += f\" (\\\"{seq_origin[index]}\\\", \\\"{length[index]}\\\", \\\"{seq_count[index][0]}\\\", \\\"{seq_count[index][1]}\\\", \\\"{seq_count[index][2]}\\\",  \\\"{seq_count[index][3]}\\\")\"\n",
    "        if sql_pm.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "            sql_pm = sql_pm[:-1] #remove utimo caracter da string\n",
    "        print(sql_pm)\n",
    "        Cursor.execute(sql_pm)\n",
    "\n",
    "        DataBase.commit()\n",
    "\n",
    "        print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    Cursor.close()\n",
    "    print(pub_list)\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 689,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\n",
      "9\n",
      "[1]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"BW778499\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM pm_gb_relationship WHERE PUBMED_Id_PM = \"1\" and GENBANK_Id_GB = \"2\"\n",
      "INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (1, 2)\n",
      "INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (1, 2)\n",
      "1 record inserted.\n",
      "[1]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"BW778502\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM pm_gb_relationship WHERE PUBMED_Id_PM = \"1\" and GENBANK_Id_GB = \"3\"\n",
      "INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (1, 3)\n",
      "INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (1, 3)\n",
      "1 record inserted.\n",
      "[2]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"AAHY01223610\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM pm_gb_relationship WHERE PUBMED_Id_PM = \"2\" and GENBANK_Id_GB = \"6\"\n",
      "INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (2, 6)\n",
      "INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (2, 6)\n",
      "1 record inserted.\n",
      "[1]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"BW778501\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM pm_gb_relationship WHERE PUBMED_Id_PM = \"1\" and GENBANK_Id_GB = \"7\"\n",
      "INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (1, 7)\n",
      "INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (1, 7)\n",
      "1 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"127.0.0.1\",\n",
    "    user =\"user\",\n",
    "    password =\"password\",\n",
    "        database =\"teste\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `AUTHORS` (\n",
    "        `Id_Author` INT NOT NULL AUTO_INCREMENT,\n",
    "        `Name` VARCHAR(45) NULL,\n",
    "        `AFFILIATION_Id_affiliation` INT NULL,\n",
    "        PRIMARY KEY (`Id_Author`)\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    print(len(file_locus_pub_list))\n",
    "    print(len(loc_name_list))\n",
    "\n",
    "    for index_ids in range(len(file_locus_pub_list)):\n",
    "        '''for index in range(len(author[index_ids])):\n",
    "            #verificar se existe autor (evitar duplicados)\n",
    "            select_ids_auth = f\"SELECT Id_Author FROM AUTHORS WHERE Name = \\\"{author[index_ids]}\\\"\"\n",
    "            Cursor.execute(select_ids_auth)\n",
    "            myresult_auth = Cursor.fetchall()\n",
    "            id_auth = None\n",
    "            for res, in myresult_auth:\n",
    "                id_affiliation = res'''\n",
    "\n",
    "        if file_locus_pub_list[index_ids][0] != '':\n",
    "            db_id_pm = []\n",
    "            for pm_ids in file_locus_pub_list[index_ids]:\n",
    "                select_pm_gb = f\"SELECT Id_PM FROM PUBMED WHERE PM_ID = \\\"{pm_ids}\\\"\"\n",
    "                \n",
    "                #print(select_pm_gb)\n",
    "                Cursor.execute(select_pm_gb)\n",
    "                myresult_pm = Cursor.fetchall()\n",
    "                id_pm = None\n",
    "                for res, in myresult_pm:\n",
    "                    id_pm = res\n",
    "                db_id_pm.append(id_pm)\n",
    "\n",
    "            print(db_id_pm)\n",
    "            select_gb_pm = f\"SELECT Id_GB FROM GENBANK WHERE GB_ID= \\\"{loc_name_list[index_ids]}\\\"\"\n",
    "            print(select_gb_pm)\n",
    "            Cursor.execute(select_gb_pm)\n",
    "            myresult_gb = Cursor.fetchall()\n",
    "            id_gb = None\n",
    "            for res, in myresult_gb:\n",
    "                id_gb = res\n",
    "\n",
    "            index = 0\n",
    "            sql_pm_gb_relation = \"INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES\" \n",
    "            for id_pm in db_id_pm:\n",
    "                if id_gb is not None and id_pm is not None:      \n",
    "                    #verificar se existe relação entre ids\n",
    "                    select_rel_gb_pm = f\"SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM pm_gb_relationship WHERE PUBMED_Id_PM = \\\"{id_pm}\\\" and GENBANK_Id_GB = \\\"{id_gb}\\\"\"\n",
    "                    print(select_rel_gb_pm)\n",
    "                    Cursor.execute(select_rel_gb_pm)\n",
    "                    myresult_rel_pm_gb = Cursor.fetchall()\n",
    "                    id_rel_pm_gb = None\n",
    "                    for res_PUBMED_Id_PM,res_GENBANK_Id_GB, in myresult_rel_pm_gb:   #apenas selecionamos um dos id pq se não houver relação serão ambos null\n",
    "                        id_rel_pm_gb = res_PUBMED_Id_PM     \n",
    "\n",
    "                    #se null:\n",
    "                    if id_rel_pm_gb is None:\n",
    "                        if index == len(db_id_pm)-1:\n",
    "                            sql_pm_gb_relation += f\" ({id_pm}, {id_gb})\"\n",
    "                        else:\n",
    "                            sql_pm_gb_relation += f\" ({id_pm}, {id_gb}),\"\n",
    "                index += 1\n",
    "\n",
    "            print(sql_pm_gb_relation)\n",
    "            if sql_pm_gb_relation != \"INSERT INTO pm_gb_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES\":\n",
    "                #remove ultimo caracter se este for uma \",\"\n",
    "                if sql_pm_gb_relation.endswith(\",\"):\n",
    "                    sql_pm_gb_relation = sql_pm_gb_relation[:-1] #remove utimo caracter da string\n",
    "\n",
    "                print(sql_pm_gb_relation)\n",
    "                Cursor.execute(sql_pm_gb_relation)\n",
    "                DataBase.commit()\n",
    "                print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 690,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INSERT INTO AFFILIATION (Description) VALUES (\"Celera Genomics, 45 West Gude Drive, Rockville, MD 20850, USA. richard.mural@celera.com\")\n",
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Mural RJ\", 1), (\"Adams MD\", null), (\"Myers EW\", null), (\"Smith HO\", null), (\"Miklos GL\", null), (\"Wides R\", null), (\"Halpern A\", null), (\"Li PW\", null), (\"Sutton GG\", null), (\"Nadeau J\", null), (\"Salzberg SL\", null), (\"Holt RA\", null), (\"Kodira CD\", null), (\"Lu F\", null), (\"Chen L\", null), (\"Deng Z\", null), (\"Evangelista CC\", null), (\"Gan W\", null), (\"Heiman TJ\", null), (\"Li J\", null), (\"Li Z\", null), (\"Merkulov GV\", null), (\"Milshina NV\", null), (\"Naik AK\", null), (\"Qi R\", null), (\"Shue BC\", null), (\"Wang A\", null), (\"Wang J\", null), (\"Wang X\", null), (\"Yan X\", null), (\"Ye J\", null), (\"Yooseph S\", null), (\"Zhao Q\", null), (\"Zheng L\", null), (\"Zhu SC\", null), (\"Biddick K\", null), (\"Bolanos R\", null), (\"Delcher AL\", null), (\"Dew IM\", null), (\"Fasulo D\", null), (\"Flanigan MJ\", null), (\"Huson DH\", null), (\"Kravitz SA\", null), (\"Miller JR\", null), (\"Mobarry CM\", null), (\"Reinert K\", null), (\"Remington KA\", null), (\"Zhang Q\", null), (\"Zheng XH\", null), (\"Nusskern DR\", null), (\"Lai Z\", null), (\"Lei Y\", null), (\"Zhong W\", null), (\"Yao A\", null), (\"Guan P\", null), (\"Ji RR\", null), (\"Gu Z\", null), (\"Wang ZY\", null), (\"Zhong F\", null), (\"Xiao C\", null), (\"Chiang CC\", null), (\"Yandell M\", null), (\"Wortman JR\", null), (\"Amanatides PG\", null), (\"Hladun SL\", null), (\"Pratts EC\", null), (\"Johnson JE\", null), (\"Dodson KL\", null), (\"Woodford KJ\", null), (\"Evans CA\", null), (\"Gropman B\", null), (\"Rusch DB\", null), (\"Venter E\", null), (\"Wang M\", null), (\"Smith TJ\", null), (\"Houck JT\", null), (\"Tompkins DE\", null), (\"Haynes C\", null), (\"Jacob D\", null), (\"Chin SH\", null), (\"Allen DR\", null), (\"Dahlke CE\", null), (\"Sanders R\", null), (\"Li K\", null), (\"Liu X\", null), (\"Levitsky AA\", null), (\"Majoros WH\", null), (\"Chen Q\", null), (\"Xia AC\", null), (\"Lopez JR\", null), (\"Donnelly MT\", null), (\"Newman MH\", null), (\"Glodek A\", null), (\"Kraft CL\", null), (\"Nodell M\", null), (\"Ali F\", null), (\"An HJ\", null), (\"Baldwin-Pitts D\", null), (\"Beeson KY\", null), (\"Cai S\", null), (\"Carnes M\", null), (\"Carver A\", null), (\"Caulk PM\", null), (\"Center A\", null), (\"Chen YH\", null), (\"Cheng ML\", null), (\"Coyne MD\", null), (\"Crowder M\", null), (\"Danaher S\", null), (\"Davenport LB\", null), (\"Desilets R\", null), (\"Dietz SM\", null), (\"Doup L\", null), (\"Dullaghan P\", null), (\"Ferriera S\", null), (\"Fosler CR\", null), (\"Gire HC\", null), (\"Gluecksmann A\", null), (\"Gocayne JD\", null), (\"Gray J\", null), (\"Hart B\", null), (\"Haynes J\", null), (\"Hoover J\", null), (\"Howland T\", null), (\"Ibegwam C\", null), (\"Jalali M\", null), (\"Johns D\", null), (\"Kline L\", null), (\"Ma DS\", null), (\"MacCawley S\", null), (\"Magoon A\", null), (\"Mann F\", null), (\"May D\", null), (\"McIntosh TC\", null), (\"Mehta S\", null), (\"Moy L\", null), (\"Moy MC\", null), (\"Murphy BJ\", null), (\"Murphy SD\", null), (\"Nelson KA\", null), (\"Nuri Z\", null), (\"Parker KA\", null), (\"Prudhomme AC\", null), (\"Puri VN\", null), (\"Qureshi H\", null), (\"Raley JC\", null), (\"Reardon MS\", null), (\"Regier MA\", null), (\"Rogers YH\", null), (\"Romblad DL\", null), (\"Schutz J\", null), (\"Scott JL\", null), (\"Scott R\", null), (\"Sitter CD\", null), (\"Smallwood M\", null), (\"Sprague AC\", null), (\"Stewart E\", null), (\"Strong RV\", null), (\"Suh E\", null), (\"Sylvester K\", null), (\"Thomas R\", null), (\"Tint NN\", null), (\"Tsonis C\", null), (\"Wang G\", null), (\"Wang G\", null), (\"Williams MS\", null), (\"Williams SM\", null), (\"Windsor SM\", null), (\"Wolfe K\", null), (\"Wu MM\", null), (\"Zaveri J\", null), (\"Chaturvedi K\", null), (\"Gabrielian AE\", null), (\"Ke Z\", null), (\"Sun J\", null), (\"Subramanian G\", null), (\"Venter JC\", null), (\"Pfannkoch CM\", null), (\"Barnstead M\", null), (\"Stephenson LD\", null)\n",
      "INSERT INTO AFFILIATION (Description) VALUES (\"Mental Retardation Research Center, UCLA School of Medicine.\")\n",
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Tasker JG\", 2), (\"Hoffman NW\", null), (\"Dudek FE\", null)\n",
      "3 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"127.0.0.1\",\n",
    "    user =\"user\",\n",
    "    password =\"password\",\n",
    "        database =\"teste\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `AUTHORS` (\n",
    "        `Id_Author` INT NOT NULL AUTO_INCREMENT,\n",
    "        `Name` VARCHAR(45) NULL,\n",
    "        `AFFILIATION_Id_affiliation` INT NULL,\n",
    "        PRIMARY KEY (`Id_Author`)\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "\n",
    "    for index_pubmed in range(len(pub_list)):\n",
    "        \n",
    "        affiliation_ids = []\n",
    "        for index in range(len(affiliation[index_pubmed])):\n",
    "\n",
    "            #verificar se existe afiliação (evitar duplicados)\n",
    "            select_affiliation = f\"SELECT Id_affiliation FROM AFFILIATION WHERE Description = \\\"{affiliation[index_pubmed][index]}\\\"\"\n",
    "            Cursor.execute(select_affiliation)\n",
    "            myresult_af = Cursor.fetchall()\n",
    "            id_affiliation = None\n",
    "            for res, in myresult_af:\n",
    "                id_affiliation = res\n",
    "\n",
    "\n",
    "            if id_affiliation is not None:\n",
    "                affiliation_ids.append(id_affiliation)\n",
    "            else:\n",
    "                sql_insert_affiliation = \"INSERT INTO AFFILIATION (Description) VALUES\" \n",
    "                sql_insert_affiliation += f\" (\\\"{affiliation[index_pubmed][index]}\\\")\"\n",
    "                print(sql_insert_affiliation)\n",
    "                Cursor.execute(sql_insert_affiliation)\n",
    "                # print(Cursor.lastrowid)  #buscar o id do affiliation inserido\n",
    "                affiliation_ids.append(Cursor.lastrowid)\n",
    "\n",
    "\n",
    "        sql = \"INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES\"\n",
    "        \n",
    "        for index in range(len(author[index_pubmed])):\n",
    "            select_author = f\"SELECT Id_Author FROM AUTHORS WHERE Name = \\\"{author[index_pubmed][index]}\\\"\"\n",
    "            Cursor.execute(select_author)\n",
    "            myresult_author = Cursor.fetchall()\n",
    "            id_author = None\n",
    "            for res, in myresult_author:\n",
    "                id_author = res\n",
    "\n",
    "\n",
    "            if id_author is None:\n",
    "                if index > len(affiliation_ids)-1:\n",
    "                    if index == len(author[index_pubmed])-1:\n",
    "                        sql += f\" (\\\"{author[index_pubmed][index]}\\\", null)\"\n",
    "                    else:\n",
    "                        sql += f\" (\\\"{author[index_pubmed][index]}\\\", null),\"\n",
    "                else:\n",
    "                    if index == len(author[index_pubmed])-1:\n",
    "                        sql += f\" (\\\"{author[index_pubmed][index]}\\\", {affiliation_ids[index]})\"\n",
    "                    else:\n",
    "                        sql += f\" (\\\"{author[index_pubmed][index]}\\\", {affiliation_ids[index]}),\"\n",
    "    \n",
    "        print(sql)\n",
    "        if sql != \"INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES\":\n",
    "                #remove ultimo caracter se este for uma \",\"\n",
    "            if sql.endswith(\",\"):\n",
    "                sql = sql[:-1]\n",
    "\n",
    "            Cursor.execute(sql)\n",
    "\n",
    "    DataBase.commit()\n",
    "\n",
    "    print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 691,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\n",
      "9\n",
      "1\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"1\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"2\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"3\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"4\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"5\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"6\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"7\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"8\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"9\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"10\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"11\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"12\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"13\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"14\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"15\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"16\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"17\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"18\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"19\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"20\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"21\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"22\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"23\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"24\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"25\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"26\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"27\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"28\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"29\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"30\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"31\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"32\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"33\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"34\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"35\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"36\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"37\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"38\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"39\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"40\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"41\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"42\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"43\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"44\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"45\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"46\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"47\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"48\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"49\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"50\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"51\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"52\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"53\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"54\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"55\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"56\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"57\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"58\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"59\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"60\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"61\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"62\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"63\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"64\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"65\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"66\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"67\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"68\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"69\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"70\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"71\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"72\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"73\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"74\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"75\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"76\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"77\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"78\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"79\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"80\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"81\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"82\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"83\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"84\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"85\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"86\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"87\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"88\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"89\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"90\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"91\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"92\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"93\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"94\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"95\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"96\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"97\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"98\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"99\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"100\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"101\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"102\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"103\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"104\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"105\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"106\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"107\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"108\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"109\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"110\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"111\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"112\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"113\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"114\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"115\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"116\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"117\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"118\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"119\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"120\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"121\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"122\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"123\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"124\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"125\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"126\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"127\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"128\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"129\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"130\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"131\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"132\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"133\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"134\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"135\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"136\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"137\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"138\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"139\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"140\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"141\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"142\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"143\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"144\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"145\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"146\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"147\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"148\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"149\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"150\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"151\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"152\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"153\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"154\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"155\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"156\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"157\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"158\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"159\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"160\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"161\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"162\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"163\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"165\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"166\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"167\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"168\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"169\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"170\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"171\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"172\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"173\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"174\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"175\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"176\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"177\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"178\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"179\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"1\" and AUTHORS_Id_Author = \"180\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (1, 1), (1, 2), (1, 3), (1, 4), (1, 5), (1, 6), (1, 7), (1, 8), (1, 9), (1, 10), (1, 11), (1, 12), (1, 13), (1, 14), (1, 15), (1, 16), (1, 17), (1, 18), (1, 19), (1, 20), (1, 21), (1, 22), (1, 23), (1, 24), (1, 25), (1, 26), (1, 27), (1, 28), (1, 29), (1, 30), (1, 31), (1, 32), (1, 33), (1, 34), (1, 35), (1, 36), (1, 37), (1, 38), (1, 39), (1, 40), (1, 41), (1, 42), (1, 43), (1, 44), (1, 45), (1, 46), (1, 47), (1, 48), (1, 49), (1, 50), (1, 51), (1, 52), (1, 53), (1, 54), (1, 55), (1, 56), (1, 57), (1, 58), (1, 59), (1, 60), (1, 61), (1, 62), (1, 63), (1, 64), (1, 65), (1, 66), (1, 67), (1, 68), (1, 69), (1, 70), (1, 71), (1, 72), (1, 73), (1, 74), (1, 75), (1, 76), (1, 77), (1, 78), (1, 79), (1, 80), (1, 81), (1, 82), (1, 83), (1, 84), (1, 85), (1, 86), (1, 87), (1, 88), (1, 89), (1, 90), (1, 91), (1, 92), (1, 93), (1, 94), (1, 95), (1, 96), (1, 97), (1, 98), (1, 99), (1, 100), (1, 101), (1, 102), (1, 103), (1, 104), (1, 105), (1, 106), (1, 107), (1, 108), (1, 109), (1, 110), (1, 111), (1, 112), (1, 113), (1, 114), (1, 115), (1, 116), (1, 117), (1, 118), (1, 119), (1, 120), (1, 121), (1, 122), (1, 123), (1, 124), (1, 125), (1, 126), (1, 127), (1, 128), (1, 129), (1, 130), (1, 131), (1, 132), (1, 133), (1, 134), (1, 135), (1, 136), (1, 137), (1, 138), (1, 139), (1, 140), (1, 141), (1, 142), (1, 143), (1, 144), (1, 145), (1, 146), (1, 147), (1, 148), (1, 149), (1, 150), (1, 151), (1, 152), (1, 153), (1, 154), (1, 155), (1, 156), (1, 157), (1, 158), (1, 159), (1, 160), (1, 161), (1, 162), (1, 163), (1, 165), (1, 166), (1, 167), (1, 168), (1, 169), (1, 170), (1, 171), (1, 172), (1, 173), (1, 174), (1, 175), (1, 176), (1, 177), (1, 178), (1, 179), (1, 180)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (1, 1), (1, 2), (1, 3), (1, 4), (1, 5), (1, 6), (1, 7), (1, 8), (1, 9), (1, 10), (1, 11), (1, 12), (1, 13), (1, 14), (1, 15), (1, 16), (1, 17), (1, 18), (1, 19), (1, 20), (1, 21), (1, 22), (1, 23), (1, 24), (1, 25), (1, 26), (1, 27), (1, 28), (1, 29), (1, 30), (1, 31), (1, 32), (1, 33), (1, 34), (1, 35), (1, 36), (1, 37), (1, 38), (1, 39), (1, 40), (1, 41), (1, 42), (1, 43), (1, 44), (1, 45), (1, 46), (1, 47), (1, 48), (1, 49), (1, 50), (1, 51), (1, 52), (1, 53), (1, 54), (1, 55), (1, 56), (1, 57), (1, 58), (1, 59), (1, 60), (1, 61), (1, 62), (1, 63), (1, 64), (1, 65), (1, 66), (1, 67), (1, 68), (1, 69), (1, 70), (1, 71), (1, 72), (1, 73), (1, 74), (1, 75), (1, 76), (1, 77), (1, 78), (1, 79), (1, 80), (1, 81), (1, 82), (1, 83), (1, 84), (1, 85), (1, 86), (1, 87), (1, 88), (1, 89), (1, 90), (1, 91), (1, 92), (1, 93), (1, 94), (1, 95), (1, 96), (1, 97), (1, 98), (1, 99), (1, 100), (1, 101), (1, 102), (1, 103), (1, 104), (1, 105), (1, 106), (1, 107), (1, 108), (1, 109), (1, 110), (1, 111), (1, 112), (1, 113), (1, 114), (1, 115), (1, 116), (1, 117), (1, 118), (1, 119), (1, 120), (1, 121), (1, 122), (1, 123), (1, 124), (1, 125), (1, 126), (1, 127), (1, 128), (1, 129), (1, 130), (1, 131), (1, 132), (1, 133), (1, 134), (1, 135), (1, 136), (1, 137), (1, 138), (1, 139), (1, 140), (1, 141), (1, 142), (1, 143), (1, 144), (1, 145), (1, 146), (1, 147), (1, 148), (1, 149), (1, 150), (1, 151), (1, 152), (1, 153), (1, 154), (1, 155), (1, 156), (1, 157), (1, 158), (1, 159), (1, 160), (1, 161), (1, 162), (1, 163), (1, 165), (1, 166), (1, 167), (1, 168), (1, 169), (1, 170), (1, 171), (1, 172), (1, 173), (1, 174), (1, 175), (1, 176), (1, 177), (1, 178), (1, 179), (1, 180)\n",
      "179 record inserted.\n",
      "2\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"2\" and AUTHORS_Id_Author = \"181\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"2\" and AUTHORS_Id_Author = \"182\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"2\" and AUTHORS_Id_Author = \"183\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (2, 181), (2, 182), (2, 183)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (2, 181), (2, 182), (2, 183)\n",
      "3 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"127.0.0.1\",\n",
    "    user =\"user\",\n",
    "    password =\"password\",\n",
    "        database =\"teste\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `teste`.`PM_AUTHORS_relationship` (\n",
    "        `PUBMED_Id_PM` INT NOT NULL,\n",
    "        `AUTHORS_Id_Author` INT NOT NULL,\n",
    "        PRIMARY KEY (`PUBMED_Id_PM`, `AUTHORS_Id_Author`)\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    print(len(file_locus_pub_list))\n",
    "    print(len(loc_name_list))\n",
    "\n",
    "    for index_pm in range(len(pub_list)):\n",
    "\n",
    "        if pub_list[index_pm] != '':\n",
    "            select_pm_at = f\"SELECT Id_PM FROM PUBMED WHERE PM_ID = \\\"{pub_list[index_pm]}\\\"\"\n",
    "        \n",
    "            Cursor.execute(select_pm_at)\n",
    "            myresult_pm_at = Cursor.fetchall()\n",
    "            id_pm_at = None\n",
    "            for res, in myresult_pm_at:\n",
    "                id_pm_at = res\n",
    "\n",
    "            print(id_pm_at)\n",
    "            author_id_list = []\n",
    "            for xx in author[index_pm]:\n",
    "                select_at_pm = f\"SELECT Id_Author FROM AUTHORS WHERE Name= \\\"{xx}\\\"\"\n",
    "                Cursor.execute(select_at_pm)\n",
    "                myresult_at = Cursor.fetchall()\n",
    "                id_at = None\n",
    "                for res, in myresult_at:\n",
    "                    id_at = res\n",
    "                author_id_list.append(id_at)\n",
    "\n",
    "            index = 0\n",
    "            sql_at_pm_relation = \"INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES\" \n",
    "            author_id_list = list(dict.fromkeys(author_id_list))\n",
    "            for id_at in author_id_list:\n",
    "                if id_at is not None and id_pm_at is not None:      \n",
    "                    #verificar se existe relação entre ids\n",
    "                    select_rel_at_pm = f\"SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \\\"{id_pm_at}\\\" and AUTHORS_Id_Author = \\\"{id_at}\\\"\"\n",
    "                    print(select_rel_at_pm)\n",
    "                    Cursor.execute(select_rel_at_pm)\n",
    "                    myresult_rel_pm_at = Cursor.fetchall()\n",
    "                    id_rel_pm_at = None\n",
    "                    for res_PUBMED_Id_PM,res_AUTHORS_Id_Author, in myresult_rel_pm_at:   #apenas selecionamos um dos id pq se não houver relação serão ambos null\n",
    "                        id_rel_pm_at = res_PUBMED_Id_PM     \n",
    "\n",
    "                    #se null:\n",
    "                    if id_rel_pm_at is None:\n",
    "                        if index == len(author_id_list)-1:\n",
    "                            sql_at_pm_relation += f\" ({id_pm_at}, {id_at})\"\n",
    "                        else:\n",
    "                            sql_at_pm_relation += f\" ({id_pm_at}, {id_at}),\"\n",
    "                index += 1\n",
    "\n",
    "            print(sql_at_pm_relation)\n",
    "            if sql_at_pm_relation != \"INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES\":\n",
    "                #remove ultimo caracter se este for uma \",\"\n",
    "                if sql_at_pm_relation.endswith(\",\"):\n",
    "                    sql_at_pm_relation = sql_at_pm_relation[:-1] #remove utimo caracter da string\n",
    "\n",
    "                print(sql_at_pm_relation)\n",
    "                Cursor.execute(sql_at_pm_relation)\n",
    "                DataBase.commit()\n",
    "                print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 702,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[['Oct 2']], [], [], [], [], [], [], [], []]"
      ]
     },
     "execution_count": 702,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 695,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "[[], [], ['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']], [], [], []]\n",
      "['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>369](+)', ['Oct 2 Pou-Homeo domain protein'], ['CAA47715.1'], ['FTQGDVGLAMGKLYGNDFSQTTISRFEALNLSFKNMCKLKPLLEKWLNDAETMSVDSSLPSPNQLSSPSLGFDGLPGRRRKKRTSIETNVRFALEKSFLANQKPTSEEILLIAEQLHMEKEVI']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "2\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "3\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "4\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "5\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "6\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "7\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "8\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "9\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"127.0.0.1\",\n",
    "    user =\"user\",\n",
    "    password =\"password\",\n",
    "        database =\"teste\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `teste`.`FEATURES` (\n",
    "        `Id_F` INT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n",
    "        `Gene` VARCHAR(45) NULL,\n",
    "        `Source_location` VARCHAR(45) NULL\n",
    "    \n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `teste`.`CDS` (\n",
    "        `Id_CDS` INT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n",
    "        `Translation_seq` longtext  NULL,\n",
    "        `Location` VARCHAR(45)  NULL,\n",
    "        `Product` VARCHAR(45)  NULL,\n",
    "        `Protein_ID` VARCHAR(45) NULL\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "\n",
    "    for index_gb in range(len(loc_name_list)):\n",
    "\n",
    "        #get genbank id\n",
    "        select_gb = f\"SELECT Id_GB FROM GENBANK WHERE GB_ID= \\\"{loc_name_list[index_gb]}\\\"\"\n",
    "        Cursor.execute(select_gb)\n",
    "        myresult_gb = Cursor.fetchall()\n",
    "        id_gb = None\n",
    "        for res, in myresult_gb:\n",
    "            id_gb = res\n",
    "        \n",
    "        print(id_gb)\n",
    "        if id_gb is not None:\n",
    "            for index_ft in range(len(source[index_gb])):\n",
    "                if source[index_gb][index_ft] != '':  #verificar se existe \"source\" na bd\n",
    "                    sql_insert_source = \"INSERT INTO FEATURES (Gene, Source_location) VALUES\"\n",
    "                    select_ft = f\"SELECT Id_F FROM FEATURES WHERE Source_location = \\\"{source[index_gb][index_ft]}\\\"\"\n",
    "            \n",
    "                    Cursor.execute(select_ft)\n",
    "                    myresult_ft = Cursor.fetchall()\n",
    "                    id_ft = None\n",
    "                    for res_ft, in myresult_ft:\n",
    "                        id_ft = res_ft\n",
    "\n",
    "                    if id_ft is None: #se n existe, insere\n",
    "                        if len(gene[index_gb]) > 0:\n",
    "                            sql_insert_source += f\" (\\\"{gene[index_gb][index_ft]}\\\", \\\"{source[index_gb][index_ft]}\\\")\"\n",
    "                        else:\n",
    "                            sql_insert_source += f\" ( null, \\\"{source[index_gb][index_ft]}\\\")\"\n",
    "\n",
    "                        if sql_insert_source.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                            sql_insert_source = sql_insert_source[:-1] #remove utimo caracter da string\n",
    "                        Cursor.execute(sql_insert_source)\n",
    "\n",
    "                        # print(Cursor.lastrowid)  #buscar o id do feature inserido\n",
    "                        id_inserted_feature = Cursor.lastrowid\n",
    "                    else:\n",
    "                        id_inserted_feature = id_ft\n",
    "\n",
    "                    #relationship between genbank and feature\n",
    "                    sql_insert_gb_ft = \"INSERT INTO GB_FT_relationship (GENBANK_Id_GB, FEATURES_Id_F) VALUES\"\n",
    "                    select_gb_ft = f\"SELECT GENBANK_Id_GB FROM GB_FT_relationship WHERE GENBANK_Id_GB = {id_gb} and FEATURES_Id_F = {id_inserted_feature}\"\n",
    "            \n",
    "                    Cursor.execute(select_gb_ft)\n",
    "                    myresult_gb_ft = Cursor.fetchall()\n",
    "                    id_gb_ft = None\n",
    "                    for res_gb_ft, in myresult_gb_ft:\n",
    "                        id_gb_ft = res_gb_ft\n",
    "\n",
    "                    if id_gb_ft is None: #se n existe, insere\n",
    "                        sql_insert_gb_ft += f\" ({id_gb}, {id_inserted_feature})\"\n",
    "\n",
    "                        if sql_insert_gb_ft.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                            sql_insert_gb_ft = sql_insert_gb_ft[:-1] #remove utimo caracter da string\n",
    "                        Cursor.execute(sql_insert_gb_ft)\n",
    "\n",
    "            for index_cds in range(len(CDS[index_gb])):\n",
    "                print(CDS[index_gb])\n",
    "                print(CDS[index_gb][index_cds])\n",
    "                if CDS[index_gb][index_cds]:  #verificar se existe \"cds\" na bd\n",
    "                    sql_insert_cds = \"INSERT INTO CDS (Translation_seq, Location,Product,Protein_ID) VALUES\"\n",
    "                    select_cds = f\"SELECT Id_CDS FROM CDS WHERE Translation_seq = \\\"{CDS[index_gb][index_cds][3]}\\\"\"\n",
    "            \n",
    "                    Cursor.execute(select_cds)\n",
    "                    myresult_cds = Cursor.fetchall()\n",
    "                    id_cds = None\n",
    "                    for res_cds, in myresult_cds:\n",
    "                        id_cds = res_cds\n",
    "\n",
    "                    if id_cds is None: #se n existe, insere\n",
    "                        sql_insert_cds += f\" (\\\"{CDS[index_gb][index_cds][3]}\\\",\\\"{CDS[index_gb][index_cds][0]}\\\",\\\"{CDS[index_gb][index_cds][1]}\\\", \\\"{CDS[index_gb][index_cds][2]}\\\")\"\n",
    "\n",
    "                        if sql_insert_cds.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                            sql_insert_cds = sql_insert_cds[:-1] #remove utimo caracter da string\n",
    "                        Cursor.execute(sql_insert_cds)\n",
    "\n",
    "                        # print(Cursor.lastrowid)  #buscar o id do cds inserido\n",
    "                        id_inserted_cds = Cursor.lastrowid\n",
    "                    else:\n",
    "                        id_inserted_cds = id_cds\n",
    "                    \n",
    "                    #relationship between cds and genbank\n",
    "                    sql_insert_gb_cds = \"INSERT INTO CDS_GB_relationship (GENBANK_Id_GB, CDS_Id_CDS) VALUES\"\n",
    "                    select_gb_cds = f\"SELECT GENBANK_Id_GB FROM CDS_GB_relationship WHERE GENBANK_Id_GB = {id_gb} and CDS_Id_CDS = {id_inserted_cds}\"\n",
    "            \n",
    "                    Cursor.execute(select_gb_cds)\n",
    "                    myresult_gb_cds = Cursor.fetchall()\n",
    "                    id_gb_cds = None\n",
    "                    for res_gb_cds, in myresult_gb_cds:\n",
    "                        id_gb_ft = res_gb_cds\n",
    "\n",
    "                    if id_gb_ft is None: #se n existe, insere\n",
    "                        sql_insert_gb_cds += f\" ({id_gb}, {id_inserted_cds})\"\n",
    "\n",
    "                        if sql_insert_gb_cds.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                            sql_insert_gb_cds = sql_insert_gb_cds[:-1] #remove utimo caracter da string\n",
    "                        Cursor.execute(sql_insert_gb_cds)\n",
    "\n",
    "                DataBase.commit()\n",
    "                print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.5"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "fb4569285eef3a3450cb62085a5b1e0da4bce0af555edc33dcf29baf3acc1368"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
