{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TRABALHO BD"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A partir de webscrapping, ir buscar o número de ids escolhido (máx 20) de termos à escolha do utilizador e importar as informações para a base de dados"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "WEBSCRAPPING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['https://www.ncbi.nlm.nih.gov/gene/?term=malaria']\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "\n",
    "def validateInsertedText(item):\n",
    "    return re.match(r'[\\d|a-z|A-Z]+(\\_?[\\d|a-z|A-Z]+)*', item)\n",
    "\n",
    "def url_get(item):\n",
    "    url_list = []\n",
    "    url = \"https://www.ncbi.nlm.nih.gov/gene/?term={}\".format(item)\n",
    "    url_list.append(url)\n",
    "    return url_list\n",
    "\n",
    "query = input(\"Defina o termo que pretende pesquisar [bacteria/yeast/human...] \\n Se o termo tiver mais do que uma palavra, separe por _: \")\n",
    "try:\n",
    "    \n",
    "    if not validateInsertedText(query):\n",
    "        raise Exception\n",
    "    else:   \n",
    "        print(url_get(query))\n",
    "\n",
    "except Exception:\n",
    "    print(f'Inválido. Insira um termo válido. O termo que inseriu foi: \"{query}\"')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests, sys, json\n",
    "content = []\n",
    "for url in url_get(query):\n",
    "    r = requests.get(url)\n",
    "    content.append(r.content)\n",
    "# print(content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# Parsing the HTML\n",
    "for c in content:\n",
    "    soup = BeautifulSoup(r.content, 'html.parser')\n",
    "    res = soup.get_text()\n",
    "# print(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['3043',\n",
       " '112314',\n",
       " '116906',\n",
       " '112313',\n",
       " '117578',\n",
       " '100101431',\n",
       " '100035820',\n",
       " '100034930',\n",
       " '493123',\n",
       " '493122',\n",
       " '100036393',\n",
       " '100036129',\n",
       " '100035705',\n",
       " '246749',\n",
       " '100036439',\n",
       " '100036484',\n",
       " '104057',\n",
       " '2539',\n",
       " '28',\n",
       " '118446222101']"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "existe = re.findall(r'\\s*ID:\\s*\\d*(?=\\D)', res, re.DOTALL)\n",
    "id_genes = ','.join(existe)\n",
    "id_g = re.sub(r'\\s*ID:\\n*\\s+', '', id_genes)\n",
    "id_gene = id_g.split(',')\n",
    "id_gene\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10\n",
      "['3043', '112314', '116906', '112313', '117578', '100101431', '100035820', '100034930', '493123', '493122']\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    number_of_genes = int(input(\"Número de genes a obter (máx: 20): \"))\n",
    "    print(number_of_genes)\n",
    "    genes = id_gene[0:number_of_genes]\n",
    "    print(genes)\n",
    "except Exception:\n",
    "    print(\"INVALID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#BUSCAR LINKS DOS GENES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['https://www.ncbi.nlm.nih.gov/nuccore/3043',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/112314',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/116906',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/112313',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/117578',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/100101431',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/100035820',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/100034930',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/493123',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/493122']"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import time\n",
    "def url_get_id(item):\n",
    "    url_id = []\n",
    "    for id in genes:\n",
    "        url = f\"https://www.ncbi.nlm.nih.gov/nuccore/{id}\"\n",
    "        url_id.append(url)\n",
    "    return url_id\n",
    "url_get_id(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "id_content = []\n",
    "for url in url_get_id(genes):\n",
    "    r = requests.get(url)\n",
    "    id_content.append(r.content)\n",
    "# id_content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       X56237                  1077 bp    mRNA    linear   PLN 12-JUN-2006\n",
      "DEFINITION  N. crassa NUO-32 mRNA for NADH dehydrogenase 32 kD subunit.\n",
      "ACCESSION   X56237\n",
      "VERSION     X56237.1\n",
      "KEYWORDS    NADH dehydrogenase (ubiquinone); NADH dehydrogenase subunit; NUO-32\n",
      "            gene.\n",
      "SOURCE      Neurospora crassa\n",
      "  ORGANISM  Neurospora crassa\n",
      "            Eukaryota; Fungi; Dikarya; Ascomycota; Pezizomycotina;\n",
      "            Sordariomycetes; Sordariomycetidae; Sordariales; Sordariaceae;\n",
      "            Neurospora.\n",
      "REFERENCE   1  (bases 1 to 1077)\n",
      "  AUTHORS   Van der Pas,J.C., Rohlen,D.A., Weidner,U. and Weiss,H.\n",
      "  TITLE     Primary structure of the nuclear-encoded 29.9 kDa subunit of NADH:\n",
      "            ubiquinone reductase from Neurospora crassa mitochondria\n",
      "  JOURNAL   Biochim. Biophys. Acta 1089 (3), 389-390 (1991)\n",
      "   PUBMED   1830489\n",
      "REFERENCE   2  (bases 1 to 1077)\n",
      "  AUTHORS   Roelen,D.A.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (25-NOV-1990) Roehlen D.A., Institut fuer Biochemie,\n",
      "            Universitaet Duesseldorf, Universitaetsstr. 1, 4000 Duesseldorf 2\n",
      "            Germany\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..1077\n",
      "                     /organism=\"Neurospora crassa\"\n",
      "                     /mol_type=\"mRNA\"\n",
      "                     /strain=\"SL74\"\n",
      "                     /db_xref=\"taxon:5141\"\n",
      "     gene            1..1077\n",
      "                     /gene=\"NUO-32\"\n",
      "     mRNA            1..1077\n",
      "                     /gene=\"NUO-32\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "     CDS             68..889\n",
      "                     /gene=\"NUO-32\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"NADH dehydrogenase subunit\"\n",
      "                     /protein_id=\"CAA39694.1\"\n",
      "                     /db_xref=\"GOA:P24919\"\n",
      "                     /db_xref=\"InterPro:IPR006806\"\n",
      "                     /db_xref=\"UniProtKB/Swiss-Prot:P24919\"\n",
      "                     /translation=\"MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALL\n",
      "                     YLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKS\n",
      "                     DLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVV\n",
      "                     GAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQ\n",
      "                     IAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE\"\n",
      "     sig_peptide     68..91\n",
      "                     /gene=\"NUO-32\"\n",
      "                     /note=\"for NADH dehydrogenase subunit\"\n",
      "     mat_peptide     92..886\n",
      "                     /gene=\"NUO-32\"\n",
      "                     /product=\"NADH dehydrogenase subunit\"\n",
      "     polyA_site      1077\n",
      "                     /gene=\"NUO-32\"\n",
      "ORIGIN      \n",
      "        1 ccacaatcca ccccacccca ccggaccccg agtcagtcaa caacctccac gacacgactc\n",
      "       61 ggaaaccatg cgcgcggcac tccgattgct cgcgaccgcg acggctaccg tccgcccgtc\n",
      "      121 cgcccgcttc ctcaaacccg gctccccgac cggcctcacc ggcctcggca cccacccgtc\n",
      "      181 gccgcgctcc gccctgctgt acctctacaa ccacaccctc gacaagctca agcagatccc\n",
      "      241 cgagcactcg ctgtaccgcc agtccgccga ggccctgacc aagcaccgcc tcgccatcgt\n",
      "      301 cgagcagtac gtgcccgacg gctacgacgc ctggcaggag cgcgcccgca agctgctcga\n",
      "      361 gaagcacaag agcgacctca cggcccgcca gttcgacggc cagcatgccc gcctagtcga\n",
      "      421 aggccccgac ggccgcgctt acttcatccg ccagatggtc cccccgcagg actggcgcga\n",
      "      481 cgtcgagtgg gatggtgccg tcctggatcc gcacttttcc tgggttcaga ccggcgagga\n",
      "      541 tgtcgtcggc gccgtcaagc tggaagacag cgacaagctg ctcgagctgg acaagatcag\n",
      "      601 ggaatcagac ccggttgcct atcgccaggg tctcagggat ctcggtatca agatgggcgg\n",
      "      661 tgttgtcgag gacaagagcc ccgtcgagtg ggagtcggag ccgccgttga gcgctgaaca\n",
      "      721 gattgctgag atggaggcca ggattggttc tggcttgatt gaggaggttg ttcaggttgc\n",
      "      781 cgagggcgag cttaagctgg ttgacattat gacccaggcc aggccttggg aagctcttga\n",
      "      841 ggaggaggcg ccagagggcc aatggactta ctttgagcgt aaggagtaaa caaacatcga\n",
      "      901 cagacataga gaatgggaac acaggaccaa ccaggccatg acggccacct attttatgag\n",
      "      961 cgttacgaag aagaagggag gacaggcatt tctcagccag ttgtacctat accataccac\n",
      "     1021 gccgggacat gtatataatc atcgactttg gcaggcggga atgactatgc cctcctt\n",
      "//\n",
      "\n",
      "Error: F a i l e d  t o  r e t r i e v e  s e q u e n c e :  1 1 2 3 1 4 \n",
      "\n",
      "\n",
      "Error: F a i l e d  t o  r e t r i e v e  s e q u e n c e :  1 1 6 9 0 6 \n",
      "\n",
      "\n",
      "Error: F a i l e d  t o  r e t r i e v e  s e q u e n c e :  1 1 2 3 1 3 \n",
      "\n",
      "\n",
      "Error: F a i l e d  t o  r e t r i e v e  s e q u e n c e :  1 1 7 5 7 8 \n",
      "\n",
      "\n",
      "LOCUS       AAADK0077081              20 bp    RNA     linear   ROD 20-SEP-2005\n",
      "DEFINITION  Mus musculus strain C57BL/6Jx129/Sv.\n",
      "ACCESSION   AAADK0077081\n",
      "VERSION     AAADK0077081.1\n",
      "KEYWORDS    MGA; 5'-end tag; CAGE (Cap Analysis Gene Expression).\n",
      "SOURCE      Mus musculus (house mouse)\n",
      "  ORGANISM  Mus musculus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Mus; Mus.\n",
      "REFERENCE   1\n",
      "  AUTHORS   Shiraki,T., Kondo,S., Katayama,S., Waki,K., Kasukawa,T., Kawaji,H.,\n",
      "            Kodzius,R., Watahiki,A., Nakamura,M., Arakawa,T., Fukuda,S.,\n",
      "            Sasaki,D., Podhajska,A., Harbers,M., Kawai,J., Carninci,P. and\n",
      "            Hayashizaki,Y.\n",
      "  TITLE     Cap analysis gene expression for high-throughput analysis of\n",
      "            transcriptional starting point and identification of promoter usage\n",
      "  JOURNAL   Proc. Natl. Acad. Sci. U.S.A. 100 (26), 15776-15781 (2003)\n",
      "   PUBMED   14663149\n",
      "REFERENCE   2\n",
      "  AUTHORS   Kodzius,R., Matsumura,Y., Kasukawa,T., Shimokawa,K., Fukuda,S.,\n",
      "            Shiraki,T., Nakamura,M., Arakawa,T., Sasaki,D., Kawai,J.,\n",
      "            Harbers,M., Carninci,P. and Hayashizaki,Y.\n",
      "  TITLE     Absolute expression values for mouse transcripts: re-annotation of\n",
      "            the READ expression database by the use of CAGE and EST sequence\n",
      "            tags\n",
      "  JOURNAL   FEBS Lett. 559 (1-3), 22-26 (2004)\n",
      "   PUBMED   14960301\n",
      "REFERENCE   3\n",
      "  AUTHORS   Carninci,P., Kasukawa,T., Katayama,S., Gough,J., Frith,M.C.,\n",
      "            Maeda,N., Oyama,R., Ravasi,T., Lenhard,B., Wells,C., Kodzius,R.,\n",
      "            Shimokawa,K., Bajic,V.B., Brenner,S.E., Batalov,S., Forrest,A.R.,\n",
      "            Zavolan,M., Davis,M.J., Wilming,L.G., Aidinis,V., Allen,J.E.,\n",
      "            Ambesi-Impiombato,A., Apweiler,R., Aturaliya,R.N., Bailey,T.L.,\n",
      "            Bansal,M., Baxter,L., Beisel,K.W., Bersano,T., Bono,H., Chalk,A.M.,\n",
      "            Chiu,K.P., Choudhary,V., Christoffels,A., Clutterbuck,D.R.,\n",
      "            Crowe,M.L., Dalla,E., Dalrymple,B.P., de Bono,B., Della Gatta,G.,\n",
      "            di Bernardo,D., Down,T., Engstrom,P., Fagiolini,M., Faulkner,G.,\n",
      "            Fletcher,C.F., Fukushima,T., Furuno,M., Futaki,S., Gariboldi,M.,\n",
      "            Georgii-Hemming,P., Gingeras,T.R., Gojobori,T., Green,R.E.,\n",
      "            Gustincich,S., Harbers,M., Hayashi,Y., Hensch,T.K., Hirokawa,N.,\n",
      "            Hill,D., Huminiecki,L., Iacono,M., Ikeo,K., Iwama,A., Ishikawa,T.,\n",
      "            Jakt,M., Kanapin,A., Katoh,M., Kawasawa,Y., Kelso,J., Kitamura,H.,\n",
      "            Kitano,H., Kollias,G., Krishnan,S.P., Kruger,A., Kummerfeld,S.K.,\n",
      "            Kurochkin,I.V., Lareau,L.F., Lazarevic,D., Lipovich,L., Liu,J.,\n",
      "            Liuni,S., McWilliam,S., Madan Babu,M., Madera,M., Marchionni,L.,\n",
      "            Matsuda,H., Matsuzawa,S., Miki,H., Mignone,F., Miyake,S.,\n",
      "            Morris,K., Mottagui-Tabar,S., Mulder,N., Nakano,N., Nakauchi,H.,\n",
      "            Ng,P., Nilsson,R., Nishiguchi,S., Nishikawa,S., Nori,F., Ohara,O.,\n",
      "            Okazaki,Y., Orlando,V., Pang,K.C., Pavan,W.J., Pavesi,G.,\n",
      "            Pesole,G., Petrovsky,N., Piazza,S., Reed,J., Reid,J.F., Ring,B.Z.,\n",
      "            Ringwald,M., Rost,B., Ruan,Y., Salzberg,S.L., Sandelin,A.,\n",
      "            Schneider,C., Schonbach,C., Sekiguchi,K., Semple,C.A., Seno,S.,\n",
      "            Sessa,L., Sheng,Y., Shibata,Y., Shimada,H., Shimada,K., Silva,D.,\n",
      "            Sinclair,B., Sperling,S., Stupka,E., Sugiura,K., Sultana,R.,\n",
      "            Takenaka,Y., Taki,K., Tammoja,K., Tan,S.L., Tang,S., Taylor,M.S.,\n",
      "            Tegner,J., Teichmann,S.A., Ueda,H.R., van Nimwegen,E., Verardo,R.,\n",
      "            Wei,C.L., Yagi,K., Yamanishi,H., Zabarovsky,E., Zhu,S., Zimmer,A.,\n",
      "            Hide,W., Bult,C., Grimmond,S.M., Teasdale,R.D., Liu,E.T.,\n",
      "            Brusic,V., Quackenbush,J., Wahlestedt,C., Mattick,J.S., Hume,D.A.,\n",
      "            Kai,C., Sasaki,D., Tomaru,Y., Fukuda,S., Kanamori-Katayama,M.,\n",
      "            Suzuki,M., Aoki,J., Arakawa,T., Iida,J., Imamura,K., Itoh,M.,\n",
      "            Kato,T., Kawaji,H., Kawagashira,N., Kawashima,T., Kojima,M.,\n",
      "            Kondo,S., Konno,H., Nakano,K., Ninomiya,N., Nishio,T., Okada,M.,\n",
      "            Plessy,C., Shibata,K., Shiraki,T., Suzuki,S., Tagami,M., Waki,K.,\n",
      "            Watahiki,A., Okamura-Oho,Y., Suzuki,H., Kawai,J. and Hayashizaki,Y.\n",
      "  CONSRTM   FANTOM Consortium; RIKEN Genome Exploration Research Group and\n",
      "            Genome Science Group (Genome Network Project Core Group)\n",
      "  TITLE     The transcriptional landscape of the mammalian genome\n",
      "  JOURNAL   Science 309 (5740), 1559-1563 (2005)\n",
      "   PUBMED   16141072\n",
      "  REMARK    Erratum:[Science. 2006 Mar 24;311(5768):1713]\n",
      "REFERENCE   4\n",
      "  AUTHORS   Katayama,S., Tomaru,Y., Kasukawa,T., Waki,K., Nakanishi,M.,\n",
      "            Nakamura,M., Nishida,H., Yap,C.C., Suzuki,M., Kawai,J., Suzuki,H.,\n",
      "            Carninci,P., Hayashizaki,Y., Wells,C., Frith,M., Ravasi,T.,\n",
      "            Pang,K.C., Hallinan,J., Mattick,J., Hume,D.A., Lipovich,L.,\n",
      "            Batalov,S., Engstrom,P.G., Mizuno,Y., Faghihi,M.A., Sandelin,A.,\n",
      "            Chalk,A.M., Mottagui-Tabar,S., Liang,Z., Lenhard,B. and\n",
      "            Wahlestedt,C.\n",
      "  CONSRTM   RIKEN Genome Exploration Research Group; Genome Science Group\n",
      "            (Genome Network Project Core Group); FANTOM Consortium\n",
      "  TITLE     Antisense transcription in the mammalian transcriptome\n",
      "  JOURNAL   Science 309 (5740), 1564-1566 (2005)\n",
      "   PUBMED   16141073\n",
      "REFERENCE   5\n",
      "  AUTHORS   Carninci,P., Sandelin,A., Lenhard,B., Katayama,S., Shimokawa,K.,\n",
      "            Ponjavic,J., Semple,C.A., Taylor,M.S., Engstrom,P.G., Frith,M.C.,\n",
      "            Forrest,A.R., Alkema,W.B., Tan,S.L., Plessy,C., Kodzius,R.,\n",
      "            Ravasi,T., Kasukawa,T., Fukuda,S., Kanamori-Katayama,M.,\n",
      "            Kitazume,Y., Kawaji,H., Kai,C., Nakamura,M., Konno,H., Nakano,K.,\n",
      "            Mottagui-Tabar,S., Arner,P., Chesi,A., Gustincich,S.,\n",
      "            Persichetti,F., Suzuki,H., Grimmond,S.M., Wells,C.A., Orlando,V.,\n",
      "            Wahlestedt,C., Liu,E.T., Harbers,M., Kawai,J., Bajic,V.B.,\n",
      "            Hume,D.A. and Hayashizaki,Y.\n",
      "  TITLE     Genome-wide analysis of mammalian promoter architecture and\n",
      "            evolution\n",
      "  JOURNAL   Nat. Genet. 38 (6), 626-635 (2006)\n",
      "   PUBMED   16645617\n",
      "  REMARK    Erratum:[Nat Genet. 2007 Sep;39(9):1174]\n",
      "REFERENCE   6\n",
      "  AUTHORS   Arakawa,T., Carninci,P., Fukuda,S., Hasegawa,A., Hayashida,K.,\n",
      "            Hori,F., Iida,J., Imamura,K., Kai,C., Kanagawa,S., Katayama,S.,\n",
      "            Kawai,J., Kodzius,R., Kojima,M., Kondo,S., Murata,M., Nakamura,M.,\n",
      "            Nishiyori,H., Nomura,K., Ohno,M., Sakazume,N., Sasaki,D.,\n",
      "            Shiraki,T., Tagami,M., Tagami,Y., Waki,K., Watahiki,A. and\n",
      "            Hayashizaki,Y.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (18-APR-2005) Contact:Yoshihide Hayashizaki The Institute\n",
      "            of Physical and Chemical Research (RIKEN), Omics Science Center,\n",
      "            RIKEN Yokohama Institute; 1-7-22 Suehiro-cho, Tsurumi-ku, Yokohama,\n",
      "            Kanagawa 230-0045, Japan URL :http://www.osc.riken.jp/\n",
      "COMMENT     CAGE library was prepared and sequenced in Genome Science\n",
      "            Laboratory (Wako) and Genome Exploration Research Group Genomics\n",
      "            Science Center (GSC) in RIKEN Yokohama Institute\n",
      "            Please visit our web site for further details.\n",
      "            URL:http://www.osc.riken.jp/\n",
      "            The CAGE (cap analysis gene expression) tags are obtained by\n",
      "            sequencing concatamers of DNA tags deriving from the initial 20/21\n",
      "            nucleotides from 5' end mRNAs, prepared in accordance to Shiraki et\n",
      "            al. (Proc Natl Acad Sci U S A. 100, 15776-81, 2003). At first step,\n",
      "            full-length cDNAs were selected with the Cap-Trapper. Next, a\n",
      "            specific linker (Linker1, which contains the ClassIIs restriction\n",
      "            enzyme site MmeI) was ligated to the cDNA. Linker 1 may contain\n",
      "            extra 5 bp sequences tag, and 15 of such different sequences tags\n",
      "            were used to tag different starting RNA samples. Then the second\n",
      "            strand of cDNA synthesized. Resulting double-stranded cDNAs were\n",
      "            cleaved by the restriction enzyme MmeI and a second linker\n",
      "            (Linker2) was ligated to the 2 bp overhang at the MmeI cleaved\n",
      "            site, to produce a 5' 20/21 tag having two linkers at both sides.\n",
      "            The ligation products were separated from unmodified DNA with\n",
      "            magnetic beads. The 5' end cDNA tags were released from the beads,\n",
      "            and the DNA fragments were amplified in a PCR step by using the two\n",
      "            linker-specific primers (Primer1 (uni-PCR), Primer2 (MmeI-PCR)).\n",
      "            The desired 32-37 bp tags were purified and ligated to form\n",
      "            concatamers, and then the concatamer were fractionated and ligated\n",
      "            to the plasmid ZErO-2. The ligations were finally electroporated\n",
      "            into DH10b cells (Invitrogen) and obtained plasmids were sequenced\n",
      "            with forward primers essentially as described with minor\n",
      "            modifications to use zeocin for selection of recombinants. Each\n",
      "            CAGE tag comprises short sequences of about 20bp derived from the\n",
      "            5' end of a full-length cDNA. The length of a CAGE tag may vary due\n",
      "            to the limited specificity of MmeI. Note that up to 70% of the CAGE\n",
      "            tags may have an unspecific G in the first position at the 5' end.\n",
      "            We used in-house developed algorithms for the extraction of tags\n",
      "            and for masking the vectors. CAGE tags were extracted with the\n",
      "            following parameters: vector masking, minimum 12 bp recognition\n",
      "            allowed; linker (13 bp) masking: maximum mismatch, 2 bp allowed;\n",
      "            XmaJI site maximum mismatch, 2 bp allowed; tag length, 17-24 bp.\n",
      "            Linker1: 'Upper oligonucleotide GN6':\n",
      "            biotin-agagagagacctcgagtaactataacggtcctaaggtagcgacctagg (5 bp)\n",
      "            tccgacGNNNNN and 'Upper oligonucleotide N6':\n",
      "            biotin-agagagagacctcgagtaactataacggtcctaaggtagcgacctagg (5 bp)\n",
      "            tccgacNNNNNN were mixed. 'Lower oligonucleotide': phosphate\n",
      "            group-gtcgga (5 bp) cctaggtcgctaccttaggaccgttatagttactcgaggtctctctc\n",
      "            t-NH2\n",
      "            Linker2: 'Upper-XmaJI': Pi-cctaggtcaggactcttctatagtgtcacctaaagacaca\n",
      "            cacac -NH2 'Lower-XmaJI': gtgtgtgtgtctttaggtgacactatagaagagtcctgacc\n",
      "            taggNN\n",
      "            Primer1 (uni-PCR): 5'-biotin-GTGTGTGTGTCTTTAGGTGACACTA-3' Primer2\n",
      "            (MmeI-PCR): 5'-biotin-AGAGAGAGACCTCGAGTAACTATAA-3'\n",
      "            Tissues were provided by Dr. John Reed, whose assistance we\n",
      "            gratefully acknowledge.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..20\n",
      "                     /organism=\"Mus musculus\"\n",
      "                     /mol_type=\"unassigned RNA\"\n",
      "                     /strain=\"C57BL/6Jx129/Sv\"\n",
      "                     /db_xref=\"taxon:10090\"\n",
      "                     /tissue_type=\"body (-head and whole gut)\"\n",
      "                     /clone_lib=\"mouse Cap Analysis Gene Expression (CAGE)\n",
      "                     library\"\n",
      "                     /dev_stage=\"14.5 days embryo\"\n",
      "                     /note=\"primer Oligo dT derived;\n",
      "                     condition: serum-starved cells were irradiated at 20 Gy\n",
      "                     and released into c;\n",
      "                     EMAP_id: EMAP:5719;\n",
      "                     eVOC_id: EVM:2220036\"\n",
      "ORIGIN      \n",
      "        1 ggcgtcctca gctcggggct\n",
      "//\n",
      "\n",
      "LOCUS       AAADK0017277              21 bp    RNA     linear   ROD 20-SEP-2005\n",
      "DEFINITION  Mus musculus strain C57BL/6Jx129/Sv.\n",
      "ACCESSION   AAADK0017277\n",
      "VERSION     AAADK0017277.1\n",
      "KEYWORDS    MGA; 5'-end tag; CAGE (Cap Analysis Gene Expression).\n",
      "SOURCE      Mus musculus (house mouse)\n",
      "  ORGANISM  Mus musculus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Mus; Mus.\n",
      "REFERENCE   1\n",
      "  AUTHORS   Shiraki,T., Kondo,S., Katayama,S., Waki,K., Kasukawa,T., Kawaji,H.,\n",
      "            Kodzius,R., Watahiki,A., Nakamura,M., Arakawa,T., Fukuda,S.,\n",
      "            Sasaki,D., Podhajska,A., Harbers,M., Kawai,J., Carninci,P. and\n",
      "            Hayashizaki,Y.\n",
      "  TITLE     Cap analysis gene expression for high-throughput analysis of\n",
      "            transcriptional starting point and identification of promoter usage\n",
      "  JOURNAL   Proc. Natl. Acad. Sci. U.S.A. 100 (26), 15776-15781 (2003)\n",
      "   PUBMED   14663149\n",
      "REFERENCE   2\n",
      "  AUTHORS   Kodzius,R., Matsumura,Y., Kasukawa,T., Shimokawa,K., Fukuda,S.,\n",
      "            Shiraki,T., Nakamura,M., Arakawa,T., Sasaki,D., Kawai,J.,\n",
      "            Harbers,M., Carninci,P. and Hayashizaki,Y.\n",
      "  TITLE     Absolute expression values for mouse transcripts: re-annotation of\n",
      "            the READ expression database by the use of CAGE and EST sequence\n",
      "            tags\n",
      "  JOURNAL   FEBS Lett. 559 (1-3), 22-26 (2004)\n",
      "   PUBMED   14960301\n",
      "REFERENCE   3\n",
      "  AUTHORS   Carninci,P., Kasukawa,T., Katayama,S., Gough,J., Frith,M.C.,\n",
      "            Maeda,N., Oyama,R., Ravasi,T., Lenhard,B., Wells,C., Kodzius,R.,\n",
      "            Shimokawa,K., Bajic,V.B., Brenner,S.E., Batalov,S., Forrest,A.R.,\n",
      "            Zavolan,M., Davis,M.J., Wilming,L.G., Aidinis,V., Allen,J.E.,\n",
      "            Ambesi-Impiombato,A., Apweiler,R., Aturaliya,R.N., Bailey,T.L.,\n",
      "            Bansal,M., Baxter,L., Beisel,K.W., Bersano,T., Bono,H., Chalk,A.M.,\n",
      "            Chiu,K.P., Choudhary,V., Christoffels,A., Clutterbuck,D.R.,\n",
      "            Crowe,M.L., Dalla,E., Dalrymple,B.P., de Bono,B., Della Gatta,G.,\n",
      "            di Bernardo,D., Down,T., Engstrom,P., Fagiolini,M., Faulkner,G.,\n",
      "            Fletcher,C.F., Fukushima,T., Furuno,M., Futaki,S., Gariboldi,M.,\n",
      "            Georgii-Hemming,P., Gingeras,T.R., Gojobori,T., Green,R.E.,\n",
      "            Gustincich,S., Harbers,M., Hayashi,Y., Hensch,T.K., Hirokawa,N.,\n",
      "            Hill,D., Huminiecki,L., Iacono,M., Ikeo,K., Iwama,A., Ishikawa,T.,\n",
      "            Jakt,M., Kanapin,A., Katoh,M., Kawasawa,Y., Kelso,J., Kitamura,H.,\n",
      "            Kitano,H., Kollias,G., Krishnan,S.P., Kruger,A., Kummerfeld,S.K.,\n",
      "            Kurochkin,I.V., Lareau,L.F., Lazarevic,D., Lipovich,L., Liu,J.,\n",
      "            Liuni,S., McWilliam,S., Madan Babu,M., Madera,M., Marchionni,L.,\n",
      "            Matsuda,H., Matsuzawa,S., Miki,H., Mignone,F., Miyake,S.,\n",
      "            Morris,K., Mottagui-Tabar,S., Mulder,N., Nakano,N., Nakauchi,H.,\n",
      "            Ng,P., Nilsson,R., Nishiguchi,S., Nishikawa,S., Nori,F., Ohara,O.,\n",
      "            Okazaki,Y., Orlando,V., Pang,K.C., Pavan,W.J., Pavesi,G.,\n",
      "            Pesole,G., Petrovsky,N., Piazza,S., Reed,J., Reid,J.F., Ring,B.Z.,\n",
      "            Ringwald,M., Rost,B., Ruan,Y., Salzberg,S.L., Sandelin,A.,\n",
      "            Schneider,C., Schonbach,C., Sekiguchi,K., Semple,C.A., Seno,S.,\n",
      "            Sessa,L., Sheng,Y., Shibata,Y., Shimada,H., Shimada,K., Silva,D.,\n",
      "            Sinclair,B., Sperling,S., Stupka,E., Sugiura,K., Sultana,R.,\n",
      "            Takenaka,Y., Taki,K., Tammoja,K., Tan,S.L., Tang,S., Taylor,M.S.,\n",
      "            Tegner,J., Teichmann,S.A., Ueda,H.R., van Nimwegen,E., Verardo,R.,\n",
      "            Wei,C.L., Yagi,K., Yamanishi,H., Zabarovsky,E., Zhu,S., Zimmer,A.,\n",
      "            Hide,W., Bult,C., Grimmond,S.M., Teasdale,R.D., Liu,E.T.,\n",
      "            Brusic,V., Quackenbush,J., Wahlestedt,C., Mattick,J.S., Hume,D.A.,\n",
      "            Kai,C., Sasaki,D., Tomaru,Y., Fukuda,S., Kanamori-Katayama,M.,\n",
      "            Suzuki,M., Aoki,J., Arakawa,T., Iida,J., Imamura,K., Itoh,M.,\n",
      "            Kato,T., Kawaji,H., Kawagashira,N., Kawashima,T., Kojima,M.,\n",
      "            Kondo,S., Konno,H., Nakano,K., Ninomiya,N., Nishio,T., Okada,M.,\n",
      "            Plessy,C., Shibata,K., Shiraki,T., Suzuki,S., Tagami,M., Waki,K.,\n",
      "            Watahiki,A., Okamura-Oho,Y., Suzuki,H., Kawai,J. and Hayashizaki,Y.\n",
      "  CONSRTM   FANTOM Consortium; RIKEN Genome Exploration Research Group and\n",
      "            Genome Science Group (Genome Network Project Core Group)\n",
      "  TITLE     The transcriptional landscape of the mammalian genome\n",
      "  JOURNAL   Science 309 (5740), 1559-1563 (2005)\n",
      "   PUBMED   16141072\n",
      "  REMARK    Erratum:[Science. 2006 Mar 24;311(5768):1713]\n",
      "REFERENCE   4\n",
      "  AUTHORS   Katayama,S., Tomaru,Y., Kasukawa,T., Waki,K., Nakanishi,M.,\n",
      "            Nakamura,M., Nishida,H., Yap,C.C., Suzuki,M., Kawai,J., Suzuki,H.,\n",
      "            Carninci,P., Hayashizaki,Y., Wells,C., Frith,M., Ravasi,T.,\n",
      "            Pang,K.C., Hallinan,J., Mattick,J., Hume,D.A., Lipovich,L.,\n",
      "            Batalov,S., Engstrom,P.G., Mizuno,Y., Faghihi,M.A., Sandelin,A.,\n",
      "            Chalk,A.M., Mottagui-Tabar,S., Liang,Z., Lenhard,B. and\n",
      "            Wahlestedt,C.\n",
      "  CONSRTM   RIKEN Genome Exploration Research Group; Genome Science Group\n",
      "            (Genome Network Project Core Group); FANTOM Consortium\n",
      "  TITLE     Antisense transcription in the mammalian transcriptome\n",
      "  JOURNAL   Science 309 (5740), 1564-1566 (2005)\n",
      "   PUBMED   16141073\n",
      "REFERENCE   5\n",
      "  AUTHORS   Carninci,P., Sandelin,A., Lenhard,B., Katayama,S., Shimokawa,K.,\n",
      "            Ponjavic,J., Semple,C.A., Taylor,M.S., Engstrom,P.G., Frith,M.C.,\n",
      "            Forrest,A.R., Alkema,W.B., Tan,S.L., Plessy,C., Kodzius,R.,\n",
      "            Ravasi,T., Kasukawa,T., Fukuda,S., Kanamori-Katayama,M.,\n",
      "            Kitazume,Y., Kawaji,H., Kai,C., Nakamura,M., Konno,H., Nakano,K.,\n",
      "            Mottagui-Tabar,S., Arner,P., Chesi,A., Gustincich,S.,\n",
      "            Persichetti,F., Suzuki,H., Grimmond,S.M., Wells,C.A., Orlando,V.,\n",
      "            Wahlestedt,C., Liu,E.T., Harbers,M., Kawai,J., Bajic,V.B.,\n",
      "            Hume,D.A. and Hayashizaki,Y.\n",
      "  TITLE     Genome-wide analysis of mammalian promoter architecture and\n",
      "            evolution\n",
      "  JOURNAL   Nat. Genet. 38 (6), 626-635 (2006)\n",
      "   PUBMED   16645617\n",
      "  REMARK    Erratum:[Nat Genet. 2007 Sep;39(9):1174]\n",
      "REFERENCE   6\n",
      "  AUTHORS   Arakawa,T., Carninci,P., Fukuda,S., Hasegawa,A., Hayashida,K.,\n",
      "            Hori,F., Iida,J., Imamura,K., Kai,C., Kanagawa,S., Katayama,S.,\n",
      "            Kawai,J., Kodzius,R., Kojima,M., Kondo,S., Murata,M., Nakamura,M.,\n",
      "            Nishiyori,H., Nomura,K., Ohno,M., Sakazume,N., Sasaki,D.,\n",
      "            Shiraki,T., Tagami,M., Tagami,Y., Waki,K., Watahiki,A. and\n",
      "            Hayashizaki,Y.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (18-APR-2005) Contact:Yoshihide Hayashizaki The Institute\n",
      "            of Physical and Chemical Research (RIKEN), Omics Science Center,\n",
      "            RIKEN Yokohama Institute; 1-7-22 Suehiro-cho, Tsurumi-ku, Yokohama,\n",
      "            Kanagawa 230-0045, Japan URL :http://www.osc.riken.jp/\n",
      "COMMENT     CAGE library was prepared and sequenced in Genome Science\n",
      "            Laboratory (Wako) and Genome Exploration Research Group Genomics\n",
      "            Science Center (GSC) in RIKEN Yokohama Institute\n",
      "            Please visit our web site for further details.\n",
      "            URL:http://www.osc.riken.jp/\n",
      "            The CAGE (cap analysis gene expression) tags are obtained by\n",
      "            sequencing concatamers of DNA tags deriving from the initial 20/21\n",
      "            nucleotides from 5' end mRNAs, prepared in accordance to Shiraki et\n",
      "            al. (Proc Natl Acad Sci U S A. 100, 15776-81, 2003). At first step,\n",
      "            full-length cDNAs were selected with the Cap-Trapper. Next, a\n",
      "            specific linker (Linker1, which contains the ClassIIs restriction\n",
      "            enzyme site MmeI) was ligated to the cDNA. Linker 1 may contain\n",
      "            extra 5 bp sequences tag, and 15 of such different sequences tags\n",
      "            were used to tag different starting RNA samples. Then the second\n",
      "            strand of cDNA synthesized. Resulting double-stranded cDNAs were\n",
      "            cleaved by the restriction enzyme MmeI and a second linker\n",
      "            (Linker2) was ligated to the 2 bp overhang at the MmeI cleaved\n",
      "            site, to produce a 5' 20/21 tag having two linkers at both sides.\n",
      "            The ligation products were separated from unmodified DNA with\n",
      "            magnetic beads. The 5' end cDNA tags were released from the beads,\n",
      "            and the DNA fragments were amplified in a PCR step by using the two\n",
      "            linker-specific primers (Primer1 (uni-PCR), Primer2 (MmeI-PCR)).\n",
      "            The desired 32-37 bp tags were purified and ligated to form\n",
      "            concatamers, and then the concatamer were fractionated and ligated\n",
      "            to the plasmid ZErO-2. The ligations were finally electroporated\n",
      "            into DH10b cells (Invitrogen) and obtained plasmids were sequenced\n",
      "            with forward primers essentially as described with minor\n",
      "            modifications to use zeocin for selection of recombinants. Each\n",
      "            CAGE tag comprises short sequences of about 20bp derived from the\n",
      "            5' end of a full-length cDNA. The length of a CAGE tag may vary due\n",
      "            to the limited specificity of MmeI. Note that up to 70% of the CAGE\n",
      "            tags may have an unspecific G in the first position at the 5' end.\n",
      "            We used in-house developed algorithms for the extraction of tags\n",
      "            and for masking the vectors. CAGE tags were extracted with the\n",
      "            following parameters: vector masking, minimum 12 bp recognition\n",
      "            allowed; linker (13 bp) masking: maximum mismatch, 2 bp allowed;\n",
      "            XmaJI site maximum mismatch, 2 bp allowed; tag length, 17-24 bp.\n",
      "            Linker1: 'Upper oligonucleotide GN6':\n",
      "            biotin-agagagagacctcgagtaactataacggtcctaaggtagcgacctagg (5 bp)\n",
      "            tccgacGNNNNN and 'Upper oligonucleotide N6':\n",
      "            biotin-agagagagacctcgagtaactataacggtcctaaggtagcgacctagg (5 bp)\n",
      "            tccgacNNNNNN were mixed. 'Lower oligonucleotide': phosphate\n",
      "            group-gtcgga (5 bp) cctaggtcgctaccttaggaccgttatagttactcgaggtctctctc\n",
      "            t-NH2\n",
      "            Linker2: 'Upper-XmaJI': Pi-cctaggtcaggactcttctatagtgtcacctaaagacaca\n",
      "            cacac -NH2 'Lower-XmaJI': gtgtgtgtgtctttaggtgacactatagaagagtcctgacc\n",
      "            taggNN\n",
      "            Primer1 (uni-PCR): 5'-biotin-GTGTGTGTGTCTTTAGGTGACACTA-3' Primer2\n",
      "            (MmeI-PCR): 5'-biotin-AGAGAGAGACCTCGAGTAACTATAA-3'\n",
      "            Tissues were provided by Dr. John Reed, whose assistance we\n",
      "            gratefully acknowledge.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..21\n",
      "                     /organism=\"Mus musculus\"\n",
      "                     /mol_type=\"unassigned RNA\"\n",
      "                     /strain=\"C57BL/6Jx129/Sv\"\n",
      "                     /db_xref=\"taxon:10090\"\n",
      "                     /tissue_type=\"body (-head and whole gut)\"\n",
      "                     /clone_lib=\"mouse Cap Analysis Gene Expression (CAGE)\n",
      "                     library\"\n",
      "                     /dev_stage=\"14.5 days embryo\"\n",
      "                     /note=\"primer Oligo dT derived;\n",
      "                     condition: serum-starved cells were irradiated at 20 Gy\n",
      "                     and released into c;\n",
      "                     EMAP_id: EMAP:5719;\n",
      "                     eVOC_id: EVM:2220036\"\n",
      "ORIGIN      \n",
      "        1 ggcaagtgct tccggagagg t\n",
      "//\n",
      "\n",
      "LOCUS       AAADK0016387              21 bp    RNA     linear   ROD 20-SEP-2005\n",
      "DEFINITION  Mus musculus strain C57BL/6Jx129/Sv.\n",
      "ACCESSION   AAADK0016387\n",
      "VERSION     AAADK0016387.1\n",
      "KEYWORDS    MGA; 5'-end tag; CAGE (Cap Analysis Gene Expression).\n",
      "SOURCE      Mus musculus (house mouse)\n",
      "  ORGANISM  Mus musculus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Mus; Mus.\n",
      "REFERENCE   1\n",
      "  AUTHORS   Shiraki,T., Kondo,S., Katayama,S., Waki,K., Kasukawa,T., Kawaji,H.,\n",
      "            Kodzius,R., Watahiki,A., Nakamura,M., Arakawa,T., Fukuda,S.,\n",
      "            Sasaki,D., Podhajska,A., Harbers,M., Kawai,J., Carninci,P. and\n",
      "            Hayashizaki,Y.\n",
      "  TITLE     Cap analysis gene expression for high-throughput analysis of\n",
      "            transcriptional starting point and identification of promoter usage\n",
      "  JOURNAL   Proc. Natl. Acad. Sci. U.S.A. 100 (26), 15776-15781 (2003)\n",
      "   PUBMED   14663149\n",
      "REFERENCE   2\n",
      "  AUTHORS   Kodzius,R., Matsumura,Y., Kasukawa,T., Shimokawa,K., Fukuda,S.,\n",
      "            Shiraki,T., Nakamura,M., Arakawa,T., Sasaki,D., Kawai,J.,\n",
      "            Harbers,M., Carninci,P. and Hayashizaki,Y.\n",
      "  TITLE     Absolute expression values for mouse transcripts: re-annotation of\n",
      "            the READ expression database by the use of CAGE and EST sequence\n",
      "            tags\n",
      "  JOURNAL   FEBS Lett. 559 (1-3), 22-26 (2004)\n",
      "   PUBMED   14960301\n",
      "REFERENCE   3\n",
      "  AUTHORS   Carninci,P., Kasukawa,T., Katayama,S., Gough,J., Frith,M.C.,\n",
      "            Maeda,N., Oyama,R., Ravasi,T., Lenhard,B., Wells,C., Kodzius,R.,\n",
      "            Shimokawa,K., Bajic,V.B., Brenner,S.E., Batalov,S., Forrest,A.R.,\n",
      "            Zavolan,M., Davis,M.J., Wilming,L.G., Aidinis,V., Allen,J.E.,\n",
      "            Ambesi-Impiombato,A., Apweiler,R., Aturaliya,R.N., Bailey,T.L.,\n",
      "            Bansal,M., Baxter,L., Beisel,K.W., Bersano,T., Bono,H., Chalk,A.M.,\n",
      "            Chiu,K.P., Choudhary,V., Christoffels,A., Clutterbuck,D.R.,\n",
      "            Crowe,M.L., Dalla,E., Dalrymple,B.P., de Bono,B., Della Gatta,G.,\n",
      "            di Bernardo,D., Down,T., Engstrom,P., Fagiolini,M., Faulkner,G.,\n",
      "            Fletcher,C.F., Fukushima,T., Furuno,M., Futaki,S., Gariboldi,M.,\n",
      "            Georgii-Hemming,P., Gingeras,T.R., Gojobori,T., Green,R.E.,\n",
      "            Gustincich,S., Harbers,M., Hayashi,Y., Hensch,T.K., Hirokawa,N.,\n",
      "            Hill,D., Huminiecki,L., Iacono,M., Ikeo,K., Iwama,A., Ishikawa,T.,\n",
      "            Jakt,M., Kanapin,A., Katoh,M., Kawasawa,Y., Kelso,J., Kitamura,H.,\n",
      "            Kitano,H., Kollias,G., Krishnan,S.P., Kruger,A., Kummerfeld,S.K.,\n",
      "            Kurochkin,I.V., Lareau,L.F., Lazarevic,D., Lipovich,L., Liu,J.,\n",
      "            Liuni,S., McWilliam,S., Madan Babu,M., Madera,M., Marchionni,L.,\n",
      "            Matsuda,H., Matsuzawa,S., Miki,H., Mignone,F., Miyake,S.,\n",
      "            Morris,K., Mottagui-Tabar,S., Mulder,N., Nakano,N., Nakauchi,H.,\n",
      "            Ng,P., Nilsson,R., Nishiguchi,S., Nishikawa,S., Nori,F., Ohara,O.,\n",
      "            Okazaki,Y., Orlando,V., Pang,K.C., Pavan,W.J., Pavesi,G.,\n",
      "            Pesole,G., Petrovsky,N., Piazza,S., Reed,J., Reid,J.F., Ring,B.Z.,\n",
      "            Ringwald,M., Rost,B., Ruan,Y., Salzberg,S.L., Sandelin,A.,\n",
      "            Schneider,C., Schonbach,C., Sekiguchi,K., Semple,C.A., Seno,S.,\n",
      "            Sessa,L., Sheng,Y., Shibata,Y., Shimada,H., Shimada,K., Silva,D.,\n",
      "            Sinclair,B., Sperling,S., Stupka,E., Sugiura,K., Sultana,R.,\n",
      "            Takenaka,Y., Taki,K., Tammoja,K., Tan,S.L., Tang,S., Taylor,M.S.,\n",
      "            Tegner,J., Teichmann,S.A., Ueda,H.R., van Nimwegen,E., Verardo,R.,\n",
      "            Wei,C.L., Yagi,K., Yamanishi,H., Zabarovsky,E., Zhu,S., Zimmer,A.,\n",
      "            Hide,W., Bult,C., Grimmond,S.M., Teasdale,R.D., Liu,E.T.,\n",
      "            Brusic,V., Quackenbush,J., Wahlestedt,C., Mattick,J.S., Hume,D.A.,\n",
      "            Kai,C., Sasaki,D., Tomaru,Y., Fukuda,S., Kanamori-Katayama,M.,\n",
      "            Suzuki,M., Aoki,J., Arakawa,T., Iida,J., Imamura,K., Itoh,M.,\n",
      "            Kato,T., Kawaji,H., Kawagashira,N., Kawashima,T., Kojima,M.,\n",
      "            Kondo,S., Konno,H., Nakano,K., Ninomiya,N., Nishio,T., Okada,M.,\n",
      "            Plessy,C., Shibata,K., Shiraki,T., Suzuki,S., Tagami,M., Waki,K.,\n",
      "            Watahiki,A., Okamura-Oho,Y., Suzuki,H., Kawai,J. and Hayashizaki,Y.\n",
      "  CONSRTM   FANTOM Consortium; RIKEN Genome Exploration Research Group and\n",
      "            Genome Science Group (Genome Network Project Core Group)\n",
      "  TITLE     The transcriptional landscape of the mammalian genome\n",
      "  JOURNAL   Science 309 (5740), 1559-1563 (2005)\n",
      "   PUBMED   16141072\n",
      "  REMARK    Erratum:[Science. 2006 Mar 24;311(5768):1713]\n",
      "REFERENCE   4\n",
      "  AUTHORS   Katayama,S., Tomaru,Y., Kasukawa,T., Waki,K., Nakanishi,M.,\n",
      "            Nakamura,M., Nishida,H., Yap,C.C., Suzuki,M., Kawai,J., Suzuki,H.,\n",
      "            Carninci,P., Hayashizaki,Y., Wells,C., Frith,M., Ravasi,T.,\n",
      "            Pang,K.C., Hallinan,J., Mattick,J., Hume,D.A., Lipovich,L.,\n",
      "            Batalov,S., Engstrom,P.G., Mizuno,Y., Faghihi,M.A., Sandelin,A.,\n",
      "            Chalk,A.M., Mottagui-Tabar,S., Liang,Z., Lenhard,B. and\n",
      "            Wahlestedt,C.\n",
      "  CONSRTM   RIKEN Genome Exploration Research Group; Genome Science Group\n",
      "            (Genome Network Project Core Group); FANTOM Consortium\n",
      "  TITLE     Antisense transcription in the mammalian transcriptome\n",
      "  JOURNAL   Science 309 (5740), 1564-1566 (2005)\n",
      "   PUBMED   16141073\n",
      "REFERENCE   5\n",
      "  AUTHORS   Carninci,P., Sandelin,A., Lenhard,B., Katayama,S., Shimokawa,K.,\n",
      "            Ponjavic,J., Semple,C.A., Taylor,M.S., Engstrom,P.G., Frith,M.C.,\n",
      "            Forrest,A.R., Alkema,W.B., Tan,S.L., Plessy,C., Kodzius,R.,\n",
      "            Ravasi,T., Kasukawa,T., Fukuda,S., Kanamori-Katayama,M.,\n",
      "            Kitazume,Y., Kawaji,H., Kai,C., Nakamura,M., Konno,H., Nakano,K.,\n",
      "            Mottagui-Tabar,S., Arner,P., Chesi,A., Gustincich,S.,\n",
      "            Persichetti,F., Suzuki,H., Grimmond,S.M., Wells,C.A., Orlando,V.,\n",
      "            Wahlestedt,C., Liu,E.T., Harbers,M., Kawai,J., Bajic,V.B.,\n",
      "            Hume,D.A. and Hayashizaki,Y.\n",
      "  TITLE     Genome-wide analysis of mammalian promoter architecture and\n",
      "            evolution\n",
      "  JOURNAL   Nat. Genet. 38 (6), 626-635 (2006)\n",
      "   PUBMED   16645617\n",
      "  REMARK    Erratum:[Nat Genet. 2007 Sep;39(9):1174]\n",
      "REFERENCE   6\n",
      "  AUTHORS   Arakawa,T., Carninci,P., Fukuda,S., Hasegawa,A., Hayashida,K.,\n",
      "            Hori,F., Iida,J., Imamura,K., Kai,C., Kanagawa,S., Katayama,S.,\n",
      "            Kawai,J., Kodzius,R., Kojima,M., Kondo,S., Murata,M., Nakamura,M.,\n",
      "            Nishiyori,H., Nomura,K., Ohno,M., Sakazume,N., Sasaki,D.,\n",
      "            Shiraki,T., Tagami,M., Tagami,Y., Waki,K., Watahiki,A. and\n",
      "            Hayashizaki,Y.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (18-APR-2005) Contact:Yoshihide Hayashizaki The Institute\n",
      "            of Physical and Chemical Research (RIKEN), Omics Science Center,\n",
      "            RIKEN Yokohama Institute; 1-7-22 Suehiro-cho, Tsurumi-ku, Yokohama,\n",
      "            Kanagawa 230-0045, Japan URL :http://www.osc.riken.jp/\n",
      "COMMENT     CAGE library was prepared and sequenced in Genome Science\n",
      "            Laboratory (Wako) and Genome Exploration Research Group Genomics\n",
      "            Science Center (GSC) in RIKEN Yokohama Institute\n",
      "            Please visit our web site for further details.\n",
      "            URL:http://www.osc.riken.jp/\n",
      "            The CAGE (cap analysis gene expression) tags are obtained by\n",
      "            sequencing concatamers of DNA tags deriving from the initial 20/21\n",
      "            nucleotides from 5' end mRNAs, prepared in accordance to Shiraki et\n",
      "            al. (Proc Natl Acad Sci U S A. 100, 15776-81, 2003). At first step,\n",
      "            full-length cDNAs were selected with the Cap-Trapper. Next, a\n",
      "            specific linker (Linker1, which contains the ClassIIs restriction\n",
      "            enzyme site MmeI) was ligated to the cDNA. Linker 1 may contain\n",
      "            extra 5 bp sequences tag, and 15 of such different sequences tags\n",
      "            were used to tag different starting RNA samples. Then the second\n",
      "            strand of cDNA synthesized. Resulting double-stranded cDNAs were\n",
      "            cleaved by the restriction enzyme MmeI and a second linker\n",
      "            (Linker2) was ligated to the 2 bp overhang at the MmeI cleaved\n",
      "            site, to produce a 5' 20/21 tag having two linkers at both sides.\n",
      "            The ligation products were separated from unmodified DNA with\n",
      "            magnetic beads. The 5' end cDNA tags were released from the beads,\n",
      "            and the DNA fragments were amplified in a PCR step by using the two\n",
      "            linker-specific primers (Primer1 (uni-PCR), Primer2 (MmeI-PCR)).\n",
      "            The desired 32-37 bp tags were purified and ligated to form\n",
      "            concatamers, and then the concatamer were fractionated and ligated\n",
      "            to the plasmid ZErO-2. The ligations were finally electroporated\n",
      "            into DH10b cells (Invitrogen) and obtained plasmids were sequenced\n",
      "            with forward primers essentially as described with minor\n",
      "            modifications to use zeocin for selection of recombinants. Each\n",
      "            CAGE tag comprises short sequences of about 20bp derived from the\n",
      "            5' end of a full-length cDNA. The length of a CAGE tag may vary due\n",
      "            to the limited specificity of MmeI. Note that up to 70% of the CAGE\n",
      "            tags may have an unspecific G in the first position at the 5' end.\n",
      "            We used in-house developed algorithms for the extraction of tags\n",
      "            and for masking the vectors. CAGE tags were extracted with the\n",
      "            following parameters: vector masking, minimum 12 bp recognition\n",
      "            allowed; linker (13 bp) masking: maximum mismatch, 2 bp allowed;\n",
      "            XmaJI site maximum mismatch, 2 bp allowed; tag length, 17-24 bp.\n",
      "            Linker1: 'Upper oligonucleotide GN6':\n",
      "            biotin-agagagagacctcgagtaactataacggtcctaaggtagcgacctagg (5 bp)\n",
      "            tccgacGNNNNN and 'Upper oligonucleotide N6':\n",
      "            biotin-agagagagacctcgagtaactataacggtcctaaggtagcgacctagg (5 bp)\n",
      "            tccgacNNNNNN were mixed. 'Lower oligonucleotide': phosphate\n",
      "            group-gtcgga (5 bp) cctaggtcgctaccttaggaccgttatagttactcgaggtctctctc\n",
      "            t-NH2\n",
      "            Linker2: 'Upper-XmaJI': Pi-cctaggtcaggactcttctatagtgtcacctaaagacaca\n",
      "            cacac -NH2 'Lower-XmaJI': gtgtgtgtgtctttaggtgacactatagaagagtcctgacc\n",
      "            taggNN\n",
      "            Primer1 (uni-PCR): 5'-biotin-GTGTGTGTGTCTTTAGGTGACACTA-3' Primer2\n",
      "            (MmeI-PCR): 5'-biotin-AGAGAGAGACCTCGAGTAACTATAA-3'\n",
      "            Tissues were provided by Dr. John Reed, whose assistance we\n",
      "            gratefully acknowledge.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..21\n",
      "                     /organism=\"Mus musculus\"\n",
      "                     /mol_type=\"unassigned RNA\"\n",
      "                     /strain=\"C57BL/6Jx129/Sv\"\n",
      "                     /db_xref=\"taxon:10090\"\n",
      "                     /tissue_type=\"body (-head and whole gut)\"\n",
      "                     /clone_lib=\"mouse Cap Analysis Gene Expression (CAGE)\n",
      "                     library\"\n",
      "                     /dev_stage=\"14.5 days embryo\"\n",
      "                     /note=\"primer Oligo dT derived;\n",
      "                     condition: serum-starved cells were irradiated at 20 Gy\n",
      "                     and released into c;\n",
      "                     EMAP_id: EMAP:5719;\n",
      "                     eVOC_id: EVM:2220036\"\n",
      "ORIGIN      \n",
      "        1 ggcatacagc atccacttcc t\n",
      "//\n",
      "\n",
      "LOCUS       HIV1LH22M                184 bp    DNA     linear   VRL 31-JUL-1995\n",
      "DEFINITION  Human immunodeficiency virus type 1 proviral envelope glycoprotein\n",
      "            (env) gene V4/CD4-binding region, partial sequence (LH20-2).\n",
      "ACCESSION   L33068\n",
      "VERSION     L33068.1\n",
      "KEYWORDS    env gene; envelope glycoprotein.\n",
      "SOURCE      Human immunodeficiency virus 1 (HIV-1)\n",
      "  ORGANISM  Human immunodeficiency virus 1\n",
      "            Viruses; Riboviria; Pararnavirae; Artverviricota; Revtraviricetes;\n",
      "            Ortervirales; Retroviridae; Orthoretrovirinae; Lentivirus.\n",
      "REFERENCE   1  (bases 1 to 184)\n",
      "  AUTHORS   Bryson,Y.J., Pang,S., Wei,L.S., Dickover,R., Diagne,A. and\n",
      "            Chen,I.S.\n",
      "  TITLE     Clearance of HIV infection in a perinatally infected infant\n",
      "  JOURNAL   N. Engl. J. Med. 332 (13), 833-838 (1995)\n",
      "   PUBMED   7870139\n",
      "COMMENT     Original source text: Human immunodeficiency virus type 1 blood\n",
      "            DNA.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..184\n",
      "                     /organism=\"Human immunodeficiency virus 1\"\n",
      "                     /proviral\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /db_xref=\"taxon:11676\"\n",
      "                     /tissue_type=\"blood\"\n",
      "     gene            1..184\n",
      "                     /gene=\"env\"\n",
      "     CDS             <1..>184\n",
      "                     /gene=\"env\"\n",
      "                     /function=\"involved in binding CD4 receptor\"\n",
      "                     /standard_name=\"env V4 sequence\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"envelope glycoprotein\"\n",
      "                     /protein_id=\"AAA73551.1\"\n",
      "                     /translation=\"CNTSRLFNSTWNGTEGLPDITLPCRIKQIINMWQEVGKAMYAPP\n",
      "                     IKGQIRCSSNITGLLLT\"\n",
      "ORIGIN      \n",
      "        1 tgtaatacat cacgactatt taatagtact tggaatggta ctgaagggtt acctgacatc\n",
      "       61 acactcccat gcagaataaa acaaatcata aacatgtggc aggaagtagg aaaagcaatg\n",
      "      121 tatgcccctc ccatcaaagg acaaattaga tgttcatcaa atattacagg actgctatta\n",
      "      181 acaa\n",
      "//\n",
      "\n",
      "LOCUS       HIV1LH21M                193 bp    DNA     linear   VRL 31-JUL-1995\n",
      "DEFINITION  Human immunodeficiency virus type 1 proviral envelope glycoprotein\n",
      "            (env) gene V4/CD4-binding region, partial sequence (LH20-1).\n",
      "ACCESSION   L33067\n",
      "VERSION     L33067.1\n",
      "KEYWORDS    env gene; envelope glycoprotein.\n",
      "SOURCE      Human immunodeficiency virus 1 (HIV-1)\n",
      "  ORGANISM  Human immunodeficiency virus 1\n",
      "            Viruses; Riboviria; Pararnavirae; Artverviricota; Revtraviricetes;\n",
      "            Ortervirales; Retroviridae; Orthoretrovirinae; Lentivirus.\n",
      "REFERENCE   1  (bases 1 to 193)\n",
      "  AUTHORS   Bryson,Y.J., Pang,S., Wei,L.S., Dickover,R., Diagne,A. and\n",
      "            Chen,I.S.\n",
      "  TITLE     Clearance of HIV infection in a perinatally infected infant\n",
      "  JOURNAL   N. Engl. J. Med. 332 (13), 833-838 (1995)\n",
      "   PUBMED   7870139\n",
      "COMMENT     Original source text: Human immunodeficiency virus type 1 blood\n",
      "            DNA.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..193\n",
      "                     /organism=\"Human immunodeficiency virus 1\"\n",
      "                     /proviral\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /db_xref=\"taxon:11676\"\n",
      "                     /tissue_type=\"blood\"\n",
      "     gene            1..193\n",
      "                     /gene=\"env\"\n",
      "     CDS             <1..>193\n",
      "                     /gene=\"env\"\n",
      "                     /function=\"involved in binding CD4 receptor\"\n",
      "                     /standard_name=\"env V4 sequence\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"envelope glycoprotein\"\n",
      "                     /protein_id=\"AAA73550.1\"\n",
      "                     /translation=\"CNTSQLFNSTWNGTEGFPNTENITLPCRIKQIINLWQEVGKAMY\n",
      "                     APPIRGQIRCSSNITGLLLT\"\n",
      "ORIGIN      \n",
      "        1 tgtaatacat cacaactgtt taatagtact tggaatggta ctgaagggtt tcctaacact\n",
      "       61 gaaaatatca cactcccatg cagaataaaa caaatcataa acctgtggca ggaagtagga\n",
      "      121 aaagcaatgt atgcccctcc catcagagga caaattagat gttcatcaaa tattacagga\n",
      "      181 ctgctattaa caa\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# Making a GET request\n",
    "#r = requests.get('https://www.ncbi.nlm.nih.gov/nuccore/{}'.format(genes))\n",
    "# Parsing the HTML\n",
    "locus = \"\"\n",
    "locus_list = []\n",
    "for c in id_content:\n",
    "\tsoup = BeautifulSoup(c, 'html.parser')\n",
    "\n",
    "# Procurar um tag meta com um determinado atributo\n",
    "\n",
    "\tlines = soup.find_all('meta', {'name':\"ncbi_uidlist\"} )\n",
    "\n",
    "\tid = \"\"\n",
    "\turl = \"\"\n",
    "\tfor line in lines:\n",
    "\t\t# print(line)\n",
    "\t\t# if 'name' in line.attrs:\n",
    "\t\t# \tprint(line.attrs['name'])\n",
    "\t\tif 'content' in line.attrs:\n",
    "\t\t\t# print(line.attrs['content'])\t\t\n",
    "\t\t\tid = line.attrs['content']\n",
    "\n",
    "\tif id:\n",
    "\t\turl =\"https://www.ncbi.nlm.nih.gov/sviewer/viewer.fcgi?id={}&db=nuccore&report=genbank&conwithfeat=on&hide-cdd=on&retmode=text&ncbi_phid=CE88F25338A215A1000000000483042A&withmarkup=on&tool=portal&log$=seqview&maxdownloadsize=1000000\".format(id)\n",
    "\n",
    "\tr2 = requests.get(url)\n",
    "\tlocus += r2.content.decode('utf-8')\n",
    "\n",
    "\tif \"Error\" not in r2.content.decode('utf-8'):\n",
    "\t\tlocus_list.append(r2.content.decode('utf-8'))\n",
    "print(locus)\n",
    "\t#utf-8 para converter de bytes para string\n",
    "\t#https://stackoverflow.com/questions/31019854/typeerror-cant-use-a-string-pattern-on-a-bytes-like-object-in-re-findall"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "LINK NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['https://www.ncbi.nlm.nih.gov/nuccore/X56237',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/AAADK0077081',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/AAADK0017277',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/AAADK0016387',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/L33068',\n",
       " 'https://www.ncbi.nlm.nih.gov/nuccore/L33067']"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "id_lista = []\n",
    "# print(genes)\n",
    "existe = re.findall(r'ACCESSION\\s+[^\\s]+', locus)\n",
    "#\\s+ um ou mais espaços\n",
    "#[^\\s]+ indica um conjunto de caracteres que se podem repetir uma ou mais vezes\n",
    "\n",
    "if existe:\n",
    "    for c in existe:\n",
    "        m = re.match(r'ACCESSION\\s+([^\\s]+)', c )\n",
    "        if m:\n",
    "            id = m.group(1)\n",
    "            id_lista.append(\"https://www.ncbi.nlm.nih.gov/nuccore/{}\".format( id ) )\n",
    "else:\n",
    "    print( \"Padrão não encontrado\" )\n",
    "id_lista\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ACCESSION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X56237 ',\n",
       " 'AAADK0077081 ',\n",
       " 'AAADK0017277 ',\n",
       " 'AAADK0016387 ',\n",
       " 'L33068 ',\n",
       " 'L33067 ']"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_list = []\n",
    "\n",
    "existe = re.findall(r'ACCESSION\\s+.*?(?=VERSION)', locus, re.DOTALL)\n",
    "#\\s+ um ou mais espaços\n",
    "#\\.* encontra o caractere ponto final zero ou mais vezes\n",
    "#?= something que faz match com VERSION mas não consome estes carateres\n",
    "if existe:\n",
    "    for accession in existe:\n",
    "        m = re.match( r'ACCESSION\\s+(.+)', accession, re.DOTALL )\n",
    "        acc_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "#(.+) agrupar um ou mais caracteres dentro de um grupo\n",
    "#re.sub(r'\\s+', ' ' ...) substitui um ou mais espaços por apenas um espaço\n",
    "acc_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "VERSION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X56237.1 ',\n",
       " 'AAADK0077081.1 ',\n",
       " 'AAADK0017277.1 ',\n",
       " 'AAADK0016387.1 ',\n",
       " 'L33068.1 ',\n",
       " 'L33067.1 ']"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "version_list = []\n",
    "\n",
    "existe = re.findall(r'VERSION.*?(?=KEYWORDS|DBLINK)', locus, re.DOTALL)\n",
    "#\\s+ um ou mais espaços\n",
    "#\\.* encontra o caractere ponto final zero ou mais vezes\n",
    "#?= something que faz match com VERSION mas não consome estes carateres\n",
    "if existe:\n",
    "    for vers in existe:\n",
    "        m = re.match( r'VERSION\\s+(.+)', vers, re.DOTALL )\n",
    "        version_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "#(.+) agrupar um ou mais caracteres dentro de um grupo\n",
    "#re.sub(r'\\s+', ' ' ...) substitui um ou mais espaços por apenas um espaço\n",
    "version_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "DEFINITION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['N. crassa NUO-32 mRNA for NADH dehydrogenase 32 kD subunit. ',\n",
       " 'Mus musculus strain C57BL/6Jx129/Sv. ',\n",
       " 'Mus musculus strain C57BL/6Jx129/Sv. ',\n",
       " 'Mus musculus strain C57BL/6Jx129/Sv. ',\n",
       " 'Human immunodeficiency virus type 1 proviral envelope glycoprotein (env) gene V4/CD4-binding region, partial sequence (LH20-2). ',\n",
       " 'Human immunodeficiency virus type 1 proviral envelope glycoprotein (env) gene V4/CD4-binding region, partial sequence (LH20-1). ']"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "defi_list = []\n",
    "existe = re.findall(r'DEFINITION\\s+.*?(?=ACCESSION)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for definition in existe:\n",
    "        m = re.match( r'DEFINITION\\s+(.+)', definition, re.DOTALL )\n",
    "        defi_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "defi_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "LOCUS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X56237 1077 bp mRNA linear PLN 12-JUN-2006 ',\n",
       " 'AAADK0077081 20 bp RNA linear ROD 20-SEP-2005 ',\n",
       " 'AAADK0017277 21 bp RNA linear ROD 20-SEP-2005 ',\n",
       " 'AAADK0016387 21 bp RNA linear ROD 20-SEP-2005 ',\n",
       " 'HIV1LH22M 184 bp DNA linear VRL 31-JUL-1995 ',\n",
       " 'HIV1LH21M 193 bp DNA linear VRL 31-JUL-1995 ']"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loc_list = []\n",
    "existe = re.findall(r'LOCUS\\s+.*?(?=DEFINITION)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for loc in existe:\n",
    "        m = re.match( r'LOCUS\\s+(.+)', loc, re.DOTALL )\n",
    "        loc_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "loc_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "LOCUS NAME"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X56237',\n",
       " 'AAADK0077081',\n",
       " 'AAADK0017277',\n",
       " 'AAADK0016387',\n",
       " 'HIV1LH22M',\n",
       " 'HIV1LH21M']"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loc_name_list = []\n",
    "existe = re.findall(r'LOCUS\\s + [^\\s]+', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for id in existe:\n",
    "        m = re.match( r'LOCUS\\s+(.+)', id, re.DOTALL )\n",
    "        loc_name_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "loc_name_list\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "NUMBER OF BP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1077 bp', '20 bp', '21 bp', '21 bp', '184 bp', '193 bp']"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loc_bp_list = []\n",
    "loc = []\n",
    "\n",
    "existe = re.findall(r'LOCUS\\s+.*?(?=DEFINITION)', locus, re.DOTALL)\n",
    "# print(existe)\n",
    "if existe:\n",
    "    for c in existe:\n",
    "        loc.append(list(filter(None, c.split(' '))))\n",
    "for x in range(len(loc)):\n",
    "    loc_bp_list.append(' '.join(loc[x][2:4])) \n",
    "\n",
    "\n",
    "loc_bp_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MOLECULE TYPE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['mRNA linear',\n",
       " 'RNA linear',\n",
       " 'RNA linear',\n",
       " 'RNA linear',\n",
       " 'DNA linear',\n",
       " 'DNA linear']"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mol_type = []\n",
    "for m in range(len(loc)):\n",
    "    if \"linear\" in loc[m][5]:\n",
    "        mol_type.append(' '.join(loc[m][4:6]))\n",
    "    else:\n",
    "        mol_type.append(' '.join(loc[m][4:5]))\n",
    "\n",
    "mol_type"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GENBANK DIVISION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PLN', 'ROD', 'ROD', 'ROD', 'VRL', 'VRL']"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gb_div = []\n",
    "for gb in range(len(loc)):\n",
    "    gb_div.append(' '.join(loc[gb][6:7]))\n",
    "gb_div"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MODIFICATION DATE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['12-JUN-2006',\n",
       " '20-SEP-2005',\n",
       " '20-SEP-2005',\n",
       " '20-SEP-2005',\n",
       " '31-JUL-1995',\n",
       " '31-JUL-1995']"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mod_date = []\n",
    "for md in range(len(loc)):\n",
    "    mod_date.append(''.join(loc[md][7]).replace(\"\\n\",\"\"))\n",
    "mod_date"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SOURCE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Neurospora crassa ',\n",
       " 'Mus musculus (house mouse) ',\n",
       " 'Mus musculus (house mouse) ',\n",
       " 'Mus musculus (house mouse) ',\n",
       " 'Human immunodeficiency virus 1 (HIV-1) ',\n",
       " 'Human immunodeficiency virus 1 (HIV-1) ']"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "source_list = []\n",
    "existe = re.findall(r'SOURCE\\s+.*?(?=ORGANISM)', locus, re.DOTALL)\n",
    "# print(existe)\n",
    "if existe:\n",
    "    for source in existe:\n",
    "        m = re.match( r'SOURCE\\s+(.+)', source, re.DOTALL )\n",
    "        source_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "source_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ORGANISM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Neurospora crassa',\n",
       " 'Mus musculus',\n",
       " 'Mus musculus',\n",
       " 'Mus musculus',\n",
       " 'Human immunodeficiency virus 1',\n",
       " 'Human immunodeficiency virus 1']"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_list = []\n",
    "existe = re.findall(r'ORGANISM\\s+.*?(?=\\n)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for org in existe:\n",
    "        m = re.match( r'ORGANISM\\s+(.+)', org, re.DOTALL )\n",
    "        org_list.append(re.sub(r'\\s+', ' ', m.group(1) ) )\n",
    "org_list"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "FEATURES - CDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['source', 'gene', 'mRNA', 'CDS', 'sig_peptide', 'mat_peptide', 'polyA_site', 'source', 'source', 'source', 'source', 'gene', 'CDS', 'source', 'gene', 'CDS']\n",
      "* [['[0:1077](+)'], ['[0:20](+)'], ['[0:21](+)'], ['[0:21](+)'], ['[0:184](+)'], ['[0:193](+)']]\n",
      "-> [[['NUO-32']], [], [], [], [['env']], [['env']]]\n",
      "6\n",
      "p [[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []], [[]], [[]], [[]], [[], [], ['[<0:>184](+)', ['envelope glycoprotein'], ['AAA73551.1'], ['CNTSRLFNSTWNGTEGLPDITLPCRIKQIINMWQEVGKAMYAPPIKGQIRCSSNITGLLLT']]], [[], [], ['[<0:>193](+)', ['envelope glycoprotein'], ['AAA73550.1'], ['CNTSQLFNSTWNGTEGFPNTENITLPCRIKQIINLWQEVGKAMYAPPIRGQIRCSSNITGLLLT']]]]\n",
      "-> [[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []], [[]], [[]], [[]], [[], [], ['[<0:>184](+)', ['envelope glycoprotein'], ['AAA73551.1'], ['CNTSRLFNSTWNGTEGLPDITLPCRIKQIINMWQEVGKAMYAPPIKGQIRCSSNITGLLLT']]], [[], [], ['[<0:>193](+)', ['envelope glycoprotein'], ['AAA73550.1'], ['CNTSQLFNSTWNGTEGFPNTENITLPCRIKQIINLWQEVGKAMYAPPIRGQIRCSSNITGLLLT']]]]\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "from Bio import SeqFeature\n",
    "from Bio.SeqFeature import SeqFeature, FeatureLocation\n",
    "\n",
    "Entrez.email = \"pg49849@alunos.uminho.pt\"\n",
    "\n",
    "features=[]\n",
    "CDS=[] \n",
    "source = []\n",
    "gene = []\n",
    "\n",
    "\n",
    "for acc in acc_list:\n",
    "    handle = Entrez.efetch(db=\"nucleotide\", rettype = \"gb\", retmode= \"text\", id=acc)\n",
    "    \n",
    "    for record in SeqIO.parse(handle,\"gb\"):\n",
    "        \n",
    "        for feature in record.features:\n",
    "            features.append(feature.type)\n",
    "\n",
    "        aux_source = []\n",
    "        aux_gene = []\n",
    "\n",
    "\n",
    "        for feature in record.features:\n",
    "            if feature.type == \"source\":\n",
    "                aux_source.append(str(feature.location))\n",
    "\n",
    "            if feature.type == 'gene':\n",
    "                aux_gene.append(feature.qualifiers[\"gene\"])\n",
    " \n",
    "        gene.append(aux_gene)\n",
    "        source.append(aux_source)\n",
    "\n",
    "        record_CDS = []\n",
    "        for feature in record.features:\n",
    "            CDS_aux=[]\n",
    "            if feature.type == \"CDS\":\n",
    "                CDS_aux.append(str(feature.location))\n",
    "                CDS_aux.append(feature.qualifiers[\"product\"])\n",
    "                CDS_aux.append(feature.qualifiers[\"protein_id\"])\n",
    "                CDS_aux.append(feature.qualifiers[\"translation\"])\n",
    "                # print('->' , CDS_aux)\n",
    "            record_CDS.append(CDS_aux)\n",
    "            \n",
    "        CDS.append(record_CDS)\n",
    "\n",
    "# print(source)\n",
    "# print(gene)\n",
    "# print(exon)\n",
    "# print(intron)\n",
    "print(features)\n",
    "print('*',source)\n",
    "print('->',gene)\n",
    "print(len(CDS))\n",
    "print('p', CDS)  \n",
    "print('->', CDS) "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "SEQUENCE/ORIGIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ccacaatccaccccaccccaccggaccccgagtcagtcaacaacctccacgacacgactcggaaaccatgcgcgcggcactccgattgctcgcgaccgcgacggctaccgtccgcccgtccgcccgcttcctcaaacccggctccccgaccggcctcaccggcctcggcacccacccgtcgccgcgctccgccctgctgtacctctacaaccacaccctcgacaagctcaagcagatccccgagcactcgctgtaccgccagtccgccgaggccctgaccaagcaccgcctcgccatcgtcgagcagtacgtgcccgacggctacgacgcctggcaggagcgcgcccgcaagctgctcgagaagcacaagagcgacctcacggcccgccagttcgacggccagcatgcccgcctagtcgaaggccccgacggccgcgcttacttcatccgccagatggtccccccgcaggactggcgcgacgtcgagtgggatggtgccgtcctggatccgcacttttcctgggttcagaccggcgaggatgtcgtcggcgccgtcaagctggaagacagcgacaagctgctcgagctggacaagatcagggaatcagacccggttgcctatcgccagggtctcagggatctcggtatcaagatgggcggtgttgtcgaggacaagagccccgtcgagtgggagtcggagccgccgttgagcgctgaacagattgctgagatggaggccaggattggttctggcttgattgaggaggttgttcaggttgccgagggcgagcttaagctggttgacattatgacccaggccaggccttgggaagctcttgaggaggaggcgccagagggccaatggacttactttgagcgtaaggagtaaacaaacatcgacagacatagagaatgggaacacaggaccaaccaggccatgacggccacctattttatgagcgttacgaagaagaagggaggacaggcatttctcagccagttgtacctataccataccacgccgggacatgtatataatcatcgactttggcaggcgggaatgactatgccctcctt',\n",
       " 'ggcgtcctcagctcggggct',\n",
       " 'ggcaagtgcttccggagaggt',\n",
       " 'ggcatacagcatccacttcct',\n",
       " 'tgtaatacatcacgactatttaatagtacttggaatggtactgaagggttacctgacatcacactcccatgcagaataaaacaaatcataaacatgtggcaggaagtaggaaaagcaatgtatgcccctcccatcaaaggacaaattagatgttcatcaaatattacaggactgctattaacaa',\n",
       " 'tgtaatacatcacaactgtttaatagtacttggaatggtactgaagggtttcctaacactgaaaatatcacactcccatgcagaataaaacaaatcataaacctgtggcaggaagtaggaaaagcaatgtatgcccctcccatcagaggacaaattagatgttcatcaaatattacaggactgctattaacaa']"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_origin = []\n",
    "existe = re.findall(r'ORIGIN\\s+.*?(?=//)', locus, re.DOTALL)\n",
    "if existe:\n",
    "    for nuc in existe:\n",
    "        m = re.match( r'ORIGIN\\s+(.+)', nuc, re.DOTALL)\n",
    "        a = re.sub(r'\\s+', '', m.group(1) ) \n",
    "        num = re.sub(r'\\d+','', a)\n",
    "        seq_origin.append(num)\n",
    "seq_origin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[233, 353, 179, 312],\n",
       " [1, 7, 4, 8],\n",
       " [4, 4, 4, 9],\n",
       " [5, 8, 5, 3],\n",
       " [71, 36, 45, 32],\n",
       " [74, 38, 48, 33]]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_count = []\n",
    "for seq in seq_origin:\n",
    "    #contagem de nucleotidos\n",
    "    A= []\n",
    "    C= []\n",
    "    T= []\n",
    "    G= []\n",
    "    for n in seq:\n",
    "        if n == 'a':\n",
    "            A.append('a')\n",
    "        if n == 'c':\n",
    "            C.append('c')\n",
    "        if n == 't':\n",
    "            T.append('t')\n",
    "        if n == 'g':\n",
    "            G.append('g')\n",
    "        \n",
    "    a = len(A)\n",
    "    c = len(C)\n",
    "    t = len(T)\n",
    "    g = len(G)\n",
    "    seq_count.append([a,c,t,g])\n",
    "    #fazer append de um array\n",
    "seq_count\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1077, 20, 21, 21, 184, 193]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "length = []\n",
    "for seq in seq_origin:\n",
    "    length.append(len(seq))\n",
    "length"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "PUBMED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['183048', '1466314', '1496030', '16141072 ', '1614107', '16645617 ', '1466314', '1496030', '16141072 ', '1614107', '16645617 ', '1466314', '1496030', '16141072 ', '1614107', '16645617 ', '787013', '787013']\n",
      "[['183048'], ['1466314', '1496030', '16141072 ', '1614107', '16645617 '], ['1466314', '1496030', '16141072 ', '1614107', '16645617 '], ['1466314', '1496030', '16141072 ', '1614107', '16645617 '], ['787013'], ['787013']]\n"
     ]
    }
   ],
   "source": [
    "pub_list = []\n",
    "existe = re.findall(r'PUBMED\\s+.*?(?=REFERENCE|COMMENT)', locus, re.DOTALL)\n",
    "# print(existe)\n",
    "if existe:\n",
    "    for pubm in existe:\n",
    "        m = re.match( r'PUBMED\\s+(.+)?(?=\\w)', pubm, re.DOTALL )\n",
    "        pub_list.append(re.sub(r'\\s+\\w+.*', ' ', m.group(1) ) )\n",
    "print(pub_list)\n",
    "\n",
    "#procurar os ficheiros que tem o campo PUBMED, se não tiver vai valor vazio na lista (para fazer a relação genbank-PM)\n",
    "file_locus_pub_list = []\n",
    "for x in locus_list:\n",
    "    aux_pub_list = []\n",
    "    existe = re.findall(r'PUBMED\\s+.*?(?=REFERENCE|COMMENT)', x, re.DOTALL)\n",
    "    # print(existe)\n",
    "    if existe:\n",
    "        for pubm in existe:\n",
    "            m = re.match( r'PUBMED\\s+(.+)?(?=\\w)', pubm, re.DOTALL )\n",
    "\n",
    "            aux_pub_list.append(re.sub(r'\\s+\\w+.*', ' ', m.group(1) ) )\n",
    "    else:\n",
    "        aux_pub_list.append('')\n",
    "    file_locus_pub_list.append(aux_pub_list)\n",
    "\n",
    "print(file_locus_pub_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['https://pubmed.ncbi.nlm.nih.gov/183048/', 'https://pubmed.ncbi.nlm.nih.gov/1466314/', 'https://pubmed.ncbi.nlm.nih.gov/1496030/', 'https://pubmed.ncbi.nlm.nih.gov/16141072 /', 'https://pubmed.ncbi.nlm.nih.gov/1614107/', 'https://pubmed.ncbi.nlm.nih.gov/16645617 /', 'https://pubmed.ncbi.nlm.nih.gov/1466314/', 'https://pubmed.ncbi.nlm.nih.gov/1496030/', 'https://pubmed.ncbi.nlm.nih.gov/16141072 /', 'https://pubmed.ncbi.nlm.nih.gov/1614107/', 'https://pubmed.ncbi.nlm.nih.gov/16645617 /', 'https://pubmed.ncbi.nlm.nih.gov/1466314/', 'https://pubmed.ncbi.nlm.nih.gov/1496030/', 'https://pubmed.ncbi.nlm.nih.gov/16141072 /', 'https://pubmed.ncbi.nlm.nih.gov/1614107/', 'https://pubmed.ncbi.nlm.nih.gov/16645617 /', 'https://pubmed.ncbi.nlm.nih.gov/787013/', 'https://pubmed.ncbi.nlm.nih.gov/787013/']\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "def url_get_id_pm(item):\n",
    "    url_id_pm = []\n",
    "    for id_pm in item:\n",
    "        url = f\"https://pubmed.ncbi.nlm.nih.gov/{id_pm}/\"\n",
    "        url_id_pm.append(url)\n",
    "    return url_id_pm\n",
    "\n",
    "pm = \"\"\n",
    "soup = []\n",
    "content_pm = []\n",
    "urls = url_get_id_pm(pub_list)\n",
    "print(urls)\n",
    "\n",
    "for url in urls:\n",
    "\tr2 = requests.get(url)\n",
    "\tpm += r2.content.decode('utf-8')\n",
    "\tsoup.append(BeautifulSoup(r2.content, 'html.parser'))\n",
    "\tcontent_pm.append(r2.content.decode('utf-8'))\n",
    "\n",
    "#print(soup)\n",
    "#print(content_pm)\n",
    "\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "AUTHOR, JOURNAL, TITLE, AFFILIATION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "from Bio import SeqIO\n",
    "from Bio import Entrez\n",
    "from Bio import Medline\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[\"[Short recapitulation for the next thirsty spell: our kidneys - the body's water police].\", 'Therapeutic management of triglycerides: an international perspective.', '[The relation between venous stasis and the occurrence of pain].', 'The transcriptional landscape of the mammalian genome.', '[The fate of corneal transplants from the Zurich eye bank].', 'Genome-wide analysis of mammalian promoter architecture and evolution.', 'A comparison between conventional gingivectomy and a non-surgical regime in the treatment of periodontitis.']\n",
      "['-', 'Current recommendations from various international expert committees generally concur in their definitions of borderline and high triglyceride levels, with small but important differences between recommendations in the definition of normal levels. However, population-based data on triglyceride levels are poorly developed in most countries, making difficult any international comparisons of prevalences of hypertriglyceridemia using the new definitions. However, it is probable that there should be considerable differences in the prevalence of hypertriglyceridemia, probably due to a mixture of genetic and environmental influences. The management of hypertriglyceridemia must continue to emphasize the detection and correction of secondary causes, even though the specific secondary causes may vary between countries. Dietary and exercise interventions must deal with local customs and resources, including striking international differences in alcohol consumption. Pharmacologic therapies will likely increase in use if they follow the trends in countries with available data. Although various drugs are available, nicotinic acid and fibric acid derivatives remain the drugs of choice. Considerably more research is needed to describe these international differences in etiology, prevalence and management practices of hypertriglyceridemia.', 'Venous stasis is a situation encountered commonly in varicose disorders. The potential implications of this decrease in oxygen levels in terms of the status of the cells of the vein were assessed. When endothelial cells are subjected to hypoxia, there is stimulation of the cells which shows itself as increased synthesis of prostaglandins and of PAF (Platelet Activating Factor). The synthesis of these typical mediators of inflammation results from activation by the calcium of phospholipase A2 which releases the arachidonic acid of phospholipids and this increase in intracellular calcium results itself from a fall in efficacy of calcium pumps due to the fall in ATP caused by hypoxia. Thus the fall in oxygen leads to the production of mediators of inflammation which activate leucocytes and result in local micro-inflammation which can be very rapidly eliminated if the circulation is restored but which can also cause irreversible damage to the vein by changes in venous tissue due to activated leucocytes which release proteases and free radicals after having penetrated the intima of the vein. These processes offer an explanation for the histological changes seen in varicose veins and the onset of localised pain during the development of such disease.', \"This study describes comprehensive polling of transcription start and termination sites and analysis of previously unidentified full-length complementary DNAs derived from the mouse genome. We identify the 5' and 3' boundaries of 181,047 transcripts with extensive variation in transcripts arising from alternative promoter usage, splicing, and polyadenylation. There are 16,247 new mouse protein-coding transcripts, including 5154 encoding previously unidentified proteins. Genomic mapping of the transcriptome reveals transcriptional forests, with overlapping transcription on both strands, separated by deserts in which few transcripts are observed. The data provide a comprehensive platform for the comparative analysis of mammalian transcriptional regulation in differentiation and development.\", \"In a retrospective study, we investigated the survival of transplanted corneal material, which had been sent between 1988 and 1989 from the Zurich Eye Bank to both domestic and foreign physicians and clinics. A questionnaire was used to determine diagnosis, transplant survival, cause of any opacification, the occurrence of problems of epithelialisation, loosening of sutures, as well as vascularization of the host cornea. Of a total of 416 corneas, 327 or 79% could be evaluated. The mean follow-up periods of the various diagnostic groups ranged from 11 to 20 months, with a range of 1 to 35 months. After 18 months, the rate of clear transplants was 96% in the keratokonus group. This rate was significantly better than that of the bullous keratopathy group (77%, p less than 0.013) or that of all other diagnosis groups (72%, p less than 0.001). The difference to the 81% survival rate of the group with Fuchs' endothelial dystrophy was not significant. The most frequent cause of transplant opacification was primary transplant failure. Analysis of possible risk factors further confirmed that transplant opacification occurs more frequent in the presence of vascularization of the host cornea.\", \"Mammalian promoters can be separated into two classes, conserved TATA box-enriched promoters, which initiate at a well-defined site, and more plastic, broad and evolvable CpG-rich promoters. We have sequenced tags corresponding to several hundred thousand transcription start sites (TSSs) in the mouse and human genomes, allowing precise analysis of the sequence architecture and evolution of distinct promoter classes. Different tissues and families of genes differentially use distinct types of promoters. Our tagging methods allow quantitative analysis of promoter usage in different tissues and show that differentially regulated alternative TSSs are a common feature in protein-coding genes and commonly generate alternative N termini. Among the TSSs, we identified new start sites associated with the majority of exons and with 3' UTRs. These data permit genome-scale identification of tissue-specific promoters and analysis of the cis-acting elements associated with them.\", 'The treatment of periodontitis by scaling and oral hygiene instruction was compared with the same regime augmented by conventional gingivectomy. The procedures were allocated randomly to contralateral sides of the maxillary arches of 28 patients. Following both regimes there was significant reduction in Plaque Index, Gingival Index, crevicular fluid and depths of pockets. There was slight gain in attachment on the non-surgical sides, and a minor loss of attachment on the surgical sides. Comparing the two regimes, over the second half of the study there was no significant difference in Plaque Index or crevicular fluid; however, on the surgical sides there was significantly greater reduction in Gingival Index and depths of pockets. Residual pockets after either technique were associated with more inflammation than where sulcus depths were less than 2 mm. Retrospective comparisons of the initial depths of pockets with the final gingival status indicated that, when initial depths of pockets had been 3 mm or over, surgical treatment was more effective than non-surgical in reducing the final values of both depths of pockets and inflammation; this finding was statistically significant for some, but not all of the parameters. Treatment of shallow pockets by either technique was more effective than the treatment of deep pockets. Final Plaque Index values of zero were associated with significantly less inflammation and pocketing than where plaque deposits could be detected.']\n",
      "[['Baranowsky W'], ['Pearson TA'], ['Remacle J', 'Arnould T', 'Michiels C'], ['Carninci P', 'Kasukawa T', 'Katayama S', 'Gough J', 'Frith MC', 'Maeda N', 'Oyama R', 'Ravasi T', 'Lenhard B', 'Wells C', 'Kodzius R', 'Shimokawa K', 'Bajic VB', 'Brenner SE', 'Batalov S', 'Forrest AR', 'Zavolan M', 'Davis MJ', 'Wilming LG', 'Aidinis V', 'Allen JE', 'Ambesi-Impiombato A', 'Apweiler R', 'Aturaliya RN', 'Bailey TL', 'Bansal M', 'Baxter L', 'Beisel KW', 'Bersano T', 'Bono H', 'Chalk AM', 'Chiu KP', 'Choudhary V', 'Christoffels A', 'Clutterbuck DR', 'Crowe ML', 'Dalla E', 'Dalrymple BP', 'de Bono B', 'Della Gatta G', 'di Bernardo D', 'Down T', 'Engstrom P', 'Fagiolini M', 'Faulkner G', 'Fletcher CF', 'Fukushima T', 'Furuno M', 'Futaki S', 'Gariboldi M', 'Georgii-Hemming P', 'Gingeras TR', 'Gojobori T', 'Green RE', 'Gustincich S', 'Harbers M', 'Hayashi Y', 'Hensch TK', 'Hirokawa N', 'Hill D', 'Huminiecki L', 'Iacono M', 'Ikeo K', 'Iwama A', 'Ishikawa T', 'Jakt M', 'Kanapin A', 'Katoh M', 'Kawasawa Y', 'Kelso J', 'Kitamura H', 'Kitano H', 'Kollias G', 'Krishnan SP', 'Kruger A', 'Kummerfeld SK', 'Kurochkin IV', 'Lareau LF', 'Lazarevic D', 'Lipovich L', 'Liu J', 'Liuni S', 'McWilliam S', 'Madan Babu M', 'Madera M', 'Marchionni L', 'Matsuda H', 'Matsuzawa S', 'Miki H', 'Mignone F', 'Miyake S', 'Morris K', 'Mottagui-Tabar S', 'Mulder N', 'Nakano N', 'Nakauchi H', 'Ng P', 'Nilsson R', 'Nishiguchi S', 'Nishikawa S', 'Nori F', 'Ohara O', 'Okazaki Y', 'Orlando V', 'Pang KC', 'Pavan WJ', 'Pavesi G', 'Pesole G', 'Petrovsky N', 'Piazza S', 'Reed J', 'Reid JF', 'Ring BZ', 'Ringwald M', 'Rost B', 'Ruan Y', 'Salzberg SL', 'Sandelin A', 'Schneider C', 'Schonbach C', 'Sekiguchi K', 'Semple CA', 'Seno S', 'Sessa L', 'Sheng Y', 'Shibata Y', 'Shimada H', 'Shimada K', 'Silva D', 'Sinclair B', 'Sperling S', 'Stupka E', 'Sugiura K', 'Sultana R', 'Takenaka Y', 'Taki K', 'Tammoja K', 'Tan SL', 'Tang S', 'Taylor MS', 'Tegner J', 'Teichmann SA', 'Ueda HR', 'van Nimwegen E', 'Verardo R', 'Wei CL', 'Yagi K', 'Yamanishi H', 'Zabarovsky E', 'Zhu S', 'Zimmer A', 'Hide W', 'Bult C', 'Grimmond SM', 'Teasdale RD', 'Liu ET', 'Brusic V', 'Quackenbush J', 'Wahlestedt C', 'Mattick JS', 'Hume DA', 'Kai C', 'Sasaki D', 'Tomaru Y', 'Fukuda S', 'Kanamori-Katayama M', 'Suzuki M', 'Aoki J', 'Arakawa T', 'Iida J', 'Imamura K', 'Itoh M', 'Kato T', 'Kawaji H', 'Kawagashira N', 'Kawashima T', 'Kojima M', 'Kondo S', 'Konno H', 'Nakano K', 'Ninomiya N', 'Nishio T', 'Okada M', 'Plessy C', 'Shibata K', 'Shiraki T', 'Suzuki S', 'Tagami M', 'Waki K', 'Watahiki A', 'Okamura-Oho Y', 'Suzuki H', 'Kawai J', 'Hayashizaki Y'], ['Hoppeler T', 'Ganzfried R', 'Gloor B'], ['Carninci P', 'Sandelin A', 'Lenhard B', 'Katayama S', 'Shimokawa K', 'Ponjavic J', 'Semple CA', 'Taylor MS', 'Engstrom PG', 'Frith MC', 'Forrest AR', 'Alkema WB', 'Tan SL', 'Plessy C', 'Kodzius R', 'Ravasi T', 'Kasukawa T', 'Fukuda S', 'Kanamori-Katayama M', 'Kitazume Y', 'Kawaji H', 'Kai C', 'Nakamura M', 'Konno H', 'Nakano K', 'Mottagui-Tabar S', 'Arner P', 'Chesi A', 'Gustincich S', 'Persichetti F', 'Suzuki H', 'Grimmond SM', 'Wells CA', 'Orlando V', 'Wahlestedt C', 'Liu ET', 'Harbers M', 'Kawai J', 'Bajic VB', 'Hume DA', 'Hayashizaki Y'], ['Waite IM']]\n",
      "['-', ['Mary Imogene Bassett Research Institute, Cooperstown, New York 13326.'], ['Laboratoire de Biochimie Cellulaire, Facultes Universitaires Notre-Dame de la Paix, Bruxelles, Belgique.'], '-', ['Augenklinik des Universitatsspitals Zurich.'], ['Genome Exploration Research Group, RIKEN Genomic Sciences Center, RIKEN Yokohama Institute, 1-7-22 Suehiro-cho, Tsurumi-ku, Yokohama, Kanagawa, 230-0045, Japan.'], '-']\n",
      "['Krankenpflege (Frankf). 1976 Jun;30(6):193-4.', 'Am J Cardiol. 1992 Dec 14;70(19):26H-31H. doi: 10.1016/0002-9149(92)91087-k.', 'Phlebologie. 1992 Jan-Mar;45(1):33-7; discussion 38-9.', 'Science. 2005 Sep 2;309(5740):1559-63. doi: 10.1126/science.1112014.', 'Klin Monbl Augenheilkd. 1992 May;200(5):388-91. doi: 10.1055/s-2008-1045772.', 'Nat Genet. 2006 Jun;38(6):626-35. doi: 10.1038/ng1789. Epub 2006 Apr 28.', 'J Clin Periodontol. 1976 Aug;3(3):173-85. doi: 10.1111/j.1600-051x.1976.tb01865.x.']\n"
     ]
    }
   ],
   "source": [
    "Ids=[]\n",
    "title_list = []\n",
    "abstract = []\n",
    "author = []\n",
    "affiliation = []\n",
    "journal = []\n",
    "database = 'PubMed'\n",
    "Entrez.email= \"pg49836@alunos.uminho.pt\"\n",
    "# idlist= pub_list\n",
    "idlist = []\n",
    "handle = Entrez.efetch(db=database, id=pub_list, rettype=\"medline\", retmode=\"text\") \n",
    "records = Medline.parse(handle)\n",
    "description=[]\n",
    "for info in records:\n",
    "    title_list.append(info.get(\"TI\", \"-\"))\n",
    "    author.append(info.get(\"AU\", \"-\"))\n",
    "    journal.append(info.get(\"SO\", \"-\"))\n",
    "    affiliation.append(info.get(\"AD\", \"-\") )\n",
    "    abstract.append(info.get(\"AB\", \"-\"))\n",
    "    print()\n",
    "print(title_list)\n",
    "print(abstract)\n",
    "print(author)\n",
    "print(affiliation)\n",
    "print(journal)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "CONNECTION TO DATABASE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INSERT INTO LOCUS (Locus_name, Sequence_length, Molecule_type, GB_division, Source, Organism, Accession) VALUES (\"X56237\", \"1077\", \"mRNA linear\", \"PLN\", \"Neurospora crassa \", \"Neurospora crassa\", \"X56237 \"), (\"AAADK0077081\", \"20\", \"RNA linear\", \"ROD\", \"Mus musculus (house mouse) \", \"Mus musculus\", \"AAADK0077081 \"), (\"AAADK0017277\", \"21\", \"RNA linear\", \"ROD\", \"Mus musculus (house mouse) \", \"Mus musculus\", \"AAADK0017277 \"), (\"AAADK0016387\", \"21\", \"RNA linear\", \"ROD\", \"Mus musculus (house mouse) \", \"Mus musculus\", \"AAADK0016387 \"), (\"HIV1LH22M\", \"184\", \"DNA linear\", \"VRL\", \"Human immunodeficiency virus 1 (HIV-1) \", \"Human immunodeficiency virus 1\", \"L33068 \"), (\"HIV1LH21M\", \"193\", \"DNA linear\", \"VRL\", \"Human immunodeficiency virus 1 (HIV-1) \", \"Human immunodeficiency virus 1\", \"L33067 \")\n",
      "6 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"geo.di.uminho.pt\",\n",
    "    user =\"bioinformatica\",\n",
    "    password =\"20221207\",\n",
    "    database =\"Project_TJM\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`LOCUS` (\n",
    "        `Id_locus` INT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n",
    "        `Accession` VARCHAR(100) NOT NULL,\n",
    "        `Sequence_length` INT NULL,\n",
    "        `Molecule_type` VARCHAR(20) NULL,\n",
    "        `GB_division` VARCHAR(10) NULL,\n",
    "        `Source` VARCHAR(200) NULL,\n",
    "        `Organism` VARCHAR(200) NULL,\n",
    "        `Locus_name` VARCHAR(45) NULL\n",
    "        );\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    sql_loc = \"INSERT INTO LOCUS (Locus_name, Sequence_length, Molecule_type, GB_division, Source, Organism, Accession) VALUES\"\n",
    "    loc_name_list = list(dict.fromkeys(loc_name_list))  \n",
    "    for index in range(len(loc_name_list)):\n",
    "       \n",
    "        select_ids_loc = f\"SELECT Id_Locus FROM LOCUS WHERE Locus_name = \\\"{loc_name_list[index]}\\\"\"\n",
    "        Cursor.execute(select_ids_loc)\n",
    "        myresult_locus = Cursor.fetchall()\n",
    "        id_loc = None\n",
    "        for res_loc, in myresult_locus:\n",
    "            id_loc = res_loc\n",
    "        if id_loc is None: \n",
    "            if index == len(loc_name_list)-1:\n",
    "                sql_loc += f\" (\\\"{loc_name_list[index]}\\\", \\\"{length[index]}\\\", \\\"{mol_type[index]}\\\", \\\"{gb_div[index]}\\\", \\\"{source_list[index]}\\\", \\\"{org_list[index]}\\\", \\\"{acc_list[index]}\\\")\"\n",
    "            else:\n",
    "                sql_loc += f\" (\\\"{loc_name_list[index]}\\\", \\\"{length[index]}\\\", \\\"{mol_type[index]}\\\", \\\"{gb_div[index]}\\\", \\\"{source_list[index]}\\\", \\\"{org_list[index]}\\\", \\\"{acc_list[index]}\\\"),\"\n",
    "    print(sql_loc)\n",
    "    if sql_loc != \"INSERT INTO LOCUS (Locus_name, Sequence_length, Molecule_type, GB_division, Source, Organism, Accession) VALUES\": \n",
    "        if sql_loc.endswith(\",\"):  \n",
    "            sql_loc = sql_loc[:-1]\n",
    "        Cursor.execute(sql_loc)\n",
    "\n",
    "    DataBase.commit()\n",
    "\n",
    "    print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"geo.di.uminho.pt\",\n",
    "    user =\"bioinformatica\",\n",
    "    password =\"20221207\",\n",
    "    database =\"Project_TJM\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`GENBANK` (\n",
    "        `Id_GB` INT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n",
    "        `GB_id` VARCHAR(45) NOT NULL,\n",
    "        `Definition` VARCHAR(400) NULL,\n",
    "        `Modification_date` VARCHAR(11) NULL,\n",
    "        `Version` VARCHAR(20) NOT NULL UNIQUE,\n",
    "        `LOCUS_Id_locus` INT NOT NULL,\n",
    "        `ORIGIN_Id_origin` INT NULL\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    sql_gb = \"INSERT INTO GENBANK (GB_Id, Modification_date, Definition, Version, LOCUS_Id_locus, ORIGIN_Id_origin) VALUES\"\n",
    "    \n",
    "\n",
    "    for index in range(len(version_list)):\n",
    "        sql_insert_origin = \"INSERT INTO `Project_TJM`.ORIGIN (Sequence, Length, Adenine, Citosine, Timine , Guanine) VALUES\"\n",
    "        select_ids_origin = f\"SELECT Id_Origin FROM ORIGIN WHERE Sequence = \\\"{seq_origin[index]}\\\"\"\n",
    "        Cursor.execute(select_ids_origin)\n",
    "        myresult_origin = Cursor.fetchall()\n",
    "        id_ori = None\n",
    "        for res_ori, in myresult_origin:\n",
    "            id_ori = res_ori\n",
    "\n",
    "        if id_ori is None: \n",
    "            if index == len(loc_name_list)-1:\n",
    "                sql_insert_origin += f\" (\\\"{seq_origin[index]}\\\", \\\"{length[index]}\\\", \\\"{seq_count[index][0]}\\\", \\\"{seq_count[index][1]}\\\", \\\"{seq_count[index][2]}\\\",  \\\"{seq_count[index][3]}\\\")\"\n",
    "            else:\n",
    "                sql_insert_origin += f\" (\\\"{seq_origin[index]}\\\", \\\"{length[index]}\\\", \\\"{seq_count[index][0]}\\\", \\\"{seq_count[index][1]}\\\", \\\"{seq_count[index][2]}\\\",  \\\"{seq_count[index][3]}\\\"),\"\n",
    "\n",
    "        if sql_insert_origin != \"INSERT INTO `Project_TJM`.ORIGIN (Sequence, Length, Adenine, Citosine, Timine , Guanine) VALUES\":\n",
    "            # select_ids_origin += f\" (\\\"{seq_origin[index]}\\\", \\\"{length[index]}\\\", \\\"{seq_count[index][0]}\\\", \\\"{seq_count[index][1]}\\\", \\\"{seq_count[index][2]}\\\",  \\\"{seq_count[index][3]}\\\")\"\n",
    "            if sql_insert_origin.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                sql_insert_origin = sql_insert_origin[:-1] #remove utimo caracter da string\n",
    "            Cursor.execute(sql_insert_origin)\n",
    "\n",
    "            # print(Cursor.lastrowid)  #buscar o id do origin inserido\n",
    "            id_inserted_origin = Cursor.lastrowid\n",
    "        else:\n",
    "            id_inserted_origin = id_ori\n",
    "\n",
    "\n",
    "        #fazer a verificação se existe locus repetido (evitar duplicados)\n",
    "        select_locus = f\"SELECT Id_locus from LOCUS where Locus_name = '{loc_name_list[index]}'\"\n",
    "        Cursor.execute(select_locus)\n",
    "        myresult = Cursor.fetchall()\n",
    "        id_locus = None\n",
    "        for res, in myresult:\n",
    "            id_locus = res\n",
    "        \n",
    "        if id_locus is not None:\n",
    "            select_gb = f\"SELECT Id_GB from GENBANK where GB_Id = '{loc_name_list[index]}'\"\n",
    "            Cursor.execute(select_gb)\n",
    "            myresult_gb = Cursor.fetchall()\n",
    "            id_genbank = None\n",
    "            for res_gb, in myresult_gb:\n",
    "                id_genbank = res_gb\n",
    "            \n",
    "            if id_genbank is None:\n",
    "                if index == len(version_list)-1:\n",
    "                    sql_gb += f\" (\\\"{loc_name_list[index]}\\\", \\\"{mod_date[index]}\\\", \\\"{defi_list[index]}\\\", \\\"{version_list[index]}\\\", \\\"{id_locus}\\\", \\\"{id_inserted_origin}\\\")\"\n",
    "                else:\n",
    "                    sql_gb += f\" (\\\"{loc_name_list[index]}\\\", \\\"{mod_date[index]}\\\", \\\"{defi_list[index]}\\\", \\\"{version_list[index]}\\\", \\\"{id_locus}\\\", \\\"{id_inserted_origin}\\\"),\"\n",
    "                \n",
    "    if sql_gb != \"INSERT INTO GENBANK (GB_Id, Modification_date, Definition, Version, LOCUS_Id_locus, ORIGIN_Id_origin) VALUES\":\n",
    "\n",
    "        if sql_gb.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "            sql_gb = sql_gb[:-1] #remove utimo caracter da string\n",
    "            \n",
    "        Cursor.execute(sql_gb)\n",
    "\n",
    "        DataBase.commit()\n",
    "\n",
    "        print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INSERT INTO PUBMED (PM_ID, Journal, Title, Abstract) VALUES (\"183048\", \"Krankenpflege (Frankf). 1976 Jun;30(6):193-4.\", \"[Short recapitulation for the next thirsty spell: our kidneys - the body's water police].\", \"-\"), (\"1466314\", \"Am J Cardiol. 1992 Dec 14;70(19):26H-31H. doi: 10.1016/0002-9149(92)91087-k.\", \"Therapeutic management of triglycerides: an international perspective.\", \"Current recommendations from various international expert committees generally concur in their definitions of borderline and high triglyceride levels, with small but important differences between recommendations in the definition of normal levels. However, population-based data on triglyceride levels are poorly developed in most countries, making difficult any international comparisons of prevalences of hypertriglyceridemia using the new definitions. However, it is probable that there should be considerable differences in the prevalence of hypertriglyceridemia, probably due to a mixture of genetic and environmental influences. The management of hypertriglyceridemia must continue to emphasize the detection and correction of secondary causes, even though the specific secondary causes may vary between countries. Dietary and exercise interventions must deal with local customs and resources, including striking international differences in alcohol consumption. Pharmacologic therapies will likely increase in use if they follow the trends in countries with available data. Although various drugs are available, nicotinic acid and fibric acid derivatives remain the drugs of choice. Considerably more research is needed to describe these international differences in etiology, prevalence and management practices of hypertriglyceridemia.\"), (\"1496030\", \"Phlebologie. 1992 Jan-Mar;45(1):33-7; discussion 38-9.\", \"[The relation between venous stasis and the occurrence of pain].\", \"Venous stasis is a situation encountered commonly in varicose disorders. The potential implications of this decrease in oxygen levels in terms of the status of the cells of the vein were assessed. When endothelial cells are subjected to hypoxia, there is stimulation of the cells which shows itself as increased synthesis of prostaglandins and of PAF (Platelet Activating Factor). The synthesis of these typical mediators of inflammation results from activation by the calcium of phospholipase A2 which releases the arachidonic acid of phospholipids and this increase in intracellular calcium results itself from a fall in efficacy of calcium pumps due to the fall in ATP caused by hypoxia. Thus the fall in oxygen leads to the production of mediators of inflammation which activate leucocytes and result in local micro-inflammation which can be very rapidly eliminated if the circulation is restored but which can also cause irreversible damage to the vein by changes in venous tissue due to activated leucocytes which release proteases and free radicals after having penetrated the intima of the vein. These processes offer an explanation for the histological changes seen in varicose veins and the onset of localised pain during the development of such disease.\"), (\"16141072 \", \"Science. 2005 Sep 2;309(5740):1559-63. doi: 10.1126/science.1112014.\", \"The transcriptional landscape of the mammalian genome.\", \"This study describes comprehensive polling of transcription start and termination sites and analysis of previously unidentified full-length complementary DNAs derived from the mouse genome. We identify the 5' and 3' boundaries of 181,047 transcripts with extensive variation in transcripts arising from alternative promoter usage, splicing, and polyadenylation. There are 16,247 new mouse protein-coding transcripts, including 5154 encoding previously unidentified proteins. Genomic mapping of the transcriptome reveals transcriptional forests, with overlapping transcription on both strands, separated by deserts in which few transcripts are observed. The data provide a comprehensive platform for the comparative analysis of mammalian transcriptional regulation in differentiation and development.\"), (\"1614107\", \"Klin Monbl Augenheilkd. 1992 May;200(5):388-91. doi: 10.1055/s-2008-1045772.\", \"[The fate of corneal transplants from the Zurich eye bank].\", \"In a retrospective study, we investigated the survival of transplanted corneal material, which had been sent between 1988 and 1989 from the Zurich Eye Bank to both domestic and foreign physicians and clinics. A questionnaire was used to determine diagnosis, transplant survival, cause of any opacification, the occurrence of problems of epithelialisation, loosening of sutures, as well as vascularization of the host cornea. Of a total of 416 corneas, 327 or 79% could be evaluated. The mean follow-up periods of the various diagnostic groups ranged from 11 to 20 months, with a range of 1 to 35 months. After 18 months, the rate of clear transplants was 96% in the keratokonus group. This rate was significantly better than that of the bullous keratopathy group (77%, p less than 0.013) or that of all other diagnosis groups (72%, p less than 0.001). The difference to the 81% survival rate of the group with Fuchs' endothelial dystrophy was not significant. The most frequent cause of transplant opacification was primary transplant failure. Analysis of possible risk factors further confirmed that transplant opacification occurs more frequent in the presence of vascularization of the host cornea.\"), (\"16645617 \", \"Nat Genet. 2006 Jun;38(6):626-35. doi: 10.1038/ng1789. Epub 2006 Apr 28.\", \"Genome-wide analysis of mammalian promoter architecture and evolution.\", \"Mammalian promoters can be separated into two classes, conserved TATA box-enriched promoters, which initiate at a well-defined site, and more plastic, broad and evolvable CpG-rich promoters. We have sequenced tags corresponding to several hundred thousand transcription start sites (TSSs) in the mouse and human genomes, allowing precise analysis of the sequence architecture and evolution of distinct promoter classes. Different tissues and families of genes differentially use distinct types of promoters. Our tagging methods allow quantitative analysis of promoter usage in different tissues and show that differentially regulated alternative TSSs are a common feature in protein-coding genes and commonly generate alternative N termini. Among the TSSs, we identified new start sites associated with the majority of exons and with 3' UTRs. These data permit genome-scale identification of tissue-specific promoters and analysis of the cis-acting elements associated with them.\"), (\"787013\", \"J Clin Periodontol. 1976 Aug;3(3):173-85. doi: 10.1111/j.1600-051x.1976.tb01865.x.\", \"A comparison between conventional gingivectomy and a non-surgical regime in the treatment of periodontitis.\", \"The treatment of periodontitis by scaling and oral hygiene instruction was compared with the same regime augmented by conventional gingivectomy. The procedures were allocated randomly to contralateral sides of the maxillary arches of 28 patients. Following both regimes there was significant reduction in Plaque Index, Gingival Index, crevicular fluid and depths of pockets. There was slight gain in attachment on the non-surgical sides, and a minor loss of attachment on the surgical sides. Comparing the two regimes, over the second half of the study there was no significant difference in Plaque Index or crevicular fluid; however, on the surgical sides there was significantly greater reduction in Gingival Index and depths of pockets. Residual pockets after either technique were associated with more inflammation than where sulcus depths were less than 2 mm. Retrospective comparisons of the initial depths of pockets with the final gingival status indicated that, when initial depths of pockets had been 3 mm or over, surgical treatment was more effective than non-surgical in reducing the final values of both depths of pockets and inflammation; this finding was statistically significant for some, but not all of the parameters. Treatment of shallow pockets by either technique was more effective than the treatment of deep pockets. Final Plaque Index values of zero were associated with significantly less inflammation and pocketing than where plaque deposits could be detected.\")\n",
      "7 record inserted.\n",
      "['183048', '1466314', '1496030', '16141072 ', '1614107', '16645617 ', '787013']\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"geo.di.uminho.pt\",\n",
    "    user =\"bioinformatica\",\n",
    "    password =\"20221207\",\n",
    "    database =\"Project_TJM\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`PUBMED` (\n",
    "        `Id_PM` INT NOT NULL AUTO_INCREMENT,\n",
    "        `PM_ID` INT NOT NULL,\n",
    "        `Journal` VARCHAR(100) NULL,\n",
    "        `Title` VARCHAR(500) NULL,\n",
    "        `Abstract` LONGTEXT NULL,\n",
    "        PRIMARY KEY (`Id_PM`)\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    \n",
    "    #print(journal)\n",
    "    #print(len(title_list))\n",
    "    #print(len(abstract))\n",
    "    #print(len(pub_list))\n",
    "    sql_pm = \"INSERT INTO PUBMED (PM_ID, Journal, Title, Abstract) VALUES\"\n",
    "    pub_list = list(dict.fromkeys(pub_list))  #tirar os duplicados (também n apanha o ultimo)\n",
    "    #print(pub_list)\n",
    "    for index in range(len(pub_list)):\n",
    "        select_pm = f\"SELECT Id_PM from PUBMED where PM_Id = '{pub_list[index]}'\"\n",
    "        Cursor.execute(select_pm)\n",
    "        myresult_pubmed = Cursor.fetchall()\n",
    "        id_pubmed = None\n",
    "        for res_pm, in myresult_pubmed:\n",
    "            id_pubmed = res_pm\n",
    "        \n",
    "        if id_pubmed is None:\n",
    "            if index == len(pub_list)-1:\n",
    "                sql_pm += f\" (\\\"{pub_list[index]}\\\", \\\"{journal[index]}\\\", \\\"{title_list[index]}\\\", \\\"{abstract[index]}\\\")\"\n",
    "            else:\n",
    "                sql_pm += f\" (\\\"{pub_list[index]}\\\", \\\"{journal[index]}\\\", \\\"{title_list[index]}\\\", \\\"{abstract[index]}\\\"),\"\n",
    "        \n",
    "    if sql_pm != \"INSERT INTO PUBMED (PM_ID, Journal, Title, Abstract) VALUES\":\n",
    "        # select_ids_origin += f\" (\\\"{seq_origin[index]}\\\", \\\"{length[index]}\\\", \\\"{seq_count[index][0]}\\\", \\\"{seq_count[index][1]}\\\", \\\"{seq_count[index][2]}\\\",  \\\"{seq_count[index][3]}\\\")\"\n",
    "        if sql_pm.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "            sql_pm = sql_pm[:-1] #remove utimo caracter da string\n",
    "        print(sql_pm)\n",
    "        Cursor.execute(sql_pm)\n",
    "\n",
    "        DataBase.commit()\n",
    "\n",
    "        print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    Cursor.close()\n",
    "    print(pub_list)\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6\n",
      "6\n",
      "[2]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"X56237\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"2\" and GENBANK_Id_GB = \"9\"\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (2, 9)\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (2, 9)\n",
      "1 record inserted.\n",
      "[3, 4, 5, 6, 7]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"AAADK0077081\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"3\" and GENBANK_Id_GB = \"10\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"4\" and GENBANK_Id_GB = \"10\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"5\" and GENBANK_Id_GB = \"10\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"6\" and GENBANK_Id_GB = \"10\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"7\" and GENBANK_Id_GB = \"10\"\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (3, 10), (4, 10), (5, 10), (6, 10), (7, 10)\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (3, 10), (4, 10), (5, 10), (6, 10), (7, 10)\n",
      "5 record inserted.\n",
      "[3, 4, 5, 6, 7]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"AAADK0017277\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"3\" and GENBANK_Id_GB = \"11\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"4\" and GENBANK_Id_GB = \"11\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"5\" and GENBANK_Id_GB = \"11\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"6\" and GENBANK_Id_GB = \"11\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"7\" and GENBANK_Id_GB = \"11\"\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (3, 11), (4, 11), (5, 11), (6, 11), (7, 11)\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (3, 11), (4, 11), (5, 11), (6, 11), (7, 11)\n",
      "5 record inserted.\n",
      "[3, 4, 5, 6, 7]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"AAADK0016387\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"3\" and GENBANK_Id_GB = \"12\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"4\" and GENBANK_Id_GB = \"12\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"5\" and GENBANK_Id_GB = \"12\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"6\" and GENBANK_Id_GB = \"12\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"7\" and GENBANK_Id_GB = \"12\"\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (3, 12), (4, 12), (5, 12), (6, 12), (7, 12)\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (3, 12), (4, 12), (5, 12), (6, 12), (7, 12)\n",
      "5 record inserted.\n",
      "[8]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"HIV1LH22M\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"8\" and GENBANK_Id_GB = \"13\"\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (8, 13)\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (8, 13)\n",
      "1 record inserted.\n",
      "[8]\n",
      "SELECT Id_GB FROM GENBANK WHERE GB_ID= \"HIV1LH21M\"\n",
      "SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \"8\" and GENBANK_Id_GB = \"14\"\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (8, 14)\n",
      "INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES (8, 14)\n",
      "1 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"geo.di.uminho.pt\",\n",
    "    user =\"bioinformatica\",\n",
    "    password =\"20221207\",\n",
    "    database =\"Project_TJM\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    '''TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`AUTHORS` (\n",
    "        `Id_Author` INT NOT NULL AUTO_INCREMENT,\n",
    "        `Name` VARCHAR(45) NULL,\n",
    "        `AFFILIATION_Id_affiliation` INT NULL,\n",
    "        PRIMARY KEY (`Id_Author`)\n",
    "    )\n",
    "    \"\"\" '''\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`PM_GB_relationship` (\n",
    "        `PUBMED_Id_PM` INT NOT NULL,\n",
    "        `GENBANK_Id_GB` INT NOT NULL,\n",
    "        PRIMARY KEY (`PUBMED_Id_PM`, `GENBANK_Id_GB`)\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    print(len(file_locus_pub_list))\n",
    "    print(len(loc_name_list))\n",
    "\n",
    "    for index_ids in range(len(file_locus_pub_list)):\n",
    "        '''for index in range(len(author[index_ids])):\n",
    "            #verificar se existe autor (evitar duplicados)\n",
    "            select_ids_auth = f\"SELECT Id_Author FROM AUTHORS WHERE Name = \\\"{author[index_ids]}\\\"\"\n",
    "            Cursor.execute(select_ids_auth)\n",
    "            myresult_auth = Cursor.fetchall()\n",
    "            id_auth = None\n",
    "            for res, in myresult_auth:\n",
    "                id_affiliation = res'''\n",
    "\n",
    "        if file_locus_pub_list[index_ids][0] != '':\n",
    "            db_id_pm = []\n",
    "            for pm_ids in file_locus_pub_list[index_ids]:\n",
    "                select_pm_gb = f\"SELECT Id_PM FROM PUBMED WHERE PM_ID = \\\"{pm_ids}\\\"\"\n",
    "                \n",
    "                #print(select_pm_gb)\n",
    "                Cursor.execute(select_pm_gb)\n",
    "                myresult_pm = Cursor.fetchall()\n",
    "                id_pm = None\n",
    "                for res, in myresult_pm:\n",
    "                    id_pm = res\n",
    "                db_id_pm.append(id_pm)\n",
    "\n",
    "            print(db_id_pm)\n",
    "            select_gb_pm = f\"SELECT Id_GB FROM GENBANK WHERE GB_ID= \\\"{loc_name_list[index_ids]}\\\"\"\n",
    "            print(select_gb_pm)\n",
    "            Cursor.execute(select_gb_pm)\n",
    "            myresult_gb = Cursor.fetchall()\n",
    "            id_gb = None\n",
    "            for res, in myresult_gb:\n",
    "                id_gb = res\n",
    "\n",
    "            index = 0\n",
    "            sql_pm_gb_relation = \"INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES\" \n",
    "            for id_pm in db_id_pm:\n",
    "                if id_gb is not None and id_pm is not None:      \n",
    "                    #verificar se existe relação entre ids\n",
    "                    select_rel_gb_pm = f\"SELECT PUBMED_Id_PM, GENBANK_Id_GB FROM PM_GB_relationship WHERE PUBMED_Id_PM = \\\"{id_pm}\\\" and GENBANK_Id_GB = \\\"{id_gb}\\\"\"\n",
    "                    print(select_rel_gb_pm)\n",
    "                    Cursor.execute(select_rel_gb_pm)\n",
    "                    myresult_rel_pm_gb = Cursor.fetchall()\n",
    "                    id_rel_pm_gb = None\n",
    "                    for res_PUBMED_Id_PM,res_GENBANK_Id_GB, in myresult_rel_pm_gb:   #apenas selecionamos um dos id pq se não houver relação serão ambos null\n",
    "                        id_rel_pm_gb = res_PUBMED_Id_PM     \n",
    "\n",
    "                    #se null:\n",
    "                    if id_rel_pm_gb is None:\n",
    "                        if index == len(db_id_pm)-1:\n",
    "                            sql_pm_gb_relation += f\" ({id_pm}, {id_gb})\"\n",
    "                        else:\n",
    "                            sql_pm_gb_relation += f\" ({id_pm}, {id_gb}),\"\n",
    "                index += 1\n",
    "\n",
    "            print(sql_pm_gb_relation)\n",
    "            if sql_pm_gb_relation != \"INSERT INTO PM_GB_relationship (PUBMED_Id_PM, GENBANK_Id_GB) VALUES\":\n",
    "                #remove ultimo caracter se este for uma \",\"\n",
    "                if sql_pm_gb_relation.endswith(\",\"):\n",
    "                    sql_pm_gb_relation = sql_pm_gb_relation[:-1] #remove utimo caracter da string\n",
    "\n",
    "                print(sql_pm_gb_relation)\n",
    "                Cursor.execute(sql_pm_gb_relation)\n",
    "                DataBase.commit()\n",
    "                print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Baranowsky W\", 1)\n",
      "INSERT INTO `Project_TJM`.AFFILIATION (Description) VALUES (\"Mary Imogene Bassett Research Institute, Cooperstown, New York 13326.\")\n",
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Pearson TA\", 2)\n",
      "INSERT INTO `Project_TJM`.AFFILIATION (Description) VALUES (\"Laboratoire de Biochimie Cellulaire, Facultes Universitaires Notre-Dame de la Paix, Bruxelles, Belgique.\")\n",
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Remacle J\", 3), (\"Arnould T\", null), (\"Michiels C\", null)\n",
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Carninci P\", 1), (\"Kasukawa T\", null), (\"Katayama S\", null), (\"Gough J\", null), (\"Frith MC\", null), (\"Maeda N\", null), (\"Oyama R\", null), (\"Ravasi T\", null), (\"Lenhard B\", null), (\"Wells C\", null), (\"Kodzius R\", null), (\"Shimokawa K\", null), (\"Bajic VB\", null), (\"Brenner SE\", null), (\"Batalov S\", null), (\"Forrest AR\", null), (\"Zavolan M\", null), (\"Davis MJ\", null), (\"Wilming LG\", null), (\"Aidinis V\", null), (\"Allen JE\", null), (\"Ambesi-Impiombato A\", null), (\"Apweiler R\", null), (\"Aturaliya RN\", null), (\"Bailey TL\", null), (\"Bansal M\", null), (\"Baxter L\", null), (\"Beisel KW\", null), (\"Bersano T\", null), (\"Bono H\", null), (\"Chalk AM\", null), (\"Chiu KP\", null), (\"Choudhary V\", null), (\"Christoffels A\", null), (\"Clutterbuck DR\", null), (\"Crowe ML\", null), (\"Dalla E\", null), (\"Dalrymple BP\", null), (\"de Bono B\", null), (\"Della Gatta G\", null), (\"di Bernardo D\", null), (\"Down T\", null), (\"Engstrom P\", null), (\"Fagiolini M\", null), (\"Faulkner G\", null), (\"Fletcher CF\", null), (\"Fukushima T\", null), (\"Furuno M\", null), (\"Futaki S\", null), (\"Gariboldi M\", null), (\"Georgii-Hemming P\", null), (\"Gingeras TR\", null), (\"Gojobori T\", null), (\"Green RE\", null), (\"Gustincich S\", null), (\"Harbers M\", null), (\"Hayashi Y\", null), (\"Hensch TK\", null), (\"Hirokawa N\", null), (\"Hill D\", null), (\"Huminiecki L\", null), (\"Iacono M\", null), (\"Ikeo K\", null), (\"Iwama A\", null), (\"Ishikawa T\", null), (\"Jakt M\", null), (\"Kanapin A\", null), (\"Katoh M\", null), (\"Kawasawa Y\", null), (\"Kelso J\", null), (\"Kitamura H\", null), (\"Kitano H\", null), (\"Kollias G\", null), (\"Krishnan SP\", null), (\"Kruger A\", null), (\"Kummerfeld SK\", null), (\"Kurochkin IV\", null), (\"Lareau LF\", null), (\"Lazarevic D\", null), (\"Lipovich L\", null), (\"Liu J\", null), (\"Liuni S\", null), (\"McWilliam S\", null), (\"Madan Babu M\", null), (\"Madera M\", null), (\"Marchionni L\", null), (\"Matsuda H\", null), (\"Matsuzawa S\", null), (\"Miki H\", null), (\"Mignone F\", null), (\"Miyake S\", null), (\"Morris K\", null), (\"Mottagui-Tabar S\", null), (\"Mulder N\", null), (\"Nakano N\", null), (\"Nakauchi H\", null), (\"Ng P\", null), (\"Nilsson R\", null), (\"Nishiguchi S\", null), (\"Nishikawa S\", null), (\"Nori F\", null), (\"Ohara O\", null), (\"Okazaki Y\", null), (\"Orlando V\", null), (\"Pang KC\", null), (\"Pavan WJ\", null), (\"Pavesi G\", null), (\"Pesole G\", null), (\"Petrovsky N\", null), (\"Piazza S\", null), (\"Reed J\", null), (\"Reid JF\", null), (\"Ring BZ\", null), (\"Ringwald M\", null), (\"Rost B\", null), (\"Ruan Y\", null), (\"Salzberg SL\", null), (\"Sandelin A\", null), (\"Schneider C\", null), (\"Schonbach C\", null), (\"Sekiguchi K\", null), (\"Semple CA\", null), (\"Seno S\", null), (\"Sessa L\", null), (\"Sheng Y\", null), (\"Shibata Y\", null), (\"Shimada H\", null), (\"Shimada K\", null), (\"Silva D\", null), (\"Sinclair B\", null), (\"Sperling S\", null), (\"Stupka E\", null), (\"Sugiura K\", null), (\"Sultana R\", null), (\"Takenaka Y\", null), (\"Taki K\", null), (\"Tammoja K\", null), (\"Tan SL\", null), (\"Tang S\", null), (\"Taylor MS\", null), (\"Tegner J\", null), (\"Teichmann SA\", null), (\"Ueda HR\", null), (\"van Nimwegen E\", null), (\"Verardo R\", null), (\"Wei CL\", null), (\"Yagi K\", null), (\"Yamanishi H\", null), (\"Zabarovsky E\", null), (\"Zhu S\", null), (\"Zimmer A\", null), (\"Hide W\", null), (\"Bult C\", null), (\"Grimmond SM\", null), (\"Teasdale RD\", null), (\"Liu ET\", null), (\"Brusic V\", null), (\"Quackenbush J\", null), (\"Wahlestedt C\", null), (\"Mattick JS\", null), (\"Hume DA\", null), (\"Kai C\", null), (\"Sasaki D\", null), (\"Tomaru Y\", null), (\"Fukuda S\", null), (\"Kanamori-Katayama M\", null), (\"Suzuki M\", null), (\"Aoki J\", null), (\"Arakawa T\", null), (\"Iida J\", null), (\"Imamura K\", null), (\"Itoh M\", null), (\"Kato T\", null), (\"Kawaji H\", null), (\"Kawagashira N\", null), (\"Kawashima T\", null), (\"Kojima M\", null), (\"Kondo S\", null), (\"Konno H\", null), (\"Nakano K\", null), (\"Ninomiya N\", null), (\"Nishio T\", null), (\"Okada M\", null), (\"Plessy C\", null), (\"Shibata K\", null), (\"Shiraki T\", null), (\"Suzuki S\", null), (\"Tagami M\", null), (\"Waki K\", null), (\"Watahiki A\", null), (\"Okamura-Oho Y\", null), (\"Suzuki H\", null), (\"Kawai J\", null), (\"Hayashizaki Y\", null)\n",
      "INSERT INTO `Project_TJM`.AFFILIATION (Description) VALUES (\"Augenklinik des Universitatsspitals Zurich.\")\n",
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Hoppeler T\", 4), (\"Ganzfried R\", null), (\"Gloor B\", null)\n",
      "INSERT INTO `Project_TJM`.AFFILIATION (Description) VALUES (\"Genome Exploration Research Group, RIKEN Genomic Sciences Center, RIKEN Yokohama Institute, 1-7-22 Suehiro-cho, Tsurumi-ku, Yokohama, Kanagawa, 230-0045, Japan.\")\n",
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Ponjavic J\", null), (\"Engstrom PG\", null), (\"Alkema WB\", null), (\"Kitazume Y\", null), (\"Nakamura M\", null), (\"Arner P\", null), (\"Chesi A\", null), (\"Persichetti F\", null), (\"Wells CA\", null),\n",
      "INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES (\"Waite IM\", 1)\n",
      "1 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"geo.di.uminho.pt\",\n",
    "    user =\"bioinformatica\",\n",
    "    password =\"20221207\",\n",
    "    database =\"Project_TJM\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`AUTHORS` (\n",
    "        `Id_Author` INT NOT NULL AUTO_INCREMENT,\n",
    "        `Name` VARCHAR(45) NULL,\n",
    "        `AFFILIATION_Id_affiliation` INT NULL,\n",
    "        PRIMARY KEY (`Id_Author`)\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "\n",
    "    for index_pubmed in range(len(pub_list)):\n",
    "        \n",
    "        affiliation_ids = []\n",
    "        for index in range(len(affiliation[index_pubmed])):\n",
    "\n",
    "            #verificar se existe afiliação (evitar duplicados)\n",
    "            select_affiliation = f\"SELECT Id_affiliation FROM `Project_TJM`.AFFILIATION WHERE Description = \\\"{affiliation[index_pubmed][index]}\\\"\"\n",
    "            Cursor.execute(select_affiliation)\n",
    "            myresult_af = Cursor.fetchall()\n",
    "            id_affiliation = None\n",
    "            for res, in myresult_af:\n",
    "                id_affiliation = res\n",
    "\n",
    "\n",
    "            if id_affiliation is not None:\n",
    "                affiliation_ids.append(id_affiliation)\n",
    "            else:\n",
    "                sql_insert_affiliation = \"INSERT INTO `Project_TJM`.AFFILIATION (Description) VALUES\" \n",
    "                sql_insert_affiliation += f\" (\\\"{affiliation[index_pubmed][index]}\\\")\"\n",
    "                print(sql_insert_affiliation)\n",
    "                Cursor.execute(sql_insert_affiliation)\n",
    "                # print(Cursor.lastrowid)  #buscar o id do affiliation inserido\n",
    "                affiliation_ids.append(Cursor.lastrowid)\n",
    "\n",
    "\n",
    "        sql = \"INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES\"\n",
    "        \n",
    "        for index in range(len(author[index_pubmed])):\n",
    "            select_author = f\"SELECT Id_Author FROM AUTHORS WHERE Name = \\\"{author[index_pubmed][index]}\\\"\"\n",
    "            Cursor.execute(select_author)\n",
    "            myresult_author = Cursor.fetchall()\n",
    "            id_author = None\n",
    "            for res, in myresult_author:\n",
    "                id_author = res\n",
    "\n",
    "\n",
    "            if id_author is None:\n",
    "                if index > len(affiliation_ids)-1:\n",
    "                    if index == len(author[index_pubmed])-1:\n",
    "                        sql += f\" (\\\"{author[index_pubmed][index]}\\\", null)\"\n",
    "                    else:\n",
    "                        sql += f\" (\\\"{author[index_pubmed][index]}\\\", null),\"\n",
    "                else:\n",
    "                    if index == len(author[index_pubmed])-1:\n",
    "                        sql += f\" (\\\"{author[index_pubmed][index]}\\\", {affiliation_ids[index]})\"\n",
    "                    else:\n",
    "                        sql += f\" (\\\"{author[index_pubmed][index]}\\\", {affiliation_ids[index]}),\"\n",
    "    \n",
    "        print(sql)\n",
    "        if sql != \"INSERT INTO AUTHORS (Name, AFFILIATION_Id_affiliation ) VALUES\":\n",
    "                #remove ultimo caracter se este for uma \",\"\n",
    "            if sql.endswith(\",\"):\n",
    "                sql = sql[:-1]\n",
    "\n",
    "            Cursor.execute(sql)\n",
    "\n",
    "    DataBase.commit()\n",
    "\n",
    "    print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6\n",
      "6\n",
      "2\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"2\" and AUTHORS_Id_Author = \"3\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (2, 3)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (2, 3)\n",
      "1 record inserted.\n",
      "3\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"3\" and AUTHORS_Id_Author = \"4\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (3, 4)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (3, 4)\n",
      "1 record inserted.\n",
      "4\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"4\" and AUTHORS_Id_Author = \"5\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"4\" and AUTHORS_Id_Author = \"6\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"4\" and AUTHORS_Id_Author = \"7\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (4, 5), (4, 6), (4, 7)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (4, 5), (4, 6), (4, 7)\n",
      "3 record inserted.\n",
      "5\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"8\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"9\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"10\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"11\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"12\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"13\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"14\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"15\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"16\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"17\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"18\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"19\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"20\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"21\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"22\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"23\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"24\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"25\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"26\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"27\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"28\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"29\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"30\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"31\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"32\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"33\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"34\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"35\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"36\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"37\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"38\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"39\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"40\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"41\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"42\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"43\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"44\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"45\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"46\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"47\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"48\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"49\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"50\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"51\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"52\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"53\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"54\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"55\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"56\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"57\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"58\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"59\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"60\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"61\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"62\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"63\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"64\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"65\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"66\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"67\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"68\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"69\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"70\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"71\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"72\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"73\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"74\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"75\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"76\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"77\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"78\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"79\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"80\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"81\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"82\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"83\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"84\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"85\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"86\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"87\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"88\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"89\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"90\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"91\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"92\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"93\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"94\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"95\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"96\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"97\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"98\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"99\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"100\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"101\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"102\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"103\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"104\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"105\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"106\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"107\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"108\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"109\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"110\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"111\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"112\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"113\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"114\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"115\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"116\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"117\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"118\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"119\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"120\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"121\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"122\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"123\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"124\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"125\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"126\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"127\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"128\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"129\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"130\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"131\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"132\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"133\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"134\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"135\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"136\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"137\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"138\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"139\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"140\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"141\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"142\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"143\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"144\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"145\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"146\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"147\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"148\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"149\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"150\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"151\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"152\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"153\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"154\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"155\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"156\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"157\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"158\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"159\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"160\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"161\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"162\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"163\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"164\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"165\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"166\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"167\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"168\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"169\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"170\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"171\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"172\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"173\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"174\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"175\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"176\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"177\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"178\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"179\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"180\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"181\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"182\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"183\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"184\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"185\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"186\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"187\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"188\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"189\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"190\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"191\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"192\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"193\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"194\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"195\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"196\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"197\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"198\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"199\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"200\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"5\" and AUTHORS_Id_Author = \"201\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (5, 8), (5, 9), (5, 10), (5, 11), (5, 12), (5, 13), (5, 14), (5, 15), (5, 16), (5, 17), (5, 18), (5, 19), (5, 20), (5, 21), (5, 22), (5, 23), (5, 24), (5, 25), (5, 26), (5, 27), (5, 28), (5, 29), (5, 30), (5, 31), (5, 32), (5, 33), (5, 34), (5, 35), (5, 36), (5, 37), (5, 38), (5, 39), (5, 40), (5, 41), (5, 42), (5, 43), (5, 44), (5, 45), (5, 46), (5, 47), (5, 48), (5, 49), (5, 50), (5, 51), (5, 52), (5, 53), (5, 54), (5, 55), (5, 56), (5, 57), (5, 58), (5, 59), (5, 60), (5, 61), (5, 62), (5, 63), (5, 64), (5, 65), (5, 66), (5, 67), (5, 68), (5, 69), (5, 70), (5, 71), (5, 72), (5, 73), (5, 74), (5, 75), (5, 76), (5, 77), (5, 78), (5, 79), (5, 80), (5, 81), (5, 82), (5, 83), (5, 84), (5, 85), (5, 86), (5, 87), (5, 88), (5, 89), (5, 90), (5, 91), (5, 92), (5, 93), (5, 94), (5, 95), (5, 96), (5, 97), (5, 98), (5, 99), (5, 100), (5, 101), (5, 102), (5, 103), (5, 104), (5, 105), (5, 106), (5, 107), (5, 108), (5, 109), (5, 110), (5, 111), (5, 112), (5, 113), (5, 114), (5, 115), (5, 116), (5, 117), (5, 118), (5, 119), (5, 120), (5, 121), (5, 122), (5, 123), (5, 124), (5, 125), (5, 126), (5, 127), (5, 128), (5, 129), (5, 130), (5, 131), (5, 132), (5, 133), (5, 134), (5, 135), (5, 136), (5, 137), (5, 138), (5, 139), (5, 140), (5, 141), (5, 142), (5, 143), (5, 144), (5, 145), (5, 146), (5, 147), (5, 148), (5, 149), (5, 150), (5, 151), (5, 152), (5, 153), (5, 154), (5, 155), (5, 156), (5, 157), (5, 158), (5, 159), (5, 160), (5, 161), (5, 162), (5, 163), (5, 164), (5, 165), (5, 166), (5, 167), (5, 168), (5, 169), (5, 170), (5, 171), (5, 172), (5, 173), (5, 174), (5, 175), (5, 176), (5, 177), (5, 178), (5, 179), (5, 180), (5, 181), (5, 182), (5, 183), (5, 184), (5, 185), (5, 186), (5, 187), (5, 188), (5, 189), (5, 190), (5, 191), (5, 192), (5, 193), (5, 194), (5, 195), (5, 196), (5, 197), (5, 198), (5, 199), (5, 200), (5, 201)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (5, 8), (5, 9), (5, 10), (5, 11), (5, 12), (5, 13), (5, 14), (5, 15), (5, 16), (5, 17), (5, 18), (5, 19), (5, 20), (5, 21), (5, 22), (5, 23), (5, 24), (5, 25), (5, 26), (5, 27), (5, 28), (5, 29), (5, 30), (5, 31), (5, 32), (5, 33), (5, 34), (5, 35), (5, 36), (5, 37), (5, 38), (5, 39), (5, 40), (5, 41), (5, 42), (5, 43), (5, 44), (5, 45), (5, 46), (5, 47), (5, 48), (5, 49), (5, 50), (5, 51), (5, 52), (5, 53), (5, 54), (5, 55), (5, 56), (5, 57), (5, 58), (5, 59), (5, 60), (5, 61), (5, 62), (5, 63), (5, 64), (5, 65), (5, 66), (5, 67), (5, 68), (5, 69), (5, 70), (5, 71), (5, 72), (5, 73), (5, 74), (5, 75), (5, 76), (5, 77), (5, 78), (5, 79), (5, 80), (5, 81), (5, 82), (5, 83), (5, 84), (5, 85), (5, 86), (5, 87), (5, 88), (5, 89), (5, 90), (5, 91), (5, 92), (5, 93), (5, 94), (5, 95), (5, 96), (5, 97), (5, 98), (5, 99), (5, 100), (5, 101), (5, 102), (5, 103), (5, 104), (5, 105), (5, 106), (5, 107), (5, 108), (5, 109), (5, 110), (5, 111), (5, 112), (5, 113), (5, 114), (5, 115), (5, 116), (5, 117), (5, 118), (5, 119), (5, 120), (5, 121), (5, 122), (5, 123), (5, 124), (5, 125), (5, 126), (5, 127), (5, 128), (5, 129), (5, 130), (5, 131), (5, 132), (5, 133), (5, 134), (5, 135), (5, 136), (5, 137), (5, 138), (5, 139), (5, 140), (5, 141), (5, 142), (5, 143), (5, 144), (5, 145), (5, 146), (5, 147), (5, 148), (5, 149), (5, 150), (5, 151), (5, 152), (5, 153), (5, 154), (5, 155), (5, 156), (5, 157), (5, 158), (5, 159), (5, 160), (5, 161), (5, 162), (5, 163), (5, 164), (5, 165), (5, 166), (5, 167), (5, 168), (5, 169), (5, 170), (5, 171), (5, 172), (5, 173), (5, 174), (5, 175), (5, 176), (5, 177), (5, 178), (5, 179), (5, 180), (5, 181), (5, 182), (5, 183), (5, 184), (5, 185), (5, 186), (5, 187), (5, 188), (5, 189), (5, 190), (5, 191), (5, 192), (5, 193), (5, 194), (5, 195), (5, 196), (5, 197), (5, 198), (5, 199), (5, 200), (5, 201)\n",
      "194 record inserted.\n",
      "6\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"6\" and AUTHORS_Id_Author = \"202\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"6\" and AUTHORS_Id_Author = \"203\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"6\" and AUTHORS_Id_Author = \"204\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (6, 202), (6, 203), (6, 204)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (6, 202), (6, 203), (6, 204)\n",
      "3 record inserted.\n",
      "7\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"8\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"125\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"16\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"10\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"19\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"205\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"129\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"147\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"206\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"12\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"23\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"207\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"145\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"191\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"18\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"15\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"9\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"172\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"173\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"208\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"181\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"169\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"209\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"186\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"187\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"100\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"210\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"211\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"62\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"212\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"199\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"161\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"213\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"111\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"166\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"163\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"63\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"200\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"20\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"168\"\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"7\" and AUTHORS_Id_Author = \"201\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (7, 8), (7, 125), (7, 16), (7, 10), (7, 19), (7, 205), (7, 129), (7, 147), (7, 206), (7, 12), (7, 23), (7, 207), (7, 145), (7, 191), (7, 18), (7, 15), (7, 9), (7, 172), (7, 173), (7, 208), (7, 181), (7, 169), (7, 209), (7, 186), (7, 187), (7, 100), (7, 210), (7, 211), (7, 62), (7, 212), (7, 199), (7, 161), (7, 213), (7, 111), (7, 166), (7, 163), (7, 63), (7, 200), (7, 20), (7, 168), (7, 201)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (7, 8), (7, 125), (7, 16), (7, 10), (7, 19), (7, 205), (7, 129), (7, 147), (7, 206), (7, 12), (7, 23), (7, 207), (7, 145), (7, 191), (7, 18), (7, 15), (7, 9), (7, 172), (7, 173), (7, 208), (7, 181), (7, 169), (7, 209), (7, 186), (7, 187), (7, 100), (7, 210), (7, 211), (7, 62), (7, 212), (7, 199), (7, 161), (7, 213), (7, 111), (7, 166), (7, 163), (7, 63), (7, 200), (7, 20), (7, 168), (7, 201)\n",
      "41 record inserted.\n",
      "8\n",
      "SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \"8\" and AUTHORS_Id_Author = \"214\"\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (8, 214)\n",
      "INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES (8, 214)\n",
      "1 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"geo.di.uminho.pt\",\n",
    "    user =\"bioinformatica\",\n",
    "    password =\"20221207\",\n",
    "    database =\"Project_TJM\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`PM_AUTHORS_relationship` (\n",
    "        `PUBMED_Id_PM` INT NOT NULL,\n",
    "        `AUTHORS_Id_Author` INT NOT NULL,\n",
    "        PRIMARY KEY (`PUBMED_Id_PM`, `AUTHORS_Id_Author`)\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "    print(len(file_locus_pub_list))\n",
    "    print(len(loc_name_list))\n",
    "\n",
    "    for index_pm in range(len(pub_list)):\n",
    "\n",
    "        if pub_list[index_pm] != '':\n",
    "            select_pm_at = f\"SELECT Id_PM FROM PUBMED WHERE PM_ID = \\\"{pub_list[index_pm]}\\\"\"\n",
    "        \n",
    "            Cursor.execute(select_pm_at)\n",
    "            myresult_pm_at = Cursor.fetchall()\n",
    "            id_pm_at = None\n",
    "            for res, in myresult_pm_at:\n",
    "                id_pm_at = res\n",
    "\n",
    "            print(id_pm_at)\n",
    "            author_id_list = []\n",
    "            for xx in author[index_pm]:\n",
    "                select_at_pm = f\"SELECT Id_Author FROM AUTHORS WHERE Name= \\\"{xx}\\\"\"\n",
    "                Cursor.execute(select_at_pm)\n",
    "                myresult_at = Cursor.fetchall()\n",
    "                id_at = None\n",
    "                for res, in myresult_at:\n",
    "                    id_at = res\n",
    "                author_id_list.append(id_at)\n",
    "\n",
    "            index = 0\n",
    "            sql_at_pm_relation = \"INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES\" \n",
    "            author_id_list = list(dict.fromkeys(author_id_list))\n",
    "            for id_at in author_id_list:\n",
    "                if id_at is not None and id_pm_at is not None:      \n",
    "                    #verificar se existe relação entre ids\n",
    "                    select_rel_at_pm = f\"SELECT PUBMED_Id_PM, AUTHORS_Id_Author FROM PM_AUTHORS_relationship WHERE PUBMED_Id_PM = \\\"{id_pm_at}\\\" and AUTHORS_Id_Author = \\\"{id_at}\\\"\"\n",
    "                    print(select_rel_at_pm)\n",
    "                    Cursor.execute(select_rel_at_pm)\n",
    "                    myresult_rel_pm_at = Cursor.fetchall()\n",
    "                    id_rel_pm_at = None\n",
    "                    for res_PUBMED_Id_PM,res_AUTHORS_Id_Author, in myresult_rel_pm_at:   #apenas selecionamos um dos id pq se não houver relação serão ambos null\n",
    "                        id_rel_pm_at = res_PUBMED_Id_PM     \n",
    "\n",
    "                    #se null:\n",
    "                    if id_rel_pm_at is None:\n",
    "                        if index == len(author_id_list)-1:\n",
    "                            sql_at_pm_relation += f\" ({id_pm_at}, {id_at})\"\n",
    "                        else:\n",
    "                            sql_at_pm_relation += f\" ({id_pm_at}, {id_at}),\"\n",
    "                index += 1\n",
    "\n",
    "            print(sql_at_pm_relation)\n",
    "            if sql_at_pm_relation != \"INSERT INTO PM_AUTHORS_relationship (PUBMED_Id_PM, AUTHORS_Id_Author) VALUES\":\n",
    "                #remove ultimo caracter se este for uma \",\"\n",
    "                if sql_at_pm_relation.endswith(\",\"):\n",
    "                    sql_at_pm_relation = sql_at_pm_relation[:-1] #remove utimo caracter da string\n",
    "\n",
    "                print(sql_at_pm_relation)\n",
    "                Cursor.execute(sql_at_pm_relation)\n",
    "                DataBase.commit()\n",
    "                print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\n",
      "[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []]\n",
      "['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']]\n",
      "1 record inserted.\n",
      "[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], [], ['[67:889](+)', ['NADH dehydrogenase subunit'], ['CAA39694.1'], ['MRAALRLLATATATVRPSARFLKPGSPTGLTGLGTHPSPRSALLYLYNHTLDKLKQIPEHSLYRQSAEALTKHRLAIVEQYVPDGYDAWQERARKLLEKHKSDLTARQFDGQHARLVEGPDGRAYFIRQMVPPQDWRDVEWDGAVLDPHFSWVQTGEDVVGAVKLEDSDKLLELDKIRESDPVAYRQGLRDLGIKMGGVVEDKSPVEWESEPPLSAEQIAEMEARIGSGLIEEVVQVAEGELKLVDIMTQARPWEALEEEAPEGQWTYFERKE']], [], [], []]\n",
      "[]\n",
      "1 record inserted.\n",
      "10\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "11\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "12\n",
      "[[]]\n",
      "[]\n",
      "1 record inserted.\n",
      "13\n",
      "[[], [], ['[<0:>184](+)', ['envelope glycoprotein'], ['AAA73551.1'], ['CNTSRLFNSTWNGTEGLPDITLPCRIKQIINMWQEVGKAMYAPPIKGQIRCSSNITGLLLT']]]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>184](+)', ['envelope glycoprotein'], ['AAA73551.1'], ['CNTSRLFNSTWNGTEGLPDITLPCRIKQIINMWQEVGKAMYAPPIKGQIRCSSNITGLLLT']]]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>184](+)', ['envelope glycoprotein'], ['AAA73551.1'], ['CNTSRLFNSTWNGTEGLPDITLPCRIKQIINMWQEVGKAMYAPPIKGQIRCSSNITGLLLT']]]\n",
      "['[<0:>184](+)', ['envelope glycoprotein'], ['AAA73551.1'], ['CNTSRLFNSTWNGTEGLPDITLPCRIKQIINMWQEVGKAMYAPPIKGQIRCSSNITGLLLT']]\n",
      "1 record inserted.\n",
      "14\n",
      "[[], [], ['[<0:>193](+)', ['envelope glycoprotein'], ['AAA73550.1'], ['CNTSQLFNSTWNGTEGFPNTENITLPCRIKQIINLWQEVGKAMYAPPIRGQIRCSSNITGLLLT']]]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>193](+)', ['envelope glycoprotein'], ['AAA73550.1'], ['CNTSQLFNSTWNGTEGFPNTENITLPCRIKQIINLWQEVGKAMYAPPIRGQIRCSSNITGLLLT']]]\n",
      "[]\n",
      "1 record inserted.\n",
      "[[], [], ['[<0:>193](+)', ['envelope glycoprotein'], ['AAA73550.1'], ['CNTSQLFNSTWNGTEGFPNTENITLPCRIKQIINLWQEVGKAMYAPPIRGQIRCSSNITGLLLT']]]\n",
      "['[<0:>193](+)', ['envelope glycoprotein'], ['AAA73550.1'], ['CNTSQLFNSTWNGTEGFPNTENITLPCRIKQIINLWQEVGKAMYAPPIRGQIRCSSNITGLLLT']]\n",
      "1 record inserted.\n"
     ]
    }
   ],
   "source": [
    "import mysql.connector as SQLC\n",
    "try: \n",
    "    DataBase = SQLC.connect(\n",
    "    host =\"geo.di.uminho.pt\",\n",
    "    user =\"bioinformatica\",\n",
    "    password =\"20221207\",\n",
    "    database =\"Project_TJM\"\n",
    "    )\n",
    "    Cursor = DataBase.cursor()\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`FEATURES` (\n",
    "        `Id_F` INT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n",
    "        `Gene` VARCHAR(45) NULL,\n",
    "        `Source_location` VARCHAR(45) NULL\n",
    "    \n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "\n",
    "    TableName =\"\"\"CREATE TABLE IF NOT EXISTS `Project_TJM`.`CDS` (\n",
    "        `Id_CDS` INT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n",
    "        `Translation_seq` longtext  NULL,\n",
    "        `Location` VARCHAR(45)  NULL,\n",
    "        `Product` VARCHAR(45)  NULL,\n",
    "        `Protein_ID` VARCHAR(45) NULL\n",
    "    )\n",
    "    \"\"\"\n",
    "    Cursor.execute(TableName)\n",
    "\n",
    "    for index_gb in range(len(loc_name_list)):\n",
    "\n",
    "        #get genbank id\n",
    "        select_gb = f\"SELECT Id_GB FROM GENBANK WHERE GB_ID= \\\"{loc_name_list[index_gb]}\\\"\"\n",
    "        Cursor.execute(select_gb)\n",
    "        myresult_gb = Cursor.fetchall()\n",
    "        id_gb = None\n",
    "        for res, in myresult_gb:\n",
    "            id_gb = res\n",
    "        \n",
    "        print(id_gb)\n",
    "        if id_gb is not None:\n",
    "            for index_ft in range(len(source[index_gb])):\n",
    "                if source[index_gb][index_ft] != '':  #verificar se existe \"source\" na bd\n",
    "                    sql_insert_source = \"INSERT INTO FEATURES (Gene, Source_location) VALUES\"\n",
    "                    select_ft = f\"SELECT Id_F FROM FEATURES WHERE Source_location = \\\"{source[index_gb][index_ft]}\\\"\"\n",
    "            \n",
    "                    Cursor.execute(select_ft)\n",
    "                    myresult_ft = Cursor.fetchall()\n",
    "                    id_ft = None\n",
    "                    for res_ft, in myresult_ft:\n",
    "                        id_ft = res_ft\n",
    "\n",
    "                    if id_ft is None: #se n existe, insere\n",
    "                        if len(gene[index_gb]) > 0:\n",
    "                            sql_insert_source += f\" (\\\"{gene[index_gb][index_ft]}\\\", \\\"{source[index_gb][index_ft]}\\\")\"\n",
    "                        else:\n",
    "                            sql_insert_source += f\" ( null, \\\"{source[index_gb][index_ft]}\\\")\"\n",
    "\n",
    "                        if sql_insert_source.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                            sql_insert_source = sql_insert_source[:-1] #remove utimo caracter da string\n",
    "                        Cursor.execute(sql_insert_source)\n",
    "\n",
    "                        # print(Cursor.lastrowid)  #buscar o id do feature inserido\n",
    "                        id_inserted_feature = Cursor.lastrowid\n",
    "                    else:\n",
    "                        id_inserted_feature = id_ft\n",
    "\n",
    "                    #relationship between genbank and feature\n",
    "                    sql_insert_gb_ft = \"INSERT INTO GB_FT_relationship (GENBANK_Id_GB, FEATURES_Id_F) VALUES\"\n",
    "                    select_gb_ft = f\"SELECT GENBANK_Id_GB FROM GB_FT_relationship WHERE GENBANK_Id_GB = {id_gb} and FEATURES_Id_F = {id_inserted_feature}\"\n",
    "            \n",
    "                    Cursor.execute(select_gb_ft)\n",
    "                    myresult_gb_ft = Cursor.fetchall()\n",
    "                    id_gb_ft = None\n",
    "                    for res_gb_ft, in myresult_gb_ft:\n",
    "                        id_gb_ft = res_gb_ft\n",
    "\n",
    "                    if id_gb_ft is None: #se n existe, insere\n",
    "                        sql_insert_gb_ft += f\" ({id_gb}, {id_inserted_feature})\"\n",
    "\n",
    "                        if sql_insert_gb_ft.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                            sql_insert_gb_ft = sql_insert_gb_ft[:-1] #remove utimo caracter da string\n",
    "                        Cursor.execute(sql_insert_gb_ft)\n",
    "\n",
    "            for index_cds in range(len(CDS[index_gb])):\n",
    "                print(CDS[index_gb])\n",
    "                print(CDS[index_gb][index_cds])\n",
    "                if CDS[index_gb][index_cds]:  #verificar se existe \"cds\" na bd\n",
    "                    sql_insert_cds = \"INSERT INTO CDS (Translation_seq, Location,Product,Protein_ID) VALUES\"\n",
    "                    select_cds = f\"SELECT Id_CDS FROM CDS WHERE Translation_seq = \\\"{CDS[index_gb][index_cds][3]}\\\"\"\n",
    "            \n",
    "                    Cursor.execute(select_cds)\n",
    "                    myresult_cds = Cursor.fetchall()\n",
    "                    id_cds = None\n",
    "                    for res_cds, in myresult_cds:\n",
    "                        id_cds = res_cds\n",
    "\n",
    "                    if id_cds is None: #se n existe, insere\n",
    "                        sql_insert_cds += f\" (\\\"{CDS[index_gb][index_cds][3]}\\\",\\\"{CDS[index_gb][index_cds][0]}\\\",\\\"{CDS[index_gb][index_cds][1]}\\\", \\\"{CDS[index_gb][index_cds][2]}\\\")\"\n",
    "\n",
    "                        if sql_insert_cds.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                            sql_insert_cds = sql_insert_cds[:-1] #remove utimo caracter da string\n",
    "                        Cursor.execute(sql_insert_cds)\n",
    "\n",
    "                        # print(Cursor.lastrowid)  #buscar o id do cds inserido\n",
    "                        id_inserted_cds = Cursor.lastrowid\n",
    "                    else:\n",
    "                        id_inserted_cds = id_cds\n",
    "                    \n",
    "                    #relationship between cds and genbank\n",
    "                    sql_insert_gb_cds = \"INSERT INTO CDS_GB_relationship (GENBANK_Id_GB, CDS_Id_CDS) VALUES\"\n",
    "                    select_gb_cds = f\"SELECT GENBANK_Id_GB FROM CDS_GB_relationship WHERE GENBANK_Id_GB = {id_gb} and CDS_Id_CDS = {id_inserted_cds}\"\n",
    "            \n",
    "                    Cursor.execute(select_gb_cds)\n",
    "                    myresult_gb_cds = Cursor.fetchall()\n",
    "                    id_gb_cds = None\n",
    "                    for res_gb_cds, in myresult_gb_cds:\n",
    "                        id_gb_ft = res_gb_cds\n",
    "\n",
    "                    if id_gb_ft is None: #se n existe, insere\n",
    "                        sql_insert_gb_cds += f\" ({id_gb}, {id_inserted_cds})\"\n",
    "\n",
    "                        if sql_insert_gb_cds.endswith(\",\"):  #remove ultimo caracter se este for uma \",\"\n",
    "                            sql_insert_gb_cds = sql_insert_gb_cds[:-1] #remove utimo caracter da string\n",
    "                        Cursor.execute(sql_insert_gb_cds)\n",
    "\n",
    "                DataBase.commit()\n",
    "                print(Cursor.rowcount, \"record inserted.\")\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "    Cursor.close()\n",
    "\n",
    "except SQLC.Error as error:\n",
    "    print(\"Failed to insert record into MySQL table {}\".format(error))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.5"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "fb4569285eef3a3450cb62085a5b1e0da4bce0af555edc33dcf29baf3acc1368"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
